% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
\documentclass[
]{article}
\usepackage{xcolor}
\usepackage[margin=1in]{geometry}
\usepackage{amsmath,amssymb}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math} % this also loads fontspec
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else
  % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx}
\makeatletter
\newsavebox\pandoc@box
\newcommand*\pandocbounded[1]{% scales image to fit in text height/width
  \sbox\pandoc@box{#1}%
  \Gscale@div\@tempa{\textheight}{\dimexpr\ht\pandoc@box+\dp\pandoc@box\relax}%
  \Gscale@div\@tempb{\linewidth}{\wd\pandoc@box}%
  \ifdim\@tempb\p@<\@tempa\p@\let\@tempa\@tempb\fi% select the smaller of both
  \ifdim\@tempa\p@<\p@\scalebox{\@tempa}{\usebox\pandoc@box}%
  \else\usebox{\pandoc@box}%
  \fi%
}
% Set default figure placement to htbp
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\usepackage{bookmark}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same}
\hypersetup{
  pdftitle={DEPM Project},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\title{DEPM Project}
\usepackage{etoolbox}
\makeatletter
\providecommand{\subtitle}[1]{% add subtitle to \maketitle
  \apptocmd{\@title}{\par {\large #1 \par}}{}{}
}
\makeatother
\subtitle{by Domenico Azzarito, Federico Lattanzio and Michele Pezza}
\author{}
\date{\vspace{-2.5em}}

\begin{document}
\maketitle

\textbf{Project Steps:}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Selection of the disease: Kidney renal papillary cell carcinoma (KIRP)
\item
  Differentially expressed genes (DEGs)
\item
  CoExpression Networks
\item
  Differential CoExpression Network
\item
  Patient Similarity Network
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# TCGA{-}KIRP:}
\FunctionTok{rm}\NormalTok{(}\AttributeTok{list =} \FunctionTok{ls}\NormalTok{())      }\CommentTok{\# clear workspace}
\FunctionTok{gc}\NormalTok{()                 }\CommentTok{\# garbage collection}
\end{Highlighting}
\end{Shaded}

The first step is to loading the required packages for the analysis:

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(BiocGenerics) }\CommentTok{\# Contains Bioconductor generics used internally by SummarizedExperiment and DESeq2}
\FunctionTok{library}\NormalTok{(DESeq2) }\CommentTok{\# Store RNA{-}Seq counts and normalize counts}
\FunctionTok{library}\NormalTok{(psych) }\CommentTok{\# correlation analysis + p{-}value adjustment}
\FunctionTok{library}\NormalTok{(NetworkToolbox) }\CommentTok{\# Additional functions for network analysis}
\FunctionTok{library}\NormalTok{(ggplot2) }\CommentTok{\# volcano plot, degree distribution plot and any additional plot}
\CommentTok{\# library(ggnet)    }
\FunctionTok{library}\NormalTok{(GGally) }\CommentTok{\# Package containing ggnet2(), essential function for plotting}
\FunctionTok{library}\NormalTok{(sna) }\CommentTok{\# functions for clustering coefficient, components..}
\FunctionTok{library}\NormalTok{(network) }\CommentTok{\# defines the network() object type}
\FunctionTok{library}\NormalTok{(TCGAbiolinks) }\CommentTok{\# queries the GDC portal and downloads TCGA data}
\FunctionTok{library}\NormalTok{(GenomicRanges)  }\CommentTok{\# Gene annotation is stored as GRanges}
\FunctionTok{library}\NormalTok{(SummarizedExperiment)  }\CommentTok{\# TCGA RNA{-}Seq is stored in a SummarizedExperiment object}
\FunctionTok{library}\NormalTok{(DT)  }\CommentTok{\# Interactive visualization}
\FunctionTok{library}\NormalTok{(igraph)}
\FunctionTok{library}\NormalTok{(maftools)}
\FunctionTok{library}\NormalTok{(cowplot)}
\FunctionTok{library}\NormalTok{(SNFtool)}
\FunctionTok{library}\NormalTok{(reshape2)}
\FunctionTok{library}\NormalTok{(poweRlaw)}
\FunctionTok{library}\NormalTok{(dplyr)}
\FunctionTok{library}\NormalTok{(ggraph)}
\CommentTok{\# sessionInfo()}
\end{Highlighting}
\end{Shaded}

\subsection{1. Download data}\label{download-data}

We proceed to download the required tumor RNA-Seq data (Primary Tumor)

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{proj }\OtherTok{\textless{}{-}} \StringTok{"TCGA{-}KIRP"}
\FunctionTok{dir.create}\NormalTok{(}\FunctionTok{file.path}\NormalTok{(proj))}
\end{Highlighting}
\end{Shaded}

RNA-Seq data: primary tumor samples

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{rna.query.C }\OtherTok{\textless{}{-}}\NormalTok{ TCGAbiolinks}\SpecialCharTok{::}\FunctionTok{GDCquery}\NormalTok{(}
  \AttributeTok{project =}\NormalTok{ proj,}
  \AttributeTok{data.category =} \StringTok{"Transcriptome Profiling"}\NormalTok{,}
  \AttributeTok{data.type =} \StringTok{"Gene Expression Quantification"}\NormalTok{,}
  \AttributeTok{workflow.type =} \StringTok{"STAR {-} Counts"}\NormalTok{,}
  \AttributeTok{sample.type =} \StringTok{"Primary Tumor"}
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Download the data from GDC to local cache folder ``GDCdata'' a

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{GDCdownload}\NormalTok{(}\AttributeTok{query =}\NormalTok{ rna.query.C, }\AttributeTok{directory =} \StringTok{"GDCdata"}\NormalTok{, }\AttributeTok{method =} \StringTok{"api"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Convert the downloaded files into a SummarizedExperiment object

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{rna.data.C }\OtherTok{\textless{}{-}} \FunctionTok{GDCprepare}\NormalTok{(rna.query.C, }\AttributeTok{directory =} \StringTok{"GDCdata"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Extract the raw count matrix (genes x tumor samples)

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{rna.expr.data.C }\OtherTok{\textless{}{-}} \FunctionTok{assay}\NormalTok{(rna.data.C)}
\end{Highlighting}
\end{Shaded}

Extract gene annotation (Ensembl IDs, gene symbols, chromosomes, etc.)

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{genes.info }\OtherTok{\textless{}{-}}\NormalTok{ BiocGenerics}\SpecialCharTok{::}\FunctionTok{as.data.frame}\NormalTok{(}\FunctionTok{rowRanges}\NormalTok{(rna.data.C))}
\end{Highlighting}
\end{Shaded}

Checks:

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(rna.query.C}\SpecialCharTok{$}\NormalTok{results[[}\DecValTok{1}\NormalTok{]]}\SpecialCharTok{$}\NormalTok{cases)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "TCGA-MH-A855-01A-11R-A355-07" "TCGA-P4-A5E6-01A-11R-A28H-07"
## [3] "TCGA-BQ-5875-01A-11R-1592-07" "TCGA-WN-A9G9-01A-12R-A37K-07"
## [5] "TCGA-HE-A5NH-01A-11R-A26U-07" "TCGA-F9-A8NY-01A-11R-A36F-07"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{rna.data.C}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## class: RangedSummarizedExperiment 
## dim: 60660 290 
## metadata(1): data_release
## assays(6): unstranded stranded_first ... fpkm_unstrand fpkm_uq_unstrand
## rownames(60660): ENSG00000000003.15 ENSG00000000005.6 ...
##   ENSG00000288674.1 ENSG00000288675.1
## rowData names(10): source type ... hgnc_id havana_gene
## colnames(290): TCGA-MH-A855-01A-11R-A355-07
##   TCGA-P4-A5E6-01A-11R-A28H-07 ... TCGA-IA-A40Y-01A-11R-A24Z-07
##   TCGA-J7-6720-01A-11R-2139-07
## colData names(209): barcode patient ... paper_VHL.copy..log2.normal.
##   paper_NF2.copy..log2.normal.
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{dim}\NormalTok{(rna.expr.data.C)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 60660   290
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(genes.info)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##                    seqnames     start       end  width strand source type score
## ENSG00000000003.15     chrX 100627108 100639991  12884      - HAVANA gene    NA
## ENSG00000000005.6      chrX 100584936 100599885  14950      + HAVANA gene    NA
## ENSG00000000419.13    chr20  50934867  50958555  23689      - HAVANA gene    NA
## ENSG00000000457.14     chr1 169849631 169894267  44637      - HAVANA gene    NA
## ENSG00000000460.17     chr1 169662007 169854080 192074      + HAVANA gene    NA
## ENSG00000000938.13     chr1  27612064  27635185  23122      - HAVANA gene    NA
##                    phase            gene_id      gene_type gene_name level
## ENSG00000000003.15    NA ENSG00000000003.15 protein_coding    TSPAN6     2
## ENSG00000000005.6     NA  ENSG00000000005.6 protein_coding      TNMD     2
## ENSG00000000419.13    NA ENSG00000000419.13 protein_coding      DPM1     2
## ENSG00000000457.14    NA ENSG00000000457.14 protein_coding     SCYL3     2
## ENSG00000000460.17    NA ENSG00000000460.17 protein_coding  C1orf112     2
## ENSG00000000938.13    NA ENSG00000000938.13 protein_coding       FGR     2
##                       hgnc_id          havana_gene
## ENSG00000000003.15 HGNC:11858 OTTHUMG00000022002.2
## ENSG00000000005.6  HGNC:17757 OTTHUMG00000022001.2
## ENSG00000000419.13  HGNC:3005 OTTHUMG00000032742.2
## ENSG00000000457.14 HGNC:19285 OTTHUMG00000035941.6
## ENSG00000000460.17 HGNC:25565 OTTHUMG00000035821.9
## ENSG00000000938.13  HGNC:3697 OTTHUMG00000003516.3
\end{verbatim}

Download normal samples

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# RNA{-}Seq data: solid tissue normal samples}
\NormalTok{rna.query.N }\OtherTok{\textless{}{-}}\NormalTok{ TCGAbiolinks}\SpecialCharTok{::}\FunctionTok{GDCquery}\NormalTok{(}
  \AttributeTok{project =}\NormalTok{ proj,}
  \AttributeTok{data.category =} \StringTok{"Transcriptome Profiling"}\NormalTok{,}
  \AttributeTok{data.type =} \StringTok{"Gene Expression Quantification"}\NormalTok{,}
  \AttributeTok{workflow.type =} \StringTok{"STAR {-} Counts"}\NormalTok{,}
  \AttributeTok{sample.type =} \StringTok{"Solid Tissue Normal"}
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Download normal data

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{GDCdownload}\NormalTok{(}\AttributeTok{query =}\NormalTok{ rna.query.N, }\AttributeTok{directory =} \StringTok{"GDCdata"}\NormalTok{, }\AttributeTok{method =} \StringTok{"api"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Prepare SummarizedExperiment for normal samples

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{rna.data.N }\OtherTok{\textless{}{-}} \FunctionTok{GDCprepare}\NormalTok{(rna.query.N, }\AttributeTok{directory =} \StringTok{"GDCdata"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Extract raw counts for normal samples

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{rna.expr.data.N }\OtherTok{\textless{}{-}} \FunctionTok{assay}\NormalTok{(rna.data.N)}
\end{Highlighting}
\end{Shaded}

Extract gene annotation for normal dataset

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{genes.info2 }\OtherTok{\textless{}{-}}\NormalTok{ BiocGenerics}\SpecialCharTok{::}\FunctionTok{as.data.frame}\NormalTok{(}\FunctionTok{rowRanges}\NormalTok{(rna.data.N))}
\end{Highlighting}
\end{Shaded}

Sanity check: tumor and normal gene annotations should match

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{all}\NormalTok{(}\FunctionTok{na.omit}\NormalTok{(genes.info2) }\SpecialCharTok{==} \FunctionTok{na.omit}\NormalTok{(genes.info))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] TRUE
\end{verbatim}

Checks:

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{dim}\NormalTok{(rna.expr.data.N)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 60660    32
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{rna.data.N}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## class: RangedSummarizedExperiment 
## dim: 60660 32 
## metadata(1): data_release
## assays(6): unstranded stranded_first ... fpkm_unstrand fpkm_uq_unstrand
## rownames(60660): ENSG00000000003.15 ENSG00000000005.6 ...
##   ENSG00000288674.1 ENSG00000288675.1
## rowData names(10): source type ... hgnc_id havana_gene
## colnames(32): TCGA-Y8-A8RY-11A-11R-A37K-07 TCGA-DZ-6133-11A-01R-1965-07
##   ... TCGA-BQ-7045-11A-01R-1965-07 TCGA-BQ-7059-11A-01R-1965-07
## colData names(208): barcode patient ... paper_VHL.copy..log2.normal.
##   paper_NF2.copy..log2.normal.
\end{verbatim}

Results:

\begin{itemize}
\item
  Tumor: 60,660 genes × 290 samples
\item
  Normal: 60,660 genes × 32 samples
\item
  Gene annotation: identical between tumor and normal
\end{itemize}

\subsubsection{Data cleaning}\label{data-cleaning}

\paragraph{Inspect basic structure}\label{inspect-basic-structure}

Dimensions: genes x samples

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{dim}\NormalTok{(rna.expr.data.C)  }\CommentTok{\# tumor}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 60660   290
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{dim}\NormalTok{(rna.expr.data.N)  }\CommentTok{\# normal}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 60660    32
\end{verbatim}

Quick look at column names (TCGA barcodes)

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(rna.expr.data.C))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "TCGA-MH-A855-01A-11R-A355-07" "TCGA-P4-A5E6-01A-11R-A28H-07"
## [3] "TCGA-BQ-5875-01A-11R-1592-07" "TCGA-WN-A9G9-01A-12R-A37K-07"
## [5] "TCGA-HE-A5NH-01A-11R-A26U-07" "TCGA-F9-A8NY-01A-11R-A36F-07"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(rna.expr.data.N))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "TCGA-Y8-A8RY-11A-11R-A37K-07" "TCGA-DZ-6133-11A-01R-1965-07"
## [3] "TCGA-BQ-5891-11A-01R-1592-07" "TCGA-GL-7966-11A-01R-2204-07"
## [5] "TCGA-GL-A9DE-11A-11R-A37K-07" "TCGA-BQ-5882-11A-01R-1592-07"
\end{verbatim}

Extract patient IDs (first 12 characters of barcode)

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{patients.C }\OtherTok{\textless{}{-}} \FunctionTok{substr}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(rna.expr.data.C), }\DecValTok{1}\NormalTok{, }\DecValTok{12}\NormalTok{)}
\NormalTok{patients.N }\OtherTok{\textless{}{-}} \FunctionTok{substr}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(rna.expr.data.N), }\DecValTok{1}\NormalTok{, }\DecValTok{12}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

How many unique patients in each group?

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(patients.C))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 290
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(patients.N))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 32
\end{verbatim}

Distribution of tumor samples per patient (to detect duplicates)

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{sort}\NormalTok{(}\FunctionTok{table}\NormalTok{(patients.C))}
\end{Highlighting}
\end{Shaded}

\paragraph{Keep only patients with exactly ONE tumor
sample}\label{keep-only-patients-with-exactly-one-tumor-sample}

Table: how many tumor samples per patient

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{tab.C }\OtherTok{\textless{}{-}} \FunctionTok{table}\NormalTok{(patients.C)}
\end{Highlighting}
\end{Shaded}

Select patient IDs with exactly 1 primary tumor sample

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{single.C }\OtherTok{\textless{}{-}} \FunctionTok{names}\NormalTok{(tab.C[tab.C }\SpecialCharTok{==} \DecValTok{1}\NormalTok{])}
\end{Highlighting}
\end{Shaded}

Indices of columns in tumor matrix corresponding to single-sample
patients

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{idx.single.C }\OtherTok{\textless{}{-}} \FunctionTok{which}\NormalTok{(patients.C }\SpecialCharTok{\%in\%}\NormalTok{ single.C)}
\end{Highlighting}
\end{Shaded}

Tumor expression matrix: only one tumor sample per patient

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{expr.C }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(rna.expr.data.C[, idx.single.C])}
\end{Highlighting}
\end{Shaded}

Normal expression matrix: start with all normals

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{expr.N }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(rna.expr.data.N)}
\end{Highlighting}
\end{Shaded}

\paragraph{Rename columns to pure patient
IDs}\label{rename-columns-to-pure-patient-ids}

Use only the 12-character patient ID as column names

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{colnames}\NormalTok{(expr.C) }\OtherTok{\textless{}{-}} \FunctionTok{substr}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(expr.C), }\DecValTok{1}\NormalTok{, }\DecValTok{12}\NormalTok{) }
\FunctionTok{colnames}\NormalTok{(expr.N) }\OtherTok{\textless{}{-}} \FunctionTok{substr}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(expr.N), }\DecValTok{1}\NormalTok{, }\DecValTok{12}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

quick check

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(expr.C))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "TCGA-MH-A855" "TCGA-P4-A5E6" "TCGA-BQ-5875" "TCGA-WN-A9G9" "TCGA-HE-A5NH"
## [6] "TCGA-F9-A8NY"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(expr.N))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "TCGA-Y8-A8RY" "TCGA-DZ-6133" "TCGA-BQ-5891" "TCGA-GL-7966" "TCGA-GL-A9DE"
## [6] "TCGA-BQ-5882"
\end{verbatim}

Make sure there are no duplicated column names now

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{sum}\NormalTok{(}\FunctionTok{duplicated}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(expr.C)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{sum}\NormalTok{(}\FunctionTok{duplicated}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(expr.N)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0
\end{verbatim}

\paragraph{Keep only patients with BOTH tumor and normal
samples}\label{keep-only-patients-with-both-tumor-and-normal-samples}

Patient IDs present in BOTH expr.C (tumor) and expr.N (normal)

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{common\_ids }\OtherTok{\textless{}{-}} \FunctionTok{intersect}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(expr.C), }\FunctionTok{colnames}\NormalTok{(expr.N))}
\FunctionTok{length}\NormalTok{(common\_ids) }\CommentTok{\# this is the number of paired patients}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 32
\end{verbatim}

Subset both matrices to the same set of patients

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{expr.C }\OtherTok{\textless{}{-}}\NormalTok{ expr.C[, common\_ids, drop }\OtherTok{=} \ConstantTok{FALSE}\NormalTok{] }
\NormalTok{expr.N }\OtherTok{\textless{}{-}}\NormalTok{ expr.N[, common\_ids, drop }\OtherTok{=} \ConstantTok{FALSE}\NormalTok{]}
\end{Highlighting}
\end{Shaded}

\paragraph{Remove genes with zero counts in ANY
sample}\label{remove-genes-with-zero-counts-in-any-sample}

We remove genes that have zero counts in at least one sample (tumor or
normal) since they do not provide useful information for differential
expression analysis and we aim to avoid mathematical issues during log
fold change calculations.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{nonzero\_genes }\OtherTok{\textless{}{-}} \FunctionTok{rowSums}\NormalTok{(expr.C }\SpecialCharTok{==} \DecValTok{0}\NormalTok{) }\SpecialCharTok{==} \DecValTok{0} \SpecialCharTok{\&} \FunctionTok{rowSums}\NormalTok{(expr.N }\SpecialCharTok{==} \DecValTok{0}\NormalTok{) }\SpecialCharTok{==} \DecValTok{0}

\NormalTok{expr.C }\OtherTok{\textless{}{-}}\NormalTok{ expr.C[nonzero\_genes, ]}
\NormalTok{expr.N }\OtherTok{\textless{}{-}}\NormalTok{ expr.N[nonzero\_genes, ]}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Genes kept:"}\NormalTok{, }\FunctionTok{sum}\NormalTok{(nonzero\_genes), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Genes kept: 17790
\end{verbatim}

Sanity checks

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ncol}\NormalTok{(expr.C) }\CommentTok{\# number of paired tumor samples }
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 32
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ncol}\NormalTok{(expr.N) }\CommentTok{\# number of paired normal samples }
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 32
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{all}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(expr.C) }\SpecialCharTok{==} \FunctionTok{colnames}\NormalTok{(expr.N)) }\CommentTok{\# should be TRUE}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] TRUE
\end{verbatim}

\paragraph{Final technical checks on
counts}\label{final-technical-checks-on-counts}

Check that genes (rows) are aligned between tumor and normal

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{all}\NormalTok{(}\FunctionTok{rownames}\NormalTok{(expr.C) }\SpecialCharTok{==} \FunctionTok{rownames}\NormalTok{(expr.N)) }\CommentTok{\# must be TRUE}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] TRUE
\end{verbatim}

Check data types and missingness

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{typeof}\NormalTok{(expr.C[}\DecValTok{1}\NormalTok{, }\DecValTok{1}\NormalTok{])            }\CommentTok{\# should be "integer" or "double" }
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "integer"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{any}\NormalTok{(}\FunctionTok{is.na}\NormalTok{(expr.C))              }\CommentTok{\# should be FALSE }
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] FALSE
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{any}\NormalTok{(}\FunctionTok{is.nan}\NormalTok{(}\FunctionTok{as.matrix}\NormalTok{(expr.C)))  }\CommentTok{\# should be FALSE}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] FALSE
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{typeof}\NormalTok{(expr.N[}\DecValTok{1}\NormalTok{, }\DecValTok{1}\NormalTok{]) }
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "integer"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{any}\NormalTok{(}\FunctionTok{is.na}\NormalTok{(expr.N)) }
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] FALSE
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{any}\NormalTok{(}\FunctionTok{is.nan}\NormalTok{(}\FunctionTok{as.matrix}\NormalTok{(expr.N)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] FALSE
\end{verbatim}

At this point:

\begin{itemize}
\item
  expr.C = raw tumor counts, genes x N\_paired patients
\item
  expr.N = raw normal counts, genes x N\_paired patients
\item
  Columns in expr.C and expr.N are the SAME patients, SAME order
\end{itemize}

These are the matrices we will feed into DESeq2 for normalization.

\subsubsection{Normalizing data with
DESeq2}\label{normalizing-data-with-deseq2}

Check alignment of tumor and normal matrices

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{all}\NormalTok{(}\FunctionTok{rownames}\NormalTok{(expr.C) }\SpecialCharTok{==} \FunctionTok{rownames}\NormalTok{(expr.N)) }\CommentTok{\# Must be TRUE (same gene order)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] TRUE
\end{verbatim}

Combine into full dataset: normals first, then tumors

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{full.data }\OtherTok{\textless{}{-}} \FunctionTok{cbind}\NormalTok{(expr.N, expr.C) }
\NormalTok{full.data }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(full.data)}

\FunctionTok{dim}\NormalTok{(full.data) }\CommentTok{\# should be 60660 x 64 (32 normal + 32 tumor)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 17790    64
\end{verbatim}

\subsubsection{Build metadata}\label{build-metadata}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{metad }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{( }
    \AttributeTok{condition =} \FunctionTok{factor}\NormalTok{(}\FunctionTok{c}\NormalTok{( }
        \FunctionTok{rep}\NormalTok{(}\StringTok{"normal"}\NormalTok{, }\FunctionTok{ncol}\NormalTok{(expr.N)), }
        \FunctionTok{rep}\NormalTok{(}\StringTok{"cancer"}\NormalTok{, }\FunctionTok{ncol}\NormalTok{(expr.C)) }
\NormalTok{    )) }
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Row names must match full.data column names

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{rownames}\NormalTok{(metad) }\OtherTok{\textless{}{-}} \FunctionTok{colnames}\NormalTok{(full.data)}
\end{Highlighting}
\end{Shaded}

DESeq2 expects gene\_id as a column when tidy=TRUE

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{full.data }\OtherTok{\textless{}{-}} \FunctionTok{cbind}\NormalTok{(}\AttributeTok{gene\_id =} \FunctionTok{rownames}\NormalTok{(full.data), full.data)}
\end{Highlighting}
\end{Shaded}

\subsubsection{Build DESeq2 dataset}\label{build-deseq2-dataset}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{dds }\OtherTok{\textless{}{-}} \FunctionTok{DESeqDataSetFromMatrix}\NormalTok{( }
    \AttributeTok{countData =}\NormalTok{ full.data, }
    \AttributeTok{colData =}\NormalTok{ metad, }
    \AttributeTok{design =} \SpecialCharTok{\textasciitilde{}}\NormalTok{ condition, }
    \AttributeTok{tidy =} \ConstantTok{TRUE} 
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Sanity check

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{dim}\NormalTok{(}\FunctionTok{counts}\NormalTok{(dds)) }\CommentTok{\# should be 60660 x 64}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 17790    64
\end{verbatim}

\subsubsection{Filter low-expression
genes}\label{filter-low-expression-genes}

At least 10 counts in 90\% of \textbf{all} samples (32 normal + 32
cancer = 64 → 90\% = 57)

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{total\_samples }\OtherTok{\textless{}{-}} \FunctionTok{ncol}\NormalTok{(expr.N) }\SpecialCharTok{+} \FunctionTok{ncol}\NormalTok{(expr.C)  }\CommentTok{\# 64}
\NormalTok{threshold }\OtherTok{\textless{}{-}} \FunctionTok{floor}\NormalTok{(}\FloatTok{0.9} \SpecialCharTok{*}\NormalTok{ total\_samples)       }\CommentTok{\# 57}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Total samples:"}\NormalTok{, total\_samples, }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Total samples: 64
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"Filtering threshold (90\%):"}\NormalTok{, threshold, }\StringTok{"samples}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Filtering threshold (90%): 57 samples
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{keep }\OtherTok{\textless{}{-}} \FunctionTok{rowSums}\NormalTok{(}\FunctionTok{counts}\NormalTok{(dds) }\SpecialCharTok{\textgreater{}=} \DecValTok{10}\NormalTok{) }\SpecialCharTok{\textgreater{}=}\NormalTok{ threshold}
\NormalTok{dds }\OtherTok{\textless{}{-}}\NormalTok{ dds[keep, ]}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Genes before filtering:"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(full.data), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Genes before filtering: 17790
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"Genes after filtering:"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(dds), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Genes after filtering: 15886
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"Genes removed:"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(full.data) }\SpecialCharTok{{-}} \FunctionTok{sum}\NormalTok{(keep), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Genes removed: 1904
\end{verbatim}

\subsubsection{Normalize using DESeq2 size
factors}\label{normalize-using-deseq2-size-factors}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{dds }\OtherTok{\textless{}{-}} \FunctionTok{estimateSizeFactors}\NormalTok{(dds)}

\NormalTok{normalized\_counts }\OtherTok{\textless{}{-}} \FunctionTok{counts}\NormalTok{(dds, }\AttributeTok{normalized =} \ConstantTok{TRUE}\NormalTok{)}

\CommentTok{\# No gene should be zero in all samples}
\FunctionTok{sum}\NormalTok{(}\FunctionTok{rowSums}\NormalTok{(normalized\_counts }\SpecialCharTok{==} \DecValTok{0}\NormalTok{) }\SpecialCharTok{==} \FunctionTok{ncol}\NormalTok{(normalized\_counts))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 0
\end{verbatim}

\subsubsection{Split normalized matrix back into
normal/tumor}\label{split-normalized-matrix-back-into-normaltumor}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{filtr.expr.n }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{( }
\NormalTok{    normalized\_counts[, }\DecValTok{1}\SpecialCharTok{:}\FunctionTok{ncol}\NormalTok{(expr.N)] }
\NormalTok{)}

\NormalTok{filtr.expr.c }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{( }
\NormalTok{    normalized\_counts[, (}\FunctionTok{ncol}\NormalTok{(expr.N) }\SpecialCharTok{+} \DecValTok{1}\NormalTok{)}\SpecialCharTok{:}\FunctionTok{ncol}\NormalTok{(normalized\_counts)] }
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Keep original patient IDs (DESeq2 sometimes adds .1, .2)

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{colnames}\NormalTok{(filtr.expr.n) }\OtherTok{\textless{}{-}} \FunctionTok{colnames}\NormalTok{(expr.N) }
\FunctionTok{colnames}\NormalTok{(filtr.expr.c) }\OtherTok{\textless{}{-}} \FunctionTok{colnames}\NormalTok{(expr.C)}
\end{Highlighting}
\end{Shaded}

\subsubsection{Final checks}\label{final-checks}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Same genes}
\FunctionTok{all}\NormalTok{(}\FunctionTok{rownames}\NormalTok{(filtr.expr.c) }\SpecialCharTok{==} \FunctionTok{rownames}\NormalTok{(filtr.expr.n))}

\CommentTok{\# Same patients}
\FunctionTok{all}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(filtr.expr.c) }\SpecialCharTok{==} \FunctionTok{colnames}\NormalTok{(filtr.expr.n))}

\FunctionTok{nrow}\NormalTok{(filtr.expr.c)   }\CommentTok{\# number of filtered/normalized genes}
\FunctionTok{ncol}\NormalTok{(filtr.expr.c)   }\CommentTok{\# should be 32}

\FunctionTok{sizeFactors}\NormalTok{(dds)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{summary}\NormalTok{(}\FunctionTok{sizeFactors}\NormalTok{(dds))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.3088  0.8530  1.0833  1.0569  1.2672  1.8590
\end{verbatim}

\textbf{Output:}

\begin{itemize}
\item
  \textbf{filtr.expr.n = normalized expression for 32 normal samples}
\item
  \textbf{filtr.expr.c = normalized expression for 32 tumor samples}
\end{itemize}

These matrices are what we will use for:

\begin{itemize}
\item
  DEGs
\item
  Volcano Plot
\item
  Co-expression networks
\item
  Differential networks
\item
  PSN and SNF
\end{itemize}

\subsection{2. Differential Expression Analysis
(DEGs)}\label{differential-expression-analysis-degs}

\subsubsection{Compute log2 Fold Change}\label{compute-log2-fold-change}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{log2fc }\OtherTok{\textless{}{-}} \FunctionTok{log2}\NormalTok{(}\FunctionTok{rowMeans}\NormalTok{(filtr.expr.c) }\SpecialCharTok{/} \FunctionTok{rowMeans}\NormalTok{(filtr.expr.n))}
\FunctionTok{names}\NormalTok{(log2fc) }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(filtr.expr.c)}
\end{Highlighting}
\end{Shaded}

\subsubsection{Compute paired t-test
p-values}\label{compute-paired-t-test-p-values}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pval }\OtherTok{\textless{}{-}} \FunctionTok{sapply}\NormalTok{(}
  \DecValTok{1}\SpecialCharTok{:}\FunctionTok{nrow}\NormalTok{(filtr.expr.c),}
  \ControlFlowTok{function}\NormalTok{(i) }\FunctionTok{t.test}\NormalTok{(}
    \FunctionTok{as.numeric}\NormalTok{(filtr.expr.c[i, ]),}
    \FunctionTok{as.numeric}\NormalTok{(filtr.expr.n[i, ]),}
    \AttributeTok{paired =} \ConstantTok{TRUE}
\NormalTok{  )}\SpecialCharTok{$}\NormalTok{p.value}
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsubsection{FDR correction}\label{fdr-correction}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pval\_fdr }\OtherTok{\textless{}{-}} \FunctionTok{p.adjust}\NormalTok{(pval, }\AttributeTok{method =} \StringTok{"fdr"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsubsection{Build DEG table}\label{build-deg-table}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{deg.table }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{gene        =} \FunctionTok{rownames}\NormalTok{(filtr.expr.c),}
  \AttributeTok{log2FC      =}\NormalTok{ log2fc,}
  \AttributeTok{pvalue      =}\NormalTok{ pval,}
  \AttributeTok{pvalue\_fdr  =}\NormalTok{ pval\_fdr,}
  \AttributeTok{row.names   =} \FunctionTok{rownames}\NormalTok{(filtr.expr.c)}
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\subsubsection{Apply thresholds}\label{apply-thresholds}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{FDR\_threshold }\OtherTok{\textless{}{-}} \FloatTok{0.05}
\NormalTok{FC\_threshold  }\OtherTok{\textless{}{-}} \DecValTok{2}

\NormalTok{deg.genes }\OtherTok{\textless{}{-}} \FunctionTok{subset}\NormalTok{(}
\NormalTok{  deg.table,}
  \FunctionTok{abs}\NormalTok{(log2FC) }\SpecialCharTok{\textgreater{}=}\NormalTok{ FC\_threshold }\SpecialCharTok{\&}\NormalTok{ pvalue\_fdr }\SpecialCharTok{\textless{}=}\NormalTok{ FDR\_threshold}
\NormalTok{)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Number of DEGs:"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(deg.genes), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Number of DEGs: 724
\end{verbatim}

\subsubsection{Volcano plot}\label{volcano-plot}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{deg.table}\SpecialCharTok{$}\NormalTok{color }\OtherTok{\textless{}{-}} \StringTok{"NO"}
\NormalTok{deg.table}\SpecialCharTok{$}\NormalTok{color[deg.table}\SpecialCharTok{$}\NormalTok{log2FC }\SpecialCharTok{\textgreater{}=}\NormalTok{ FC\_threshold }\SpecialCharTok{\&}\NormalTok{ deg.table}\SpecialCharTok{$}\NormalTok{pvalue\_fdr }\SpecialCharTok{\textless{}=}\NormalTok{ FDR\_threshold] }\OtherTok{\textless{}{-}} \StringTok{"UP"}
\NormalTok{deg.table}\SpecialCharTok{$}\NormalTok{color[deg.table}\SpecialCharTok{$}\NormalTok{log2FC }\SpecialCharTok{\textless{}=} \SpecialCharTok{{-}}\NormalTok{FC\_threshold }\SpecialCharTok{\&}\NormalTok{ deg.table}\SpecialCharTok{$}\NormalTok{pvalue\_fdr }\SpecialCharTok{\textless{}=}\NormalTok{ FDR\_threshold] }\OtherTok{\textless{}{-}} \StringTok{"DOWN"}
\NormalTok{deg.table}\SpecialCharTok{$}\NormalTok{color }\OtherTok{\textless{}{-}} \FunctionTok{factor}\NormalTok{(deg.table}\SpecialCharTok{$}\NormalTok{color, }\AttributeTok{levels =} \FunctionTok{c}\NormalTok{(}\StringTok{"DOWN"}\NormalTok{, }\StringTok{"NO"}\NormalTok{, }\StringTok{"UP"}\NormalTok{))}


\NormalTok{base\_theme }\OtherTok{\textless{}{-}} \FunctionTok{theme\_minimal}\NormalTok{(}\AttributeTok{base\_size =} \DecValTok{20}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}
    \AttributeTok{plot.title =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{face =} \StringTok{"bold"}\NormalTok{, }\AttributeTok{hjust =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{size =} \DecValTok{40}\NormalTok{),}
    \AttributeTok{axis.title =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{30}\NormalTok{),}
    \AttributeTok{panel.grid.major =} \FunctionTok{element\_blank}\NormalTok{(),}
    \AttributeTok{axis.text.x =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{25}\NormalTok{),   }
    \AttributeTok{axis.text.y =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{25}\NormalTok{)}
\NormalTok{  )}

\FunctionTok{ggplot}\NormalTok{(deg.table, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ log2FC, }\AttributeTok{y =} \SpecialCharTok{{-}}\FunctionTok{log10}\NormalTok{(pvalue\_fdr), }\AttributeTok{color =}\NormalTok{ color)) }\SpecialCharTok{+}
  \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{alpha =} \FloatTok{0.6}\NormalTok{, }\AttributeTok{size =} \DecValTok{5}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{title =} \StringTok{"Volcano Plot of Differential Expression"}\NormalTok{,}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{xlab}\NormalTok{(}\StringTok{"log2 Fold Change"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{ylab}\NormalTok{(}\StringTok{"{-}log10 FDR"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_vline}\NormalTok{(}\AttributeTok{xintercept =} \FunctionTok{c}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{FC\_threshold, FC\_threshold), }\AttributeTok{col =} \StringTok{"red"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_hline}\NormalTok{(}\AttributeTok{yintercept =} \SpecialCharTok{{-}}\FunctionTok{log10}\NormalTok{(FDR\_threshold), }\AttributeTok{col =} \StringTok{"red"}\NormalTok{) }\SpecialCharTok{+}
\NormalTok{  base\_theme }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"bottom"}\NormalTok{, }\AttributeTok{legend.title =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{0}\NormalTok{, }\AttributeTok{face =} \StringTok{"bold"}\NormalTok{), }\AttributeTok{legend.text =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{30}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-54-1.pdf}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{DEG\_list }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(deg.genes)}
\FunctionTok{length}\NormalTok{(DEG\_list)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 724
\end{verbatim}

\subsection{3. Co-Expression Networks}\label{co-expression-networks}

\subsubsection{3.1 Data Preparation}\label{data-preparation}

Extract DEG expression matrices and apply log transformation for
correlation analysis.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Keep only DEGs in the normalized matrices}
\NormalTok{expr\_deg\_cancer }\OtherTok{\textless{}{-}}\NormalTok{ filtr.expr.c[DEG\_list, ]}
\NormalTok{expr\_deg\_normal }\OtherTok{\textless{}{-}}\NormalTok{ filtr.expr.n[DEG\_list, ]}

\CommentTok{\# Log transform for correlation analysis}
\NormalTok{expr\_deg\_cancer }\OtherTok{\textless{}{-}} \FunctionTok{log2}\NormalTok{(expr\_deg\_cancer }\SpecialCharTok{+} \DecValTok{1}\NormalTok{)}
\NormalTok{expr\_deg\_normal }\OtherTok{\textless{}{-}} \FunctionTok{log2}\NormalTok{(expr\_deg\_normal }\SpecialCharTok{+} \DecValTok{1}\NormalTok{)}

\CommentTok{\# Verify dimensions}
\FunctionTok{cat}\NormalTok{(}\StringTok{"Expression matrix dimensions:}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Expression matrix dimensions:
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Cancer: \%d genes × \%d samples}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(expr\_deg\_cancer), }\FunctionTok{ncol}\NormalTok{(expr\_deg\_cancer)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Cancer: 724 genes × 32 samples
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Normal: \%d genes × \%d samples}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(expr\_deg\_normal), }\FunctionTok{ncol}\NormalTok{(expr\_deg\_normal)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Normal: 724 genes × 32 samples
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Genes aligned: \%s}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{all}\NormalTok{(}\FunctionTok{rownames}\NormalTok{(expr\_deg\_cancer) }\SpecialCharTok{==} \FunctionTok{rownames}\NormalTok{(expr\_deg\_normal))))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Genes aligned: TRUE
\end{verbatim}

\subsubsection{3.2 Network Computation}\label{network-computation}

Compute Pearson correlation matrices with FDR-adjusted p-values for both
conditions.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Cancer correlation with FDR{-}adjusted p{-}values}
\NormalTok{cor.mat.c }\OtherTok{\textless{}{-}} \FunctionTok{corr.test}\NormalTok{(}
  \FunctionTok{t}\NormalTok{(expr\_deg\_cancer), }
  \AttributeTok{use =} \StringTok{"pairwise"}\NormalTok{,}
  \AttributeTok{method =} \StringTok{"pearson"}\NormalTok{,}
  \AttributeTok{adjust =} \StringTok{"fdr"}\NormalTok{,}
  \AttributeTok{ci =} \ConstantTok{FALSE}
\NormalTok{)}

\NormalTok{rho.c }\OtherTok{\textless{}{-}}\NormalTok{ cor.mat.c}\SpecialCharTok{$}\NormalTok{r}
\FunctionTok{diag}\NormalTok{(rho.c) }\OtherTok{\textless{}{-}} \DecValTok{0}
\NormalTok{qval.c }\OtherTok{\textless{}{-}}\NormalTok{ cor.mat.c}\SpecialCharTok{$}\NormalTok{p}
\NormalTok{qval.c[}\FunctionTok{lower.tri}\NormalTok{(qval.c)] }\OtherTok{\textless{}{-}} \FunctionTok{t}\NormalTok{(qval.c)[}\FunctionTok{lower.tri}\NormalTok{(qval.c)]}

\CommentTok{\# Normal correlation with FDR{-}adjusted p{-}values}
\NormalTok{cor.mat.n }\OtherTok{\textless{}{-}} \FunctionTok{corr.test}\NormalTok{(}
  \FunctionTok{t}\NormalTok{(expr\_deg\_normal), }
  \AttributeTok{use =} \StringTok{"pairwise"}\NormalTok{,}
  \AttributeTok{method =} \StringTok{"pearson"}\NormalTok{,}
  \AttributeTok{adjust =} \StringTok{"fdr"}\NormalTok{,}
  \AttributeTok{ci =} \ConstantTok{FALSE}
\NormalTok{)}

\NormalTok{rho.n }\OtherTok{\textless{}{-}}\NormalTok{ cor.mat.n}\SpecialCharTok{$}\NormalTok{r}
\FunctionTok{diag}\NormalTok{(rho.n) }\OtherTok{\textless{}{-}} \DecValTok{0}
\NormalTok{qval.n }\OtherTok{\textless{}{-}}\NormalTok{ cor.mat.n}\SpecialCharTok{$}\NormalTok{p}
\NormalTok{qval.n[}\FunctionTok{lower.tri}\NormalTok{(qval.n)] }\OtherTok{\textless{}{-}} \FunctionTok{t}\NormalTok{(qval.n)[}\FunctionTok{lower.tri}\NormalTok{(qval.n)]}
\end{Highlighting}
\end{Shaded}

Build binary adjacency matrices using correlation and significance
thresholds.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Define thresholds}
\NormalTok{corr\_threshold\_c }\OtherTok{\textless{}{-}} \FloatTok{0.7}   \CommentTok{\# Cancer correlation threshold}
\NormalTok{corr\_threshold\_n }\OtherTok{\textless{}{-}} \FloatTok{0.8}   \CommentTok{\# Normal correlation threshold (more stringent)}
\NormalTok{pval\_threshold  }\OtherTok{\textless{}{-}} \FloatTok{0.05}  \CommentTok{\# FDR{-}adjusted p{-}value threshold}

\CommentTok{\# Build binary adjacency matrices}
\NormalTok{adj\_mat\_c }\OtherTok{\textless{}{-}}\NormalTok{ ((}\FunctionTok{abs}\NormalTok{(rho.c) }\SpecialCharTok{\textgreater{}=}\NormalTok{ corr\_threshold\_c) }\SpecialCharTok{\&}\NormalTok{ (qval.c }\SpecialCharTok{\textless{}=}\NormalTok{ pval\_threshold)) }\SpecialCharTok{*} \DecValTok{1}
\NormalTok{adj\_mat\_n }\OtherTok{\textless{}{-}}\NormalTok{ ((}\FunctionTok{abs}\NormalTok{(rho.n) }\SpecialCharTok{\textgreater{}=}\NormalTok{ corr\_threshold\_n) }\SpecialCharTok{\&}\NormalTok{ (qval.n }\SpecialCharTok{\textless{}=}\NormalTok{ pval\_threshold)) }\SpecialCharTok{*} \DecValTok{1}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Detailed network statistics}
\FunctionTok{cat}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{=== CANCER NETWORK STATISTICS ===}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## === CANCER NETWORK STATISTICS ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Total possible edges: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(adj\_mat\_c) }\SpecialCharTok{*}\NormalTok{ (}\FunctionTok{nrow}\NormalTok{(adj\_mat\_c) }\SpecialCharTok{{-}} \DecValTok{1}\NormalTok{) }\SpecialCharTok{/} \DecValTok{2}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Total possible edges: 261726
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Significant edges: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{sum}\NormalTok{(adj\_mat\_c }\SpecialCharTok{!=} \DecValTok{0}\NormalTok{) }\SpecialCharTok{/} \DecValTok{2}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Significant edges: 4724
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Network density: \%.4f}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{sum}\NormalTok{(adj\_mat\_c }\SpecialCharTok{!=} \DecValTok{0}\NormalTok{) }\SpecialCharTok{/}\NormalTok{ (}\FunctionTok{nrow}\NormalTok{(adj\_mat\_c) }\SpecialCharTok{*}\NormalTok{ (}\FunctionTok{nrow}\NormalTok{(adj\_mat\_c) }\SpecialCharTok{{-}} \DecValTok{1}\NormalTok{))))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Network density: 0.0180
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  {-} Positive correlations: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{sum}\NormalTok{((adj\_mat\_c }\SpecialCharTok{==} \DecValTok{1}\NormalTok{) }\SpecialCharTok{\&}\NormalTok{ (rho.c }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{)) }\SpecialCharTok{/} \DecValTok{2}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   - Positive correlations: 4620
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  {-} Negative correlations: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{sum}\NormalTok{((adj\_mat\_c }\SpecialCharTok{==} \DecValTok{1}\NormalTok{) }\SpecialCharTok{\&}\NormalTok{ (rho.c }\SpecialCharTok{\textless{}} \DecValTok{0}\NormalTok{)) }\SpecialCharTok{/} \DecValTok{2}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   - Negative correlations: 104
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{=== NORMAL NETWORK STATISTICS ===}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## === NORMAL NETWORK STATISTICS ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Total possible edges: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(adj\_mat\_n) }\SpecialCharTok{*}\NormalTok{ (}\FunctionTok{nrow}\NormalTok{(adj\_mat\_n) }\SpecialCharTok{{-}} \DecValTok{1}\NormalTok{) }\SpecialCharTok{/} \DecValTok{2}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Total possible edges: 261726
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Significant edges: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{sum}\NormalTok{(adj\_mat\_n }\SpecialCharTok{!=} \DecValTok{0}\NormalTok{) }\SpecialCharTok{/} \DecValTok{2}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Significant edges: 8243
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Network density: \%.4f}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{sum}\NormalTok{(adj\_mat\_n }\SpecialCharTok{!=} \DecValTok{0}\NormalTok{) }\SpecialCharTok{/}\NormalTok{ (}\FunctionTok{nrow}\NormalTok{(adj\_mat\_n) }\SpecialCharTok{*}\NormalTok{ (}\FunctionTok{nrow}\NormalTok{(adj\_mat\_n) }\SpecialCharTok{{-}} \DecValTok{1}\NormalTok{))))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Network density: 0.0315
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  {-} Positive correlations: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{sum}\NormalTok{((adj\_mat\_n }\SpecialCharTok{==} \DecValTok{1}\NormalTok{) }\SpecialCharTok{\&}\NormalTok{ (rho.n }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{)) }\SpecialCharTok{/} \DecValTok{2}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   - Positive correlations: 7061
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  {-} Negative correlations: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{sum}\NormalTok{((adj\_mat\_n }\SpecialCharTok{==} \DecValTok{1}\NormalTok{) }\SpecialCharTok{\&}\NormalTok{ (rho.n }\SpecialCharTok{\textless{}} \DecValTok{0}\NormalTok{)) }\SpecialCharTok{/} \DecValTok{2}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   - Negative correlations: 1182
\end{verbatim}

\subsubsection{3.3 Degree Analysis and Scale-Free
Properties}\label{degree-analysis-and-scale-free-properties}

Compute degree centrality for each network.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Compute degree for each network}
\NormalTok{degree\_c }\OtherTok{\textless{}{-}} \FunctionTok{colSums}\NormalTok{(adj\_mat\_c }\SpecialCharTok{!=} \DecValTok{0}\NormalTok{)}
\FunctionTok{names}\NormalTok{(degree\_c) }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(adj\_mat\_c)}
\NormalTok{degree\_c }\OtherTok{\textless{}{-}} \FunctionTok{sort}\NormalTok{(degree\_c, }\AttributeTok{decreasing =} \ConstantTok{TRUE}\NormalTok{)}

\NormalTok{degree\_n }\OtherTok{\textless{}{-}} \FunctionTok{colSums}\NormalTok{(adj\_mat\_n }\SpecialCharTok{!=} \DecValTok{0}\NormalTok{)}
\FunctionTok{names}\NormalTok{(degree\_n) }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(adj\_mat\_n)}
\NormalTok{degree\_n }\OtherTok{\textless{}{-}} \FunctionTok{sort}\NormalTok{(degree\_n, }\AttributeTok{decreasing =} \ConstantTok{TRUE}\NormalTok{)}

\CommentTok{\# Summary statistics}
\FunctionTok{cat}\NormalTok{(}\StringTok{"=== DEGREE DISTRIBUTION SUMMARY ===}\SpecialCharTok{\textbackslash{}n\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## === DEGREE DISTRIBUTION SUMMARY ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"Cancer Network:}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Cancer Network:
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Max degree: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{max}\NormalTok{(degree\_c)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Max degree: 69
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Mean degree: \%. 2f}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{mean}\NormalTok{(degree\_c)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Mean degree: %.0 2f
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Median degree: \%.1f}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{median}\NormalTok{(degree\_c)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Median degree: 1.0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Nodes with degree \textgreater{} 0: \%d (\%.1f\%\%)}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }
            \FunctionTok{sum}\NormalTok{(degree\_c }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{), }\DecValTok{100} \SpecialCharTok{*} \FunctionTok{sum}\NormalTok{(degree\_c }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{) }\SpecialCharTok{/} \FunctionTok{length}\NormalTok{(degree\_c)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Nodes with degree > 0: 439 (60.6%)
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{Normal Network:}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Normal Network:
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Max degree: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{max}\NormalTok{(degree\_n)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Max degree: 124
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Mean degree: \%.2f}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{mean}\NormalTok{(degree\_n)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Mean degree: 22.77
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Median degree: \%.1f}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{median}\NormalTok{(degree\_n)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Median degree: 4.0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Nodes with degree \textgreater{} 0: \%d (\%. 1f\%\%)}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }
            \FunctionTok{sum}\NormalTok{(degree\_n }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{), }\DecValTok{100} \SpecialCharTok{*} \FunctionTok{sum}\NormalTok{(degree\_n }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{) }\SpecialCharTok{/} \FunctionTok{length}\NormalTok{(degree\_n)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Nodes with degree > 0: 488 (%.0 1f%)
\end{verbatim}

\paragraph{Scale-Free Network
Analysis}\label{scale-free-network-analysis}

A scale-free network follows a power-law degree distribution. We check
this by plotting log-log degree distributions.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{par}\NormalTok{(}\AttributeTok{mfrow =} \FunctionTok{c}\NormalTok{(}\DecValTok{1}\NormalTok{, }\DecValTok{2}\NormalTok{))}

\CommentTok{\# Cancer Network}
\FunctionTok{hist}\NormalTok{(degree\_c, }\AttributeTok{breaks =} \DecValTok{50}\NormalTok{, }\AttributeTok{main =} \StringTok{"Cancer Network Degree Distribution"}\NormalTok{,}
     \AttributeTok{xlab =} \StringTok{"Degree"}\NormalTok{, }\AttributeTok{col =} \StringTok{"lightcoral"}\NormalTok{, }\AttributeTok{freq =} \ConstantTok{FALSE}\NormalTok{)}

\CommentTok{\# Normal Network}
\FunctionTok{hist}\NormalTok{(degree\_n, }\AttributeTok{breaks =} \DecValTok{50}\NormalTok{, }\AttributeTok{main =} \StringTok{"Normal Network Degree Distribution"}\NormalTok{,}
     \AttributeTok{xlab =} \StringTok{"Degree"}\NormalTok{, }\AttributeTok{col =} \StringTok{"lightgreen"}\NormalTok{, }\AttributeTok{freq =} \ConstantTok{FALSE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-60-1.pdf}}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{par}\NormalTok{(}\AttributeTok{mfrow =} \FunctionTok{c}\NormalTok{(}\DecValTok{1}\NormalTok{, }\DecValTok{1}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Function to estimate gamma using zeta free log{-}likelihood}
\NormalTok{estimate\_gamma\_zeta }\OtherTok{\textless{}{-}} \ControlFlowTok{function}\NormalTok{(deg\_freq, }\AttributeTok{kmax =} \DecValTok{100000}\NormalTok{) \{}
  \CommentTok{\# Build vector of observations from frequency table}
\NormalTok{  k\_values }\OtherTok{\textless{}{-}} \FunctionTok{as.numeric}\NormalTok{(}\FunctionTok{names}\NormalTok{(deg\_freq))}
\NormalTok{  k\_counts }\OtherTok{\textless{}{-}} \FunctionTok{as.numeric}\NormalTok{(deg\_freq)}
  
  \CommentTok{\# Expand into full sample vector}
\NormalTok{  k }\OtherTok{\textless{}{-}} \FunctionTok{rep}\NormalTok{(k\_values, k\_counts)}
  
  \CommentTok{\# Minimum degree}
\NormalTok{  k\_min }\OtherTok{\textless{}{-}} \FunctionTok{min}\NormalTok{(k)}
  
  \CommentTok{\# Free zeta log{-}likelihood function}
\NormalTok{  loglik\_zeta\_free }\OtherTok{\textless{}{-}} \ControlFlowTok{function}\NormalTok{(gamma, k, k\_min, kmax) \{}
    \ControlFlowTok{if}\NormalTok{ (gamma }\SpecialCharTok{\textless{}=} \DecValTok{1}\NormalTok{) }\FunctionTok{return}\NormalTok{(}\SpecialCharTok{{-}}\ConstantTok{Inf}\NormalTok{)}
    
\NormalTok{    N }\OtherTok{\textless{}{-}} \FunctionTok{length}\NormalTok{(k)}
\NormalTok{    M\_prime }\OtherTok{\textless{}{-}} \FunctionTok{sum}\NormalTok{(}\FunctionTok{log}\NormalTok{(k))}
    
    \CommentTok{\# Normalization constant (truncated zeta)}
\NormalTok{    Z }\OtherTok{\textless{}{-}} \FunctionTok{sum}\NormalTok{(}\DecValTok{1} \SpecialCharTok{/}\NormalTok{ (k\_min}\SpecialCharTok{:}\NormalTok{kmax)}\SpecialCharTok{\^{}}\NormalTok{gamma)}
    
    \ControlFlowTok{if}\NormalTok{ (}\SpecialCharTok{!}\FunctionTok{is.finite}\NormalTok{(Z) }\SpecialCharTok{||}\NormalTok{ Z }\SpecialCharTok{\textless{}=} \DecValTok{0}\NormalTok{) }\FunctionTok{return}\NormalTok{(}\SpecialCharTok{{-}}\ConstantTok{Inf}\NormalTok{)}
    
    \CommentTok{\# Log{-}likelihood}
    \SpecialCharTok{{-}}\NormalTok{gamma }\SpecialCharTok{*}\NormalTok{ M\_prime }\SpecialCharTok{{-}}\NormalTok{ N }\SpecialCharTok{*} \FunctionTok{log}\NormalTok{(Z)}
\NormalTok{  \}}
  
  \CommentTok{\# MLE fit}
\NormalTok{  result }\OtherTok{\textless{}{-}} \FunctionTok{tryCatch}\NormalTok{(\{}
\NormalTok{    fit }\OtherTok{\textless{}{-}} \FunctionTok{optim}\NormalTok{(}
      \AttributeTok{par =} \FloatTok{2.0}\NormalTok{,  }\CommentTok{\# Initial gamma}
      \AttributeTok{fn =} \ControlFlowTok{function}\NormalTok{(gamma) }\SpecialCharTok{{-}}\FunctionTok{loglik\_zeta\_free}\NormalTok{(gamma, k, k\_min, kmax),}
      \AttributeTok{method =} \StringTok{"Brent"}\NormalTok{,}
      \AttributeTok{lower =} \FloatTok{1.01}\NormalTok{,}
      \AttributeTok{upper =} \DecValTok{10}
\NormalTok{    )}
    
\NormalTok{    gamma\_hat }\OtherTok{\textless{}{-}}\NormalTok{ fit}\SpecialCharTok{$}\NormalTok{par}
\NormalTok{    ll }\OtherTok{\textless{}{-}} \SpecialCharTok{{-}}\NormalTok{fit}\SpecialCharTok{$}\NormalTok{value}
    
    \FunctionTok{list}\NormalTok{(}\AttributeTok{gamma =}\NormalTok{ gamma\_hat, }\AttributeTok{loglik =}\NormalTok{ ll, }\AttributeTok{success =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{  \}, }\AttributeTok{error =} \ControlFlowTok{function}\NormalTok{(e) \{}
    \FunctionTok{list}\NormalTok{(}\AttributeTok{gamma =} \ConstantTok{NA}\NormalTok{, }\AttributeTok{loglik =} \ConstantTok{NA}\NormalTok{, }\AttributeTok{success =} \ConstantTok{FALSE}\NormalTok{)}
\NormalTok{  \})}
  
  \FunctionTok{return}\NormalTok{(result)}
\NormalTok{\}}

\FunctionTok{par}\NormalTok{(}\AttributeTok{mfrow =} \FunctionTok{c}\NormalTok{(}\DecValTok{1}\NormalTok{, }\DecValTok{2}\NormalTok{))}


\CommentTok{\# Define consistent font sizes}
\NormalTok{main\_cex  }\OtherTok{\textless{}{-}} \FloatTok{1.2}   \CommentTok{\# title size}
\NormalTok{lab\_cex   }\OtherTok{\textless{}{-}} \FloatTok{1.2}   \CommentTok{\# axis label size}
\NormalTok{axis\_cex  }\OtherTok{\textless{}{-}} \FloatTok{1.0}   \CommentTok{\# tick label size}

\CommentTok{\# Cancer: Log{-}Log plot}
\NormalTok{deg\_freq\_c }\OtherTok{\textless{}{-}} \FunctionTok{table}\NormalTok{(degree\_c[degree\_c }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{])}
\ControlFlowTok{if}\NormalTok{ (}\FunctionTok{length}\NormalTok{(deg\_freq\_c) }\SpecialCharTok{\textgreater{}} \DecValTok{1}\NormalTok{) \{}
  \CommentTok{\# Estimate gamma using zeta MLE}
\NormalTok{  zeta\_result\_c }\OtherTok{\textless{}{-}} \FunctionTok{estimate\_gamma\_zeta}\NormalTok{(deg\_freq\_c)}
\NormalTok{  gamma\_c }\OtherTok{\textless{}{-}}\NormalTok{ zeta\_result\_c}\SpecialCharTok{$}\NormalTok{gamma}
  
  \CommentTok{\# Plot}
  \FunctionTok{plot}\NormalTok{(}\FunctionTok{log10}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(}\FunctionTok{names}\NormalTok{(deg\_freq\_c))), }\FunctionTok{log10}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(deg\_freq\_c)), }
       \AttributeTok{main =} \StringTok{"Cancer: Log{-}Log Degree Distribution"}\NormalTok{, }
       \AttributeTok{xlab =} \StringTok{"Degree"}\NormalTok{, }\AttributeTok{ylab =} \StringTok{"Frequency"}\NormalTok{, }
       \AttributeTok{pch =} \DecValTok{16}\NormalTok{, }\AttributeTok{col =} \StringTok{"lightcoral"}\NormalTok{, }\AttributeTok{cex =} \FloatTok{1.2}\NormalTok{,}
       \AttributeTok{cex.main =}\NormalTok{ main\_cex, }\AttributeTok{cex.lab =}\NormalTok{ lab\_cex, }\AttributeTok{cex.axis =}\NormalTok{ axis\_cex)}
  
  \CommentTok{\# Fit line for visualization (using estimated gamma)}
\NormalTok{  x\_vals }\OtherTok{\textless{}{-}} \FunctionTok{log10}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(}\FunctionTok{names}\NormalTok{(deg\_freq\_c)))}
\NormalTok{  y\_vals }\OtherTok{\textless{}{-}} \FunctionTok{log10}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(deg\_freq\_c))}

\NormalTok{  fit\_c }\OtherTok{\textless{}{-}} \FunctionTok{lm}\NormalTok{(y\_vals }\SpecialCharTok{\textasciitilde{}}\NormalTok{ x\_vals)}
  
  \FunctionTok{abline}\NormalTok{(fit\_c, }\AttributeTok{col =} \StringTok{"darkred"}\NormalTok{, }\AttributeTok{lwd =} \DecValTok{2}\NormalTok{, }\AttributeTok{lty =} \DecValTok{2}\NormalTok{)}
  
  \FunctionTok{legend}\NormalTok{(}\StringTok{"topright"}\NormalTok{, }
         \AttributeTok{legend =} \FunctionTok{paste}\NormalTok{(}\StringTok{"γ (MLE) ="}\NormalTok{, }\FunctionTok{round}\NormalTok{(gamma\_c, }\DecValTok{2}\NormalTok{)),}
         \AttributeTok{bty =} \StringTok{"n"}\NormalTok{, }\AttributeTok{cex =} \FloatTok{0.9}\NormalTok{)}
\NormalTok{\}}

\CommentTok{\# Normal: Log{-}Log plot}
\NormalTok{deg\_freq\_n }\OtherTok{\textless{}{-}} \FunctionTok{table}\NormalTok{(degree\_n[degree\_n }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{])}
\ControlFlowTok{if}\NormalTok{ (}\FunctionTok{length}\NormalTok{(deg\_freq\_n) }\SpecialCharTok{\textgreater{}} \DecValTok{1}\NormalTok{) \{}
  \CommentTok{\# Estimate gamma using zeta MLE}
\NormalTok{  zeta\_result\_n }\OtherTok{\textless{}{-}} \FunctionTok{estimate\_gamma\_zeta}\NormalTok{(deg\_freq\_n)}
\NormalTok{  gamma\_n }\OtherTok{\textless{}{-}}\NormalTok{ zeta\_result\_n}\SpecialCharTok{$}\NormalTok{gamma}
\NormalTok{  ll\_n }\OtherTok{\textless{}{-}}\NormalTok{ zeta\_result\_n}\SpecialCharTok{$}\NormalTok{loglik}
  
  \CommentTok{\# Plot}
  \FunctionTok{plot}\NormalTok{(}\FunctionTok{log10}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(}\FunctionTok{names}\NormalTok{(deg\_freq\_n))), }\FunctionTok{log10}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(deg\_freq\_n)), }
       \AttributeTok{main =} \StringTok{"Normal: Log{-}Log Degree Distribution"}\NormalTok{, }
       \AttributeTok{xlab =} \StringTok{"Degree"}\NormalTok{, }\AttributeTok{ylab =} \StringTok{"Frequency"}\NormalTok{, }
       \AttributeTok{pch =} \DecValTok{16}\NormalTok{, }\AttributeTok{col =} \StringTok{"lightgreen"}\NormalTok{, }\AttributeTok{cex =} \FloatTok{1.2}\NormalTok{,}
      \AttributeTok{cex.main =}\NormalTok{ main\_cex, }\AttributeTok{cex.lab =}\NormalTok{ lab\_cex, }\AttributeTok{cex.axis =}\NormalTok{ axis\_cex)}
  
  \CommentTok{\# Fit line for visualization}
\NormalTok{  x\_vals }\OtherTok{\textless{}{-}} \FunctionTok{log10}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(}\FunctionTok{names}\NormalTok{(deg\_freq\_n)))}
\NormalTok{  y\_vals }\OtherTok{\textless{}{-}} \FunctionTok{log10}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(deg\_freq\_n))}
  
  \CommentTok{\# Linear regression for  calculation}
\NormalTok{  fit\_n }\OtherTok{\textless{}{-}} \FunctionTok{lm}\NormalTok{(y\_vals }\SpecialCharTok{\textasciitilde{}}\NormalTok{ x\_vals)}
  
  \FunctionTok{abline}\NormalTok{(fit\_n, }\AttributeTok{col =} \StringTok{"darkgreen"}\NormalTok{, }\AttributeTok{lwd =} \DecValTok{2}\NormalTok{, }\AttributeTok{lty =} \DecValTok{2}\NormalTok{)}
  
  \FunctionTok{legend}\NormalTok{(}\StringTok{"topright"}\NormalTok{, }
         \AttributeTok{legend =} \FunctionTok{paste}\NormalTok{(}\StringTok{"γ (MLE) ="}\NormalTok{, }\FunctionTok{round}\NormalTok{(gamma\_n, }\DecValTok{2}\NormalTok{)),}
         \AttributeTok{bty =} \StringTok{"n"}\NormalTok{, }\AttributeTok{cex =} \FloatTok{0.9}\NormalTok{)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-61-1.pdf}}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{par}\NormalTok{(}\AttributeTok{mfrow =} \FunctionTok{c}\NormalTok{(}\DecValTok{1}\NormalTok{, }\DecValTok{1}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\subsubsection{3.4 Hub Identification and
Comparison}\label{hub-identification-and-comparison}

Identify hubs as the top 5\% of nodes with highest degree values.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Define hub thresholds (top 5\% of connected nodes)}
\NormalTok{hub\_threshold\_c }\OtherTok{\textless{}{-}} \FunctionTok{quantile}\NormalTok{(degree\_c[degree\_c }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{], }\FloatTok{0.95}\NormalTok{)}
\NormalTok{hub\_threshold\_n }\OtherTok{\textless{}{-}} \FunctionTok{quantile}\NormalTok{(degree\_n[degree\_n }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{], }\FloatTok{0.95}\NormalTok{)}

\CommentTok{\# Identify hubs}
\NormalTok{hubs\_c }\OtherTok{\textless{}{-}}\NormalTok{ degree\_c[degree\_c }\SpecialCharTok{\textgreater{}=}\NormalTok{ hub\_threshold\_c]}
\NormalTok{hubs\_n }\OtherTok{\textless{}{-}}\NormalTok{ degree\_n[degree\_n }\SpecialCharTok{\textgreater{}=}\NormalTok{ hub\_threshold\_n]}

\FunctionTok{cat}\NormalTok{(}\StringTok{"=== HUB IDENTIFICATION ===}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## === HUB IDENTIFICATION ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Cancer Network:}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Cancer Network:
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Hub threshold (95th percentile): degree ≥ \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, hub\_threshold\_c))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Hub threshold (95th percentile): degree ≥ 63
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Number of hubs: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{length}\NormalTok{(hubs\_c)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Number of hubs: 30
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{Normal Network:}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Normal Network:
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Hub threshold (95th percentile): degree ≥ \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, hub\_threshold\_n))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Hub threshold (95th percentile): degree ≥ 106
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Number of hubs: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{length}\NormalTok{(hubs\_n)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Number of hubs: 27
\end{verbatim}

\paragraph{Hub Comparison Between
Conditions}\label{hub-comparison-between-conditions}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Compare hub sets}
\NormalTok{common\_hubs }\OtherTok{\textless{}{-}} \FunctionTok{intersect}\NormalTok{(}\FunctionTok{names}\NormalTok{(hubs\_c), }\FunctionTok{names}\NormalTok{(hubs\_n))}
\NormalTok{selective\_hubs\_c }\OtherTok{\textless{}{-}} \FunctionTok{setdiff}\NormalTok{(}\FunctionTok{names}\NormalTok{(hubs\_c), }\FunctionTok{names}\NormalTok{(hubs\_n))}
\NormalTok{selective\_hubs\_n }\OtherTok{\textless{}{-}} \FunctionTok{setdiff}\NormalTok{(}\FunctionTok{names}\NormalTok{(hubs\_n), }\FunctionTok{names}\NormalTok{(hubs\_c))}

\FunctionTok{cat}\NormalTok{(}\StringTok{"=== HUB COMPARISON: CANCER vs NORMAL ===}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## === HUB COMPARISON: CANCER vs NORMAL ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Common hubs (present in both): \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{length}\NormalTok{(common\_hubs)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Common hubs (present in both): 0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Cancer{-}selective hubs: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{length}\NormalTok{(selective\_hubs\_c)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Cancer-selective hubs: 30
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Normal{-}selective hubs: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{length}\NormalTok{(selective\_hubs\_n)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Normal-selective hubs: 27
\end{verbatim}

\subsubsection{3.5 Gene Symbol Mapping}\label{gene-symbol-mapping}

Map Ensembl IDs to gene symbols for biological interpretation.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Create gene mapping}
\NormalTok{gene\_mapping }\OtherTok{\textless{}{-}}\NormalTok{ genes.info[, }\FunctionTok{c}\NormalTok{(}\StringTok{"gene\_id"}\NormalTok{, }\StringTok{"gene\_name"}\NormalTok{)]}
\FunctionTok{rownames}\NormalTok{(gene\_mapping) }\OtherTok{\textless{}{-}}\NormalTok{ gene\_mapping}\SpecialCharTok{$}\NormalTok{gene\_id}

\NormalTok{get\_gene\_symbol }\OtherTok{\textless{}{-}} \ControlFlowTok{function}\NormalTok{(ensembl\_id, }\AttributeTok{mapping =}\NormalTok{ gene\_mapping) \{}
  \ControlFlowTok{if}\NormalTok{ (ensembl\_id }\SpecialCharTok{\%in\%} \FunctionTok{rownames}\NormalTok{(mapping)) \{}
\NormalTok{    symbol }\OtherTok{\textless{}{-}}\NormalTok{ mapping[ensembl\_id, }\StringTok{"gene\_name"}\NormalTok{]}
    \ControlFlowTok{if}\NormalTok{ (}\SpecialCharTok{!} \FunctionTok{is.na}\NormalTok{(symbol) }\SpecialCharTok{\&\&}\NormalTok{ symbol }\SpecialCharTok{!=} \StringTok{""}\NormalTok{) }\FunctionTok{return}\NormalTok{(symbol)}
\NormalTok{  \}}
  \FunctionTok{return}\NormalTok{(ensembl\_id)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\paragraph{Top 3 Hubs in Each Network}\label{top-3-hubs-in-each-network}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Top 3 Cancer Hubs}
\NormalTok{top3\_cancer }\OtherTok{\textless{}{-}} \FunctionTok{head}\NormalTok{(degree\_c[degree\_c }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{], }\DecValTok{3}\NormalTok{)}
\NormalTok{top3\_cancer\_symbols }\OtherTok{\textless{}{-}} \FunctionTok{sapply}\NormalTok{(}\FunctionTok{names}\NormalTok{(top3\_cancer), get\_gene\_symbol)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"=== TOP 3 HUBS: CANCER NETWORK ===}\SpecialCharTok{\textbackslash{}n\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## === TOP 3 HUBS: CANCER NETWORK ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \DecValTok{1}\SpecialCharTok{:}\DecValTok{3}\NormalTok{) \{}
  \FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"\%d. \%s (\%s) {-} Degree: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }
\NormalTok{              i, top3\_cancer\_symbols[i], }\FunctionTok{names}\NormalTok{(top3\_cancer)[i], top3\_cancer[i]))}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 1. LAIR1 (ENSG00000167613.16) - Degree: 69
## 2. MYO1F (ENSG00000142347.19) - Degree: 68
## 3. LAPTM5 (ENSG00000162511.8) - Degree: 67
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Top 3 Normal Hubs}
\NormalTok{top3\_normal }\OtherTok{\textless{}{-}} \FunctionTok{head}\NormalTok{(degree\_n[degree\_n }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{], }\DecValTok{3}\NormalTok{)}
\NormalTok{top3\_normal\_symbols }\OtherTok{\textless{}{-}} \FunctionTok{sapply}\NormalTok{(}\FunctionTok{names}\NormalTok{(top3\_normal), get\_gene\_symbol)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{=== TOP 3 HUBS: NORMAL NETWORK ===}\SpecialCharTok{\textbackslash{}n\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## === TOP 3 HUBS: NORMAL NETWORK ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \DecValTok{1}\SpecialCharTok{:}\DecValTok{3}\NormalTok{) \{}
  \FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"\%d. \%s (\%s) {-} Degree: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }
\NormalTok{              i, top3\_normal\_symbols[i], }\FunctionTok{names}\NormalTok{(top3\_normal)[i], top3\_normal[i]))}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 1. SLC34A1 (ENSG00000131183.11) - Degree: 124
## 2. SLC13A3 (ENSG00000158296.14) - Degree: 122
## 3. MME (ENSG00000196549.12) - Degree: 122
\end{verbatim}

\subsubsection{3.6 Alternative Centrality Analysis
(Betweenness)}\label{alternative-centrality-analysis-betweenness}

Compare degree-based hubs with betweenness centrality to identify genes
that serve as bridges.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Convert to igraph objects}
\NormalTok{graph\_cancer }\OtherTok{\textless{}{-}} \FunctionTok{graph\_from\_adjacency\_matrix}\NormalTok{(adj\_mat\_c, }\AttributeTok{mode =} \StringTok{"undirected"}\NormalTok{, }\AttributeTok{diag =} \ConstantTok{FALSE}\NormalTok{)}
\NormalTok{graph\_normal }\OtherTok{\textless{}{-}} \FunctionTok{graph\_from\_adjacency\_matrix}\NormalTok{(adj\_mat\_n, }\AttributeTok{mode =} \StringTok{"undirected"}\NormalTok{, }\AttributeTok{diag =} \ConstantTok{FALSE}\NormalTok{)}

\CommentTok{\# Compute betweenness centrality}
\NormalTok{betweenness\_c }\OtherTok{\textless{}{-}} \FunctionTok{betweenness}\NormalTok{(graph\_cancer, }\AttributeTok{directed =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{normalized =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{betweenness\_n }\OtherTok{\textless{}{-}} \FunctionTok{betweenness}\NormalTok{(graph\_normal, }\AttributeTok{directed =} \ConstantTok{FALSE}\NormalTok{, }\AttributeTok{normalized =} \ConstantTok{TRUE}\NormalTok{)}

\FunctionTok{names}\NormalTok{(betweenness\_c) }\OtherTok{\textless{}{-}} \FunctionTok{V}\NormalTok{(graph\_cancer)}\SpecialCharTok{$}\NormalTok{name}
\FunctionTok{names}\NormalTok{(betweenness\_n) }\OtherTok{\textless{}{-}} \FunctionTok{V}\NormalTok{(graph\_normal)}\SpecialCharTok{$}\NormalTok{name}

\NormalTok{betweenness\_c }\OtherTok{\textless{}{-}} \FunctionTok{sort}\NormalTok{(betweenness\_c, }\AttributeTok{decreasing =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{betweenness\_n }\OtherTok{\textless{}{-}} \FunctionTok{sort}\NormalTok{(betweenness\_n, }\AttributeTok{decreasing =} \ConstantTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Identify top 5\% by betweenness}
\NormalTok{bc\_threshold\_c }\OtherTok{\textless{}{-}} \FunctionTok{quantile}\NormalTok{(betweenness\_c[betweenness\_c }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{], }\FloatTok{0.95}\NormalTok{)}
\NormalTok{bc\_threshold\_n }\OtherTok{\textless{}{-}} \FunctionTok{quantile}\NormalTok{(betweenness\_n[betweenness\_n }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{], }\FloatTok{0.95}\NormalTok{)}

\NormalTok{top\_bc\_c }\OtherTok{\textless{}{-}}\NormalTok{ betweenness\_c[betweenness\_c }\SpecialCharTok{\textgreater{}=}\NormalTok{ bc\_threshold\_c]}
\NormalTok{top\_bc\_n }\OtherTok{\textless{}{-}}\NormalTok{ betweenness\_n[betweenness\_n }\SpecialCharTok{\textgreater{}=}\NormalTok{ bc\_threshold\_n]}

\CommentTok{\# Compare with degree{-}based hubs}
\NormalTok{overlap\_c }\OtherTok{\textless{}{-}} \FunctionTok{intersect}\NormalTok{(}\FunctionTok{names}\NormalTok{(top\_bc\_c), }\FunctionTok{names}\NormalTok{(hubs\_c))}
\NormalTok{overlap\_n }\OtherTok{\textless{}{-}} \FunctionTok{intersect}\NormalTok{(}\FunctionTok{names}\NormalTok{(top\_bc\_n), }\FunctionTok{names}\NormalTok{(hubs\_n))}

\FunctionTok{cat}\NormalTok{(}\StringTok{"=== DEGREE vs BETWEENNESS CENTRALITY COMPARISON ===}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## === DEGREE vs BETWEENNESS CENTRALITY COMPARISON ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"Cancer Network:}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Cancer Network:
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Degree hubs: \%d | Betweenness hubs: \%d | Overlap: \%d (\%.1f\%\%)}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{,}
            \FunctionTok{length}\NormalTok{(hubs\_c), }\FunctionTok{length}\NormalTok{(top\_bc\_c), }\FunctionTok{length}\NormalTok{(overlap\_c),}
            \DecValTok{100} \SpecialCharTok{*} \FunctionTok{length}\NormalTok{(overlap\_c) }\SpecialCharTok{/} \FunctionTok{length}\NormalTok{(}\FunctionTok{union}\NormalTok{(}\FunctionTok{names}\NormalTok{(hubs\_c), }\FunctionTok{names}\NormalTok{(top\_bc\_c)))))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Degree hubs: 30 | Betweenness hubs: 17 | Overlap: 3 (6.8%)
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{Normal Network:}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Normal Network:
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Degree hubs: \%d | Betweenness hubs: \%d | Overlap: \%d (\%.1f\%\%)}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{,}
            \FunctionTok{length}\NormalTok{(hubs\_n), }\FunctionTok{length}\NormalTok{(top\_bc\_n), }\FunctionTok{length}\NormalTok{(overlap\_n),}
            \DecValTok{100} \SpecialCharTok{*} \FunctionTok{length}\NormalTok{(overlap\_n) }\SpecialCharTok{/} \FunctionTok{length}\NormalTok{(}\FunctionTok{union}\NormalTok{(}\FunctionTok{names}\NormalTok{(hubs\_n), }\FunctionTok{names}\NormalTok{(top\_bc\_n)))))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   Degree hubs: 27 | Betweenness hubs: 20 | Overlap: 1 (2.2%)
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{centrality\_cancer\_df }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{Gene =} \FunctionTok{names}\NormalTok{(degree\_c),}
  \AttributeTok{Degree =} \FunctionTok{as.numeric}\NormalTok{(degree\_c),}
  \AttributeTok{Betweenness =}\NormalTok{ betweenness\_c[}\FunctionTok{names}\NormalTok{(degree\_c)]}
\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}
    \AttributeTok{Hub\_Type =} \FunctionTok{case\_when}\NormalTok{(}
\NormalTok{      Gene }\SpecialCharTok{\%in\%}\NormalTok{ overlap\_c }\SpecialCharTok{\textasciitilde{}} \StringTok{"Both"}\NormalTok{,}
\NormalTok{      Gene }\SpecialCharTok{\%in\%} \FunctionTok{names}\NormalTok{(hubs\_c) }\SpecialCharTok{\textasciitilde{}} \StringTok{"Degree Only"}\NormalTok{,}
\NormalTok{      Gene }\SpecialCharTok{\%in\%} \FunctionTok{names}\NormalTok{(top\_bc\_c) }\SpecialCharTok{\textasciitilde{}} \StringTok{"Betweenness Only"}\NormalTok{,}
      \ConstantTok{TRUE} \SpecialCharTok{\textasciitilde{}} \StringTok{"Non{-}Hub"}
\NormalTok{    ),}
    \AttributeTok{Hub\_Type =} \FunctionTok{factor}\NormalTok{(Hub\_Type, }\AttributeTok{levels =} \FunctionTok{c}\NormalTok{(}\StringTok{"Both"}\NormalTok{, }\StringTok{"Degree Only"}\NormalTok{, }\StringTok{"Betweenness Only"}\NormalTok{, }\StringTok{"Non{-}Hub"}\NormalTok{))}
\NormalTok{  )}

\NormalTok{centrality\_normal\_df }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{Gene =} \FunctionTok{names}\NormalTok{(degree\_n),}
  \AttributeTok{Degree =} \FunctionTok{as.numeric}\NormalTok{(degree\_n),}
  \AttributeTok{Betweenness =}\NormalTok{ betweenness\_n[}\FunctionTok{names}\NormalTok{(degree\_n)]}
\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}
    \AttributeTok{Hub\_Type =} \FunctionTok{case\_when}\NormalTok{(}
\NormalTok{      Gene }\SpecialCharTok{\%in\%}\NormalTok{ overlap\_n }\SpecialCharTok{\textasciitilde{}} \StringTok{"Both"}\NormalTok{,}
\NormalTok{      Gene }\SpecialCharTok{\%in\%} \FunctionTok{names}\NormalTok{(hubs\_n) }\SpecialCharTok{\textasciitilde{}} \StringTok{"Degree Only"}\NormalTok{,}
\NormalTok{      Gene }\SpecialCharTok{\%in\%} \FunctionTok{names}\NormalTok{(top\_bc\_n) }\SpecialCharTok{\textasciitilde{}} \StringTok{"Betweenness Only"}\NormalTok{,}
      \ConstantTok{TRUE} \SpecialCharTok{\textasciitilde{}} \StringTok{"Non{-}Hub"}
\NormalTok{    ),}
    \AttributeTok{Hub\_Type =} \FunctionTok{factor}\NormalTok{(Hub\_Type, }\AttributeTok{levels =} \FunctionTok{c}\NormalTok{(}\StringTok{"Both"}\NormalTok{, }\StringTok{"Degree Only"}\NormalTok{, }\StringTok{"Betweenness Only"}\NormalTok{, }\StringTok{"Non{-}Hub"}\NormalTok{))}
\NormalTok{  )}

\NormalTok{my\_colors }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"Both"} \OtherTok{=} \StringTok{"purple"}\NormalTok{, }\StringTok{"Degree Only"} \OtherTok{=} \StringTok{"red"}\NormalTok{,}
               \StringTok{"Betweenness Only"} \OtherTok{=} \StringTok{"blue"}\NormalTok{, }\StringTok{"Non{-}Hub"} \OtherTok{=} \StringTok{"gray70"}\NormalTok{)}
\NormalTok{legend\_breaks }\OtherTok{\textless{}{-}} \FunctionTok{names}\NormalTok{(my\_colors)}

\CommentTok{\# Plot 1: Cancer}
\NormalTok{p1 }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(centrality\_cancer\_df, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ Degree, }\AttributeTok{y =}\NormalTok{ Betweenness, }\AttributeTok{color =}\NormalTok{ Hub\_Type)) }\SpecialCharTok{+}
  \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{alpha =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{size =} \DecValTok{5}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_color\_manual}\NormalTok{(}
    \AttributeTok{values =}\NormalTok{ my\_colors,}
    \AttributeTok{breaks =}\NormalTok{ legend\_breaks,}
    \AttributeTok{limits =}\NormalTok{ legend\_breaks,}
    \AttributeTok{drop =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{name =} \StringTok{"Hub Type"}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}\AttributeTok{title =} \StringTok{"Cancer Network"}\NormalTok{, }\AttributeTok{x =} \StringTok{"Degree"}\NormalTok{, }\AttributeTok{y =} \StringTok{"Betweenness"}\NormalTok{) }\SpecialCharTok{+}
\NormalTok{  base\_theme }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{)  }\CommentTok{\# legenda rimossa qui}

\CommentTok{\# Plot 2: Normal}
\NormalTok{p2 }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(centrality\_normal\_df, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ Degree, }\AttributeTok{y =}\NormalTok{ Betweenness, }\AttributeTok{color =}\NormalTok{ Hub\_Type)) }\SpecialCharTok{+}
  \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{alpha =} \FloatTok{0.7}\NormalTok{, }\AttributeTok{size =} \DecValTok{4}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_color\_manual}\NormalTok{(}
    \AttributeTok{values =}\NormalTok{ my\_colors,}
    \AttributeTok{breaks =}\NormalTok{ legend\_breaks,}
    \AttributeTok{limits =}\NormalTok{ legend\_breaks,}
    \AttributeTok{drop =} \ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{name =} \StringTok{"Hub Type"}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}\AttributeTok{title =} \StringTok{"Normal Network"}\NormalTok{, }\AttributeTok{x =} \StringTok{"Degree"}\NormalTok{, }\AttributeTok{y =} \StringTok{"Betweenness"}\NormalTok{) }\SpecialCharTok{+}
\NormalTok{  base\_theme }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"none"}\NormalTok{)  }\CommentTok{\# legenda rimossa qui}

\CommentTok{\# Estrai la legenda da p1 (o p2, sono uguali per scala)}
\NormalTok{legend }\OtherTok{\textless{}{-}}\NormalTok{ cowplot}\SpecialCharTok{::}\FunctionTok{get\_legend}\NormalTok{(}
\NormalTok{  p1 }\SpecialCharTok{+} \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"bottom"}\NormalTok{, }\AttributeTok{legend.title =} \FunctionTok{element\_blank}\NormalTok{(), , }\AttributeTok{legend.text =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{30}\NormalTok{))}
\NormalTok{)}

\CommentTok{\# Combina i due plot e la legenda centrata in basso}
\NormalTok{combined }\OtherTok{\textless{}{-}}\NormalTok{ cowplot}\SpecialCharTok{::}\FunctionTok{plot\_grid}\NormalTok{(}
\NormalTok{  cowplot}\SpecialCharTok{::}\FunctionTok{plot\_grid}\NormalTok{(p1, p2, }\AttributeTok{ncol =} \DecValTok{2}\NormalTok{, }\AttributeTok{align =} \StringTok{"hv"}\NormalTok{),}
\NormalTok{  legend,}
  \AttributeTok{ncol =} \DecValTok{1}\NormalTok{,}
  \AttributeTok{rel\_heights =} \FunctionTok{c}\NormalTok{(}\DecValTok{1}\NormalTok{, }\FloatTok{0.12}\NormalTok{)  }
\NormalTok{)}

\CommentTok{\# Aggiungi titolo globale}
\NormalTok{title }\OtherTok{\textless{}{-}} \FunctionTok{ggdraw}\NormalTok{() }\SpecialCharTok{+} 
  \FunctionTok{draw\_label}\NormalTok{(}\StringTok{"Degree vs Betweenness Centrality"}\NormalTok{, }\AttributeTok{fontface =} \StringTok{"bold"}\NormalTok{, }\AttributeTok{hjust =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{size =} \DecValTok{40}\NormalTok{)}

\NormalTok{final\_plot }\OtherTok{\textless{}{-}}\NormalTok{ cowplot}\SpecialCharTok{::}\FunctionTok{plot\_grid}\NormalTok{(title, combined, }\AttributeTok{ncol =} \DecValTok{1}\NormalTok{, }\AttributeTok{rel\_heights =} \FunctionTok{c}\NormalTok{(}\FloatTok{0.15}\NormalTok{, }\DecValTok{1}\NormalTok{))}

\NormalTok{final\_plot}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-68-1.pdf}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Correlation between centrality measures}
\NormalTok{cor\_c }\OtherTok{\textless{}{-}} \FunctionTok{cor}\NormalTok{(centrality\_cancer\_df}\SpecialCharTok{$}\NormalTok{Degree, centrality\_cancer\_df}\SpecialCharTok{$}\NormalTok{Betweenness, }
             \AttributeTok{method =} \StringTok{"spearman"}\NormalTok{, }\AttributeTok{use =} \StringTok{"complete.obs"}\NormalTok{)}
\NormalTok{cor\_n }\OtherTok{\textless{}{-}} \FunctionTok{cor}\NormalTok{(centrality\_normal\_df}\SpecialCharTok{$}\NormalTok{Degree, centrality\_normal\_df}\SpecialCharTok{$}\NormalTok{Betweenness, }
             \AttributeTok{method =} \StringTok{"spearman"}\NormalTok{, }\AttributeTok{use =} \StringTok{"complete.obs"}\NormalTok{)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"=== CENTRALITY CORRELATION ===}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## === CENTRALITY CORRELATION ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Cancer Network (Spearman): \%.3f}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, cor\_c))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Cancer Network (Spearman): 0.812
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Normal Network (Spearman): \%.3f}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, cor\_n))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Normal Network (Spearman): 0.850
\end{verbatim}

\subsection{4. Differential Co-Expression
Network}\label{differential-co-expression-network}

\subsubsection{4.1 Network Computation}\label{network-computation-1}

Compute differential co-expression using Fisher's Z-transform to
identify edges with significantly changed correlations.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Sample sizes}
\NormalTok{n\_c }\OtherTok{\textless{}{-}} \FunctionTok{ncol}\NormalTok{(expr\_deg\_cancer)}
\NormalTok{n\_n }\OtherTok{\textless{}{-}} \FunctionTok{ncol}\NormalTok{(expr\_deg\_normal)}

\CommentTok{\# Fisher\textquotesingle{}s Z{-}transform}
\NormalTok{fisher\_z }\OtherTok{\textless{}{-}} \ControlFlowTok{function}\NormalTok{(r) \{}
  \FloatTok{0.5} \SpecialCharTok{*} \FunctionTok{log}\NormalTok{((}\DecValTok{1} \SpecialCharTok{+}\NormalTok{ r) }\SpecialCharTok{/}\NormalTok{ (}\DecValTok{1} \SpecialCharTok{{-}}\NormalTok{ r))}
\NormalTok{\}}

\NormalTok{z\_c }\OtherTok{\textless{}{-}} \FunctionTok{fisher\_z}\NormalTok{(rho.c)}
\NormalTok{z\_n }\OtherTok{\textless{}{-}} \FunctionTok{fisher\_z}\NormalTok{(rho.n)}

\CommentTok{\# Z{-}score for difference}
\NormalTok{se }\OtherTok{\textless{}{-}} \FunctionTok{sqrt}\NormalTok{(}\DecValTok{1}\SpecialCharTok{/}\NormalTok{(n\_c }\SpecialCharTok{{-}} \DecValTok{3}\NormalTok{) }\SpecialCharTok{+} \DecValTok{1}\SpecialCharTok{/}\NormalTok{(n\_n }\SpecialCharTok{{-}} \DecValTok{3}\NormalTok{))}
\NormalTok{z\_diff }\OtherTok{\textless{}{-}}\NormalTok{ (z\_c }\SpecialCharTok{{-}}\NormalTok{ z\_n) }\SpecialCharTok{/}\NormalTok{ se}

\CommentTok{\# Binary adjacency matrix}
\NormalTok{z\_threshold }\OtherTok{\textless{}{-}} \DecValTok{5}  \CommentTok{\# |Z| \textgreater{}= 5 for significant differential co{-}expression}
\NormalTok{adj\_mat\_diff }\OtherTok{\textless{}{-}}\NormalTok{ (}\FunctionTok{abs}\NormalTok{(z\_diff) }\SpecialCharTok{\textgreater{}=}\NormalTok{ z\_threshold) }\SpecialCharTok{*} \DecValTok{1}
\FunctionTok{diag}\NormalTok{(adj\_mat\_diff) }\OtherTok{\textless{}{-}} \DecValTok{0}

\FunctionTok{cat}\NormalTok{(}\StringTok{"=== DIFFERENTIAL NETWORK COMPUTATION ===}\SpecialCharTok{\textbackslash{}n\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## === DIFFERENTIAL NETWORK COMPUTATION ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Z{-}score threshold: |Z| ≥ \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, z\_threshold))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Z-score threshold: |Z| ≥ 5
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Total possible edges: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(adj\_mat\_diff) }\SpecialCharTok{*}\NormalTok{ (}\FunctionTok{nrow}\NormalTok{(adj\_mat\_diff) }\SpecialCharTok{{-}} \DecValTok{1}\NormalTok{) }\SpecialCharTok{/} \DecValTok{2}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Total possible edges: 261726
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Differential edges: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{sum}\NormalTok{(adj\_mat\_diff) }\SpecialCharTok{/} \DecValTok{2}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Differential edges: 3894
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Network density: \%.4f}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{sum}\NormalTok{(adj\_mat\_diff) }\SpecialCharTok{/}\NormalTok{ (}\FunctionTok{nrow}\NormalTok{(adj\_mat\_diff) }\SpecialCharTok{*}\NormalTok{ (}\FunctionTok{nrow}\NormalTok{(adj\_mat\_diff) }\SpecialCharTok{{-}} \DecValTok{1}\NormalTok{))))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Network density: 0.0149
\end{verbatim}

\subsubsection{4.2 Degree Analysis and Scale-Free
Properties}\label{degree-analysis-and-scale-free-properties-1}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Compute differential degree}
\NormalTok{degree\_diff }\OtherTok{\textless{}{-}} \FunctionTok{colSums}\NormalTok{(adj\_mat\_diff }\SpecialCharTok{!=} \DecValTok{0}\NormalTok{)}
\FunctionTok{names}\NormalTok{(degree\_diff) }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(adj\_mat\_diff)}
\NormalTok{degree\_diff }\OtherTok{\textless{}{-}} \FunctionTok{sort}\NormalTok{(degree\_diff, }\AttributeTok{decreasing =} \ConstantTok{TRUE}\NormalTok{)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"=== DIFFERENTIAL NETWORK DEGREE SUMMARY ===}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## === DIFFERENTIAL NETWORK DEGREE SUMMARY ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Max degree: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{max}\NormalTok{(degree\_diff)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Max degree: 103
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Mean degree: \%.2f}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{mean}\NormalTok{(degree\_diff)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Mean degree: 10.76
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Nodes with degree \textgreater{} 0: \%d (\%.1f\%\%)}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }
            \FunctionTok{sum}\NormalTok{(degree\_diff }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{), }\DecValTok{100} \SpecialCharTok{*} \FunctionTok{sum}\NormalTok{(degree\_diff }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{) }\SpecialCharTok{/} \FunctionTok{length}\NormalTok{(degree\_diff)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Nodes with degree > 0: 506 (69.9%)
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{par}\NormalTok{(}\AttributeTok{mfrow =} \FunctionTok{c}\NormalTok{(}\DecValTok{1}\NormalTok{, }\DecValTok{2}\NormalTok{))}

\CommentTok{\# Histogram}
\FunctionTok{hist}\NormalTok{(degree\_diff, }\AttributeTok{breaks =} \DecValTok{50}\NormalTok{, }\AttributeTok{main =} \StringTok{"Differential Network Degree Distribution"}\NormalTok{,}
     \AttributeTok{xlab =} \StringTok{"Degree"}\NormalTok{, }\AttributeTok{col =} \StringTok{"plum"}\NormalTok{, }\AttributeTok{freq =} \ConstantTok{FALSE}\NormalTok{)}

\CommentTok{\# Log{-}log plot}
\NormalTok{deg\_freq\_diff }\OtherTok{\textless{}{-}} \FunctionTok{table}\NormalTok{(degree\_diff[degree\_diff }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{])}
\ControlFlowTok{if}\NormalTok{ (}\FunctionTok{length}\NormalTok{(deg\_freq\_diff) }\SpecialCharTok{\textgreater{}} \DecValTok{1}\NormalTok{) \{}
  \CommentTok{\# Estimate gamma using zeta MLE (function defined earlier)}
\NormalTok{  zeta\_result\_diff }\OtherTok{\textless{}{-}} \FunctionTok{estimate\_gamma\_zeta}\NormalTok{(deg\_freq\_diff)}
\NormalTok{  gamma\_diff }\OtherTok{\textless{}{-}}\NormalTok{ zeta\_result\_diff}\SpecialCharTok{$}\NormalTok{gamma}
  
  \CommentTok{\# Plot}
  \FunctionTok{plot}\NormalTok{(}\FunctionTok{log10}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(}\FunctionTok{names}\NormalTok{(deg\_freq\_diff))), }\FunctionTok{log10}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(deg\_freq\_diff)), }
       \AttributeTok{main =} \StringTok{"Differential: Log{-}Log Degree Distribution"}\NormalTok{, }
       \AttributeTok{xlab =} \StringTok{"Degree"}\NormalTok{, }\AttributeTok{ylab =} \StringTok{"Frequency"}\NormalTok{, }
       \AttributeTok{pch =} \DecValTok{16}\NormalTok{, }\AttributeTok{col =} \StringTok{"plum"}\NormalTok{, }\AttributeTok{cex =} \FloatTok{1.2}\NormalTok{)}
  
  \CommentTok{\# Fit line for visualization}
\NormalTok{  x\_vals }\OtherTok{\textless{}{-}} \FunctionTok{log10}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(}\FunctionTok{names}\NormalTok{(deg\_freq\_diff)))}
\NormalTok{  y\_vals }\OtherTok{\textless{}{-}} \FunctionTok{log10}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(deg\_freq\_diff))}
  
  \CommentTok{\# Linear regression for R² calculation}
\NormalTok{  fit\_diff }\OtherTok{\textless{}{-}} \FunctionTok{lm}\NormalTok{(y\_vals }\SpecialCharTok{\textasciitilde{}}\NormalTok{ x\_vals)}
  
  \FunctionTok{abline}\NormalTok{(fit\_diff, }\AttributeTok{col =} \StringTok{"purple"}\NormalTok{, }\AttributeTok{lwd =} \DecValTok{2}\NormalTok{, }\AttributeTok{lty =} \DecValTok{2}\NormalTok{)}
  
  \FunctionTok{legend}\NormalTok{(}\StringTok{"topright"}\NormalTok{, }
         \AttributeTok{legend =} \FunctionTok{c}\NormalTok{(}\FunctionTok{paste}\NormalTok{(}\StringTok{"γ (MLE) ="}\NormalTok{, }\FunctionTok{round}\NormalTok{(gamma\_diff, }\DecValTok{2}\NormalTok{))), }
         \AttributeTok{bty =} \StringTok{"n"}\NormalTok{, }\AttributeTok{cex =} \FloatTok{0.9}\NormalTok{)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-72-1.pdf}}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{par}\NormalTok{(}\AttributeTok{mfrow =} \FunctionTok{c}\NormalTok{(}\DecValTok{1}\NormalTok{, }\DecValTok{1}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\subsubsection{4.3 Hub Identification and Comparison with Co-expression
Networks}\label{hub-identification-and-comparison-with-co-expression-networks}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Identify differential hubs (top 5\%)}
\NormalTok{hub\_threshold\_diff }\OtherTok{\textless{}{-}} \FunctionTok{quantile}\NormalTok{(degree\_diff[degree\_diff }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{], }\FloatTok{0.95}\NormalTok{)}
\NormalTok{hubs\_diff }\OtherTok{\textless{}{-}}\NormalTok{ degree\_diff[degree\_diff }\SpecialCharTok{\textgreater{}=}\NormalTok{ hub\_threshold\_diff]}

\FunctionTok{cat}\NormalTok{(}\StringTok{"=== DIFFERENTIAL HUB IDENTIFICATION ===}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## === DIFFERENTIAL HUB IDENTIFICATION ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Hub threshold (95th percentile): degree ≥ \%.2f }\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, hub\_threshold\_diff))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Hub threshold (95th percentile): degree ≥ 57.00
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Number of differential hubs: \%d }\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{length}\NormalTok{(hubs\_diff)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Number of differential hubs: 28
\end{verbatim}

\paragraph{Compare with Task 3 Hubs}\label{compare-with-task-3-hubs}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Compare differential hubs with cancer and normal hubs}
\NormalTok{common\_diff\_c }\OtherTok{\textless{}{-}} \FunctionTok{intersect}\NormalTok{(}\FunctionTok{names}\NormalTok{(hubs\_diff), }\FunctionTok{names}\NormalTok{(hubs\_c))}
\NormalTok{common\_diff\_n }\OtherTok{\textless{}{-}} \FunctionTok{intersect}\NormalTok{(}\FunctionTok{names}\NormalTok{(hubs\_diff), }\FunctionTok{names}\NormalTok{(hubs\_n))}
\NormalTok{selective\_diff }\OtherTok{\textless{}{-}} \FunctionTok{setdiff}\NormalTok{(}\FunctionTok{names}\NormalTok{(hubs\_diff), }\FunctionTok{union}\NormalTok{(}\FunctionTok{names}\NormalTok{(hubs\_c), }\FunctionTok{names}\NormalTok{(hubs\_n)))}

\FunctionTok{cat}\NormalTok{(}\StringTok{"=== COMPARISON: DIFFERENTIAL vs CO{-}EXPRESSION HUBS ===}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## === COMPARISON: DIFFERENTIAL vs CO-EXPRESSION HUBS ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Differential hubs also in Cancer network: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{length}\NormalTok{(common\_diff\_c)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Differential hubs also in Cancer network: 0
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Differential hubs also in Normal network: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{length}\NormalTok{(common\_diff\_n)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Differential hubs also in Normal network: 14
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Differential hubs unique (not hubs in either): \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{length}\NormalTok{(selective\_diff)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Differential hubs unique (not hubs in either): 14
\end{verbatim}

\subsubsection{4.4 Top 3 Differential Hubs
Analysis}\label{top-3-differential-hubs-analysis}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Top 3 differential hubs}
\NormalTok{top3\_diff }\OtherTok{\textless{}{-}} \FunctionTok{head}\NormalTok{(degree\_diff[degree\_diff }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{], }\DecValTok{3}\NormalTok{)}
\NormalTok{top3\_diff\_symbols }\OtherTok{\textless{}{-}} \FunctionTok{sapply}\NormalTok{(}\FunctionTok{names}\NormalTok{(top3\_diff), get\_gene\_symbol)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"=== TOP 3 HUBS: DIFFERENTIAL NETWORK ===}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## === TOP 3 HUBS: DIFFERENTIAL NETWORK ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \DecValTok{1}\SpecialCharTok{:}\DecValTok{3}\NormalTok{) \{}
  \FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"\%d. \%s (\%s) {-} Differential Degree: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }
\NormalTok{              i, top3\_diff\_symbols[i], }\FunctionTok{names}\NormalTok{(top3\_diff)[i], top3\_diff[i]))}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 1. STRA6 (ENSG00000137868.19) - Differential Degree: 103
## 2. PODXL (ENSG00000128567.17) - Differential Degree: 87
## 3. CALB1 (ENSG00000104327.7) - Differential Degree: 81
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"=== DETAILED CONNECTIVITY ANALYSIS ===}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## === DETAILED CONNECTIVITY ANALYSIS ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{top\_diff\_genes }\OtherTok{\textless{}{-}} \FunctionTok{names}\NormalTok{(top3\_diff)}

\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \DecValTok{1}\SpecialCharTok{:}\DecValTok{3}\NormalTok{) \{}
\NormalTok{  gene }\OtherTok{\textless{}{-}}\NormalTok{ top\_diff\_genes[i]}
\NormalTok{  symbol }\OtherTok{\textless{}{-}} \FunctionTok{get\_gene\_symbol}\NormalTok{(gene)}
  
  \CommentTok{\# Find neighbors in each network}
\NormalTok{  neighbors\_n }\OtherTok{\textless{}{-}} \FunctionTok{colnames}\NormalTok{(adj\_mat\_n)[adj\_mat\_n[gene, ] }\SpecialCharTok{!=} \DecValTok{0}\NormalTok{]}
\NormalTok{  neighbors\_c }\OtherTok{\textless{}{-}} \FunctionTok{colnames}\NormalTok{(adj\_mat\_c)[adj\_mat\_c[gene, ] }\SpecialCharTok{!=} \DecValTok{0}\NormalTok{]}
\NormalTok{  neighbors\_diff }\OtherTok{\textless{}{-}} \FunctionTok{colnames}\NormalTok{(adj\_mat\_diff)[adj\_mat\_diff[gene, ] }\SpecialCharTok{!=} \DecValTok{0}\NormalTok{]}
  
  \CommentTok{\# Calculate overlaps}
\NormalTok{  preserved }\OtherTok{\textless{}{-}} \FunctionTok{intersect}\NormalTok{(neighbors\_n, neighbors\_c)}
\NormalTok{  lost }\OtherTok{\textless{}{-}} \FunctionTok{setdiff}\NormalTok{(neighbors\_n, neighbors\_c)}
\NormalTok{  gained }\OtherTok{\textless{}{-}} \FunctionTok{setdiff}\NormalTok{(neighbors\_c, neighbors\_n)}
  
  \CommentTok{\# Z{-}score direction}
\NormalTok{  z\_vals }\OtherTok{\textless{}{-}}\NormalTok{ z\_diff[gene, neighbors\_diff]}
\NormalTok{  n\_pos\_z }\OtherTok{\textless{}{-}} \FunctionTok{sum}\NormalTok{(z\_vals }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{  n\_neg\_z }\OtherTok{\textless{}{-}} \FunctionTok{sum}\NormalTok{(z\_vals }\SpecialCharTok{\textless{}} \DecValTok{0}\NormalTok{, }\AttributeTok{na.rm =} \ConstantTok{TRUE}\NormalTok{)}
  
  \FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"{-}{-}{-} \%d. \%s {-}{-}{-}}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, i, symbol))}
  \FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Normal Degree: \%d → Cancer Degree: \%d (Change: \%+d)}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }
\NormalTok{              degree\_n[gene], degree\_c[gene], degree\_c[gene] }\SpecialCharTok{{-}}\NormalTok{ degree\_n[gene]))}
  \FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Differential Degree: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, degree\_diff[gene]))}
  \FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Connections preserved: \%d | Lost: \%d | Gained: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }
              \FunctionTok{length}\NormalTok{(preserved), }\FunctionTok{length}\NormalTok{(lost), }\FunctionTok{length}\NormalTok{(gained)))}
  \FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"  Edges with ↑ correlation in cancer: \%d | ↓ correlation: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }
\NormalTok{              n\_pos\_z, n\_neg\_z))}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## --- 1. STRA6 ---
##   Normal Degree: 106 → Cancer Degree: 3 (Change: -103)
##   Differential Degree: 103
##   Connections preserved: 0 | Lost: 106 | Gained: 3
##   Edges with ↑ correlation in cancer: 22 | ↓ correlation: 81
## --- 2. PODXL ---
##   Normal Degree: 81 → Cancer Degree: 27 (Change: -54)
##   Differential Degree: 87
##   Connections preserved: 5 | Lost: 76 | Gained: 22
##   Edges with ↑ correlation in cancer: 23 | ↓ correlation: 64
## --- 3. CALB1 ---
##   Normal Degree: 88 → Cancer Degree: 0 (Change: -88)
##   Differential Degree: 81
##   Connections preserved: 0 | Lost: 88 | Gained: 0
##   Edges with ↑ correlation in cancer: 10 | ↓ correlation: 71
\end{verbatim}

Create Comparison Table for Top 3 Differential Hubs

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Identify the Top 3 genes from the Differential Network}
\NormalTok{top\_diff\_genes }\OtherTok{\textless{}{-}} \FunctionTok{names}\NormalTok{(top3\_diff)}

\CommentTok{\# Build the comparison dataframe}
\NormalTok{comparison\_table }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{Gene\_Symbol =} \FunctionTok{sapply}\NormalTok{(top\_diff\_genes, get\_gene\_symbol), }\CommentTok{\# Convert Ensembl ID to Symbol}
  \AttributeTok{Ensembl\_ID =}\NormalTok{ top\_diff\_genes,}
  \AttributeTok{Degree\_Normal =}\NormalTok{ degree\_n[top\_diff\_genes],              }\CommentTok{\# Degree in Normal Network}
  \AttributeTok{Degree\_Cancer =}\NormalTok{ degree\_c[top\_diff\_genes],              }\CommentTok{\# Degree in Cancer Network}
  \AttributeTok{Degree\_Differential =}\NormalTok{ degree\_diff[top\_diff\_genes]      }\CommentTok{\# Degree in Differential Network}
\NormalTok{)}

\FunctionTok{print}\NormalTok{(comparison\_table) }\CommentTok{\# Top 3 Differential Hubs}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##                    Gene_Symbol         Ensembl_ID Degree_Normal Degree_Cancer
## ENSG00000137868.19       STRA6 ENSG00000137868.19           106             3
## ENSG00000128567.17       PODXL ENSG00000128567.17            81            27
## ENSG00000104327.7        CALB1  ENSG00000104327.7            88             0
##                    Degree_Differential
## ENSG00000137868.19                 103
## ENSG00000128567.17                  87
## ENSG00000104327.7                   81
\end{verbatim}

Visualization

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{plot\_data }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{Gene =} \FunctionTok{rep}\NormalTok{(comparison\_table}\SpecialCharTok{$}\NormalTok{Gene\_Symbol, }\DecValTok{2}\NormalTok{),}
  \AttributeTok{Condition =} \FunctionTok{rep}\NormalTok{(}\FunctionTok{c}\NormalTok{(}\StringTok{"Normal"}\NormalTok{, }\StringTok{"Cancer"}\NormalTok{), }\AttributeTok{each =} \DecValTok{3}\NormalTok{),}
  \AttributeTok{Degree =} \FunctionTok{c}\NormalTok{(comparison\_table}\SpecialCharTok{$}\NormalTok{Degree\_Normal, comparison\_table}\SpecialCharTok{$}\NormalTok{Degree\_Cancer)}
\NormalTok{)}
\NormalTok{plot\_data}\SpecialCharTok{$}\NormalTok{Gene }\OtherTok{\textless{}{-}} \FunctionTok{factor}\NormalTok{(plot\_data}\SpecialCharTok{$}\NormalTok{Gene, }\AttributeTok{levels =}\NormalTok{ comparison\_table}\SpecialCharTok{$}\NormalTok{Gene\_Symbol)}
\NormalTok{plot\_data}\SpecialCharTok{$}\NormalTok{Condition }\OtherTok{\textless{}{-}} \FunctionTok{factor}\NormalTok{(plot\_data}\SpecialCharTok{$}\NormalTok{Condition, }\AttributeTok{levels =} \FunctionTok{c}\NormalTok{(}\StringTok{"Normal"}\NormalTok{, }\StringTok{"Cancer"}\NormalTok{))}

\FunctionTok{ggplot}\NormalTok{(plot\_data, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ Gene, }\AttributeTok{y =}\NormalTok{ Degree, }\AttributeTok{fill =}\NormalTok{ Condition)) }\SpecialCharTok{+}
  \FunctionTok{geom\_bar}\NormalTok{(}\AttributeTok{stat =} \StringTok{"identity"}\NormalTok{, }\AttributeTok{position =} \StringTok{"dodge"}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.8}\NormalTok{, }\AttributeTok{width =} \FloatTok{0.7}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_text}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{label =}\NormalTok{ Degree), }\AttributeTok{position =} \FunctionTok{position\_dodge}\NormalTok{(}\AttributeTok{width =} \FloatTok{0.7}\NormalTok{), }
            \AttributeTok{vjust =} \SpecialCharTok{{-}}\FloatTok{0.5}\NormalTok{, }\AttributeTok{size =} \DecValTok{10}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}\StringTok{"Normal"} \OtherTok{=} \StringTok{"\#4DAF4A"}\NormalTok{, }\StringTok{"Cancer"} \OtherTok{=} \StringTok{"\#E41A1C"}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}\AttributeTok{title =} \StringTok{"Top 3 Differential Hubs: Degree Comparison"}\NormalTok{,}
       \AttributeTok{x =} \StringTok{"Gene Symbol"}\NormalTok{, }
       \AttributeTok{y =} \StringTok{"Degree"}\NormalTok{) }\SpecialCharTok{+}
\NormalTok{        base\_theme }\SpecialCharTok{+} 
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position =} \StringTok{"bottom"}\NormalTok{,}
        \AttributeTok{legend.title =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{30}\NormalTok{, }\AttributeTok{face =} \StringTok{"bold"}\NormalTok{),}
        \AttributeTok{legend.text =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{30}\NormalTok{)}
\NormalTok{    )}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-78-1.pdf}}

\subsubsection{4.5 Hub Transition
Analysis}\label{hub-transition-analysis}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Full comparison for all differential hubs}
\NormalTok{all\_diff\_hub\_genes }\OtherTok{\textless{}{-}} \FunctionTok{names}\NormalTok{(hubs\_diff)}

\NormalTok{full\_comparison }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{Gene\_Symbol =} \FunctionTok{sapply}\NormalTok{(all\_diff\_hub\_genes, get\_gene\_symbol),}
  \AttributeTok{Ensembl\_ID =}\NormalTok{ all\_diff\_hub\_genes,}
  \AttributeTok{Degree\_Normal =}\NormalTok{ degree\_n[all\_diff\_hub\_genes],}
  \AttributeTok{Degree\_Cancer =}\NormalTok{ degree\_c[all\_diff\_hub\_genes],}
  \AttributeTok{Degree\_Differential =}\NormalTok{ degree\_diff[all\_diff\_hub\_genes],}
  \AttributeTok{Is\_Hub\_Normal =}\NormalTok{ all\_diff\_hub\_genes }\SpecialCharTok{\%in\%} \FunctionTok{names}\NormalTok{(hubs\_n),}
  \AttributeTok{Is\_Hub\_Cancer =}\NormalTok{ all\_diff\_hub\_genes }\SpecialCharTok{\%in\%} \FunctionTok{names}\NormalTok{(hubs\_c),}
  \AttributeTok{stringsAsFactors =} \ConstantTok{FALSE}
\NormalTok{)}

\NormalTok{full\_comparison}\SpecialCharTok{$}\NormalTok{Hub\_Transition }\OtherTok{\textless{}{-}} \FunctionTok{case\_when}\NormalTok{(}
\NormalTok{  full\_comparison}\SpecialCharTok{$}\NormalTok{Is\_Hub\_Normal }\SpecialCharTok{\&}\NormalTok{ full\_comparison}\SpecialCharTok{$}\NormalTok{Is\_Hub\_Cancer }\SpecialCharTok{\textasciitilde{}} \StringTok{"Hub in Both"}\NormalTok{,}
\NormalTok{  full\_comparison}\SpecialCharTok{$}\NormalTok{Is\_Hub\_Normal }\SpecialCharTok{\&} \SpecialCharTok{!}\NormalTok{full\_comparison}\SpecialCharTok{$}\NormalTok{Is\_Hub\_Cancer }\SpecialCharTok{\textasciitilde{}} \StringTok{"Lost Hub Status"}\NormalTok{,}
  \SpecialCharTok{!}\NormalTok{full\_comparison}\SpecialCharTok{$}\NormalTok{Is\_Hub\_Normal }\SpecialCharTok{\&}\NormalTok{ full\_comparison}\SpecialCharTok{$}\NormalTok{Is\_Hub\_Cancer }\SpecialCharTok{\textasciitilde{}} \StringTok{"Gained Hub Status"}\NormalTok{,}
  \ConstantTok{TRUE} \SpecialCharTok{\textasciitilde{}} \StringTok{"Differential Hub Only"}
\NormalTok{)}

\NormalTok{full\_comparison }\OtherTok{\textless{}{-}}\NormalTok{ full\_comparison[}\FunctionTok{order}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{full\_comparison}\SpecialCharTok{$}\NormalTok{Degree\_Differential), ]}

\CommentTok{\# Summary}
\NormalTok{transition\_summary }\OtherTok{\textless{}{-}} \FunctionTok{table}\NormalTok{(full\_comparison}\SpecialCharTok{$}\NormalTok{Hub\_Transition)}
\FunctionTok{cat}\NormalTok{(}\StringTok{"=== HUB TRANSITION SUMMARY ===}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## === HUB TRANSITION SUMMARY ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(transition\_summary)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## Differential Hub Only       Lost Hub Status 
##                    14                    14
\end{verbatim}

\subsubsection{4.6 Subnetwork
Visualization}\label{subnetwork-visualization}

\paragraph{Top Differential Hub
Subnetwork}\label{top-differential-hub-subnetwork}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{top\_gene }\OtherTok{\textless{}{-}} \FunctionTok{names}\NormalTok{(degree\_diff)[}\DecValTok{1}\NormalTok{]}
\NormalTok{top\_gene\_symbol }\OtherTok{\textless{}{-}} \FunctionTok{get\_gene\_symbol}\NormalTok{(top\_gene)}

\NormalTok{neighbors\_idx }\OtherTok{\textless{}{-}} \FunctionTok{which}\NormalTok{(adj\_mat\_diff[top\_gene, ] }\SpecialCharTok{!=} \DecValTok{0}\NormalTok{)}
\NormalTok{neighbor\_names }\OtherTok{\textless{}{-}} \FunctionTok{colnames}\NormalTok{(adj\_mat\_diff)[neighbors\_idx]}
\NormalTok{top\_n }\OtherTok{\textless{}{-}} \DecValTok{20}
\NormalTok{top\_neighbors }\OtherTok{\textless{}{-}} \FunctionTok{names}\NormalTok{(}\FunctionTok{sort}\NormalTok{(degree\_diff[neighbor\_names], }\AttributeTok{decreasing =} \ConstantTok{TRUE}\NormalTok{))[}\DecValTok{1}\SpecialCharTok{:}\FunctionTok{min}\NormalTok{(top\_n, }\FunctionTok{length}\NormalTok{(neighbor\_names))]}

\NormalTok{subnet\_nodes }\OtherTok{\textless{}{-}} \FunctionTok{unique}\NormalTok{(}\FunctionTok{c}\NormalTok{(top\_gene, top\_neighbors))}
\NormalTok{subnet\_adj }\OtherTok{\textless{}{-}}\NormalTok{ adj\_mat\_diff[subnet\_nodes, subnet\_nodes]}

\NormalTok{g }\OtherTok{\textless{}{-}} \FunctionTok{graph\_from\_adjacency\_matrix}\NormalTok{(subnet\_adj, }\AttributeTok{mode =} \StringTok{"undirected"}\NormalTok{, }\AttributeTok{diag =} \ConstantTok{FALSE}\NormalTok{)}
\FunctionTok{V}\NormalTok{(g)}\SpecialCharTok{$}\NormalTok{label }\OtherTok{\textless{}{-}} \FunctionTok{sapply}\NormalTok{(}\FunctionTok{V}\NormalTok{(g)}\SpecialCharTok{$}\NormalTok{name, get\_gene\_symbol)}
\FunctionTok{V}\NormalTok{(g)}\SpecialCharTok{$}\NormalTok{is\_hub }\OtherTok{\textless{}{-}} \FunctionTok{V}\NormalTok{(g)}\SpecialCharTok{$}\NormalTok{name }\SpecialCharTok{==}\NormalTok{ top\_gene}

\NormalTok{edge\_list }\OtherTok{\textless{}{-}} \FunctionTok{as\_edgelist}\NormalTok{(g)}
\NormalTok{edge\_signs }\OtherTok{\textless{}{-}} \FunctionTok{sapply}\NormalTok{(}\DecValTok{1}\SpecialCharTok{:}\FunctionTok{nrow}\NormalTok{(edge\_list), }\ControlFlowTok{function}\NormalTok{(i) \{}
\NormalTok{  node1 }\OtherTok{\textless{}{-}}\NormalTok{ edge\_list[i, }\DecValTok{1}\NormalTok{]}
\NormalTok{  node2 }\OtherTok{\textless{}{-}}\NormalTok{ edge\_list[i, }\DecValTok{2}\NormalTok{]}
\NormalTok{  z\_val }\OtherTok{\textless{}{-}}\NormalTok{ z\_diff[node1, node2]}
  \ControlFlowTok{if}\NormalTok{ (z\_val }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{) \{}
    \FunctionTok{return}\NormalTok{(}\StringTok{"Increased in Cancer"}\NormalTok{)}
\NormalTok{  \} }\ControlFlowTok{else}\NormalTok{ \{}
    \FunctionTok{return}\NormalTok{(}\StringTok{"Decreased in Cancer"}\NormalTok{)}
\NormalTok{  \}}
\NormalTok{\})}

\FunctionTok{E}\NormalTok{(g)}\SpecialCharTok{$}\NormalTok{correlation\_change }\OtherTok{\textless{}{-}}\NormalTok{ edge\_signs}


\FunctionTok{set.seed}\NormalTok{(}\DecValTok{42}\NormalTok{)}
\FunctionTok{ggraph}\NormalTok{(g, }\AttributeTok{layout =} \StringTok{"fr"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_edge\_link}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{color =}\NormalTok{ correlation\_change), }\AttributeTok{alpha =} \FloatTok{0.10}\NormalTok{, }\AttributeTok{width =} \FloatTok{0.8}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_node\_point}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{color =}\NormalTok{ is\_hub), }\AttributeTok{size =} \DecValTok{10}\NormalTok{, }\AttributeTok{alpha =} \FloatTok{0.9}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_node\_text}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{label =}\NormalTok{ label), }\AttributeTok{repel =} \ConstantTok{TRUE}\NormalTok{, }\AttributeTok{size =} \DecValTok{10}\NormalTok{, }\AttributeTok{max.overlaps =} \DecValTok{20}\NormalTok{) }\SpecialCharTok{+}


  \FunctionTok{scale\_edge\_color\_manual}\NormalTok{(}
    \AttributeTok{values =} \FunctionTok{c}\NormalTok{(}
      \StringTok{"Increased in Cancer"} \OtherTok{=} \StringTok{"\#D81B60"}\NormalTok{,}
      \StringTok{"Decreased in Cancer"} \OtherTok{=} \StringTok{"\#1E88E5"}
\NormalTok{    ),}
    \AttributeTok{name =} \StringTok{"Correlation Change"}
\NormalTok{  ) }\SpecialCharTok{+}


  \FunctionTok{scale\_color\_manual}\NormalTok{(}
    \AttributeTok{values =} \FunctionTok{c}\NormalTok{(}
      \StringTok{"FALSE"} \OtherTok{=} \StringTok{"\#b89ef4ff"}\NormalTok{,}
      \StringTok{"TRUE"}  \OtherTok{=} \StringTok{"\#ffcb8cff"}
\NormalTok{    ),}
    \AttributeTok{labels =} \FunctionTok{c}\NormalTok{(}\StringTok{"Neighbor"}\NormalTok{, }\StringTok{"Hub"}\NormalTok{),}
    \AttributeTok{name =} \StringTok{"Node Type"}
\NormalTok{  ) }\SpecialCharTok{+}

  \FunctionTok{theme\_void}\NormalTok{() }\SpecialCharTok{+}
  \FunctionTok{ggtitle}\NormalTok{(}\FunctionTok{paste0}\NormalTok{(top\_gene\_symbol, }\StringTok{" Differential Subnetwork (Top "}\NormalTok{, top\_n, }\StringTok{")"}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}
    \AttributeTok{plot.title =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{hjust =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{face =} \StringTok{"bold"}\NormalTok{, }\AttributeTok{size =} \DecValTok{40}\NormalTok{),}
    \AttributeTok{legend.position =} \StringTok{"bottom"}\NormalTok{,}
    \AttributeTok{legend.text =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{30}\NormalTok{),}
    \AttributeTok{legend.title =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{30}\NormalTok{, }\AttributeTok{face =} \StringTok{"bold"}\NormalTok{),}
    \AttributeTok{legend.box =} \StringTok{"vertical"}

\NormalTok{  )}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-80-1.pdf}}

\subsubsection{4.7 Signed Differential Network
Analysis}\label{signed-differential-network-analysis}

Distinguish between edges where correlations increased vs decreased in
cancer.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Positive subnetwork (correlations INCREASED in cancer)}
\NormalTok{adj\_mat\_diff\_positive }\OtherTok{\textless{}{-}}\NormalTok{ (z\_diff }\SpecialCharTok{\textgreater{}=}\NormalTok{ z\_threshold) }\SpecialCharTok{*} \DecValTok{1}
\FunctionTok{diag}\NormalTok{(adj\_mat\_diff\_positive) }\OtherTok{\textless{}{-}} \DecValTok{0}

\CommentTok{\# Negative subnetwork (correlations DECREASED in cancer)}
\NormalTok{adj\_mat\_diff\_negative }\OtherTok{\textless{}{-}}\NormalTok{ (z\_diff }\SpecialCharTok{\textless{}=} \SpecialCharTok{{-}}\NormalTok{z\_threshold) }\SpecialCharTok{*} \DecValTok{1}
\FunctionTok{diag}\NormalTok{(adj\_mat\_diff\_negative) }\OtherTok{\textless{}{-}} \DecValTok{0}

\FunctionTok{cat}\NormalTok{(}\StringTok{"=== SIGNED DIFFERENTIAL NETWORK STATISTICS ===}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## === SIGNED DIFFERENTIAL NETWORK STATISTICS ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Positive edges (correlation ↑ in cancer): \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{sum}\NormalTok{(adj\_mat\_diff\_positive) }\SpecialCharTok{/} \DecValTok{2}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Positive edges (correlation ↑ in cancer): 989
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Negative edges (correlation ↓ in cancer): \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{sum}\NormalTok{(adj\_mat\_diff\_negative) }\SpecialCharTok{/} \DecValTok{2}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Negative edges (correlation ↓ in cancer): 2905
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Ratio (Positive/Negative): \%.2f}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }
            \FunctionTok{sum}\NormalTok{(adj\_mat\_diff\_positive) }\SpecialCharTok{/} \FunctionTok{max}\NormalTok{(}\FunctionTok{sum}\NormalTok{(adj\_mat\_diff\_negative), }\DecValTok{1}\NormalTok{)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Ratio (Positive/Negative): 0.34
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Compute signed degrees}
\NormalTok{degree\_positive }\OtherTok{\textless{}{-}} \FunctionTok{colSums}\NormalTok{(adj\_mat\_diff\_positive)}
\FunctionTok{names}\NormalTok{(degree\_positive) }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(adj\_mat\_diff\_positive)}
\NormalTok{degree\_positive }\OtherTok{\textless{}{-}} \FunctionTok{sort}\NormalTok{(degree\_positive, }\AttributeTok{decreasing =} \ConstantTok{TRUE}\NormalTok{)}

\NormalTok{degree\_negative }\OtherTok{\textless{}{-}} \FunctionTok{colSums}\NormalTok{(adj\_mat\_diff\_negative)}
\FunctionTok{names}\NormalTok{(degree\_negative) }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(adj\_mat\_diff\_negative)}
\NormalTok{degree\_negative }\OtherTok{\textless{}{-}} \FunctionTok{sort}\NormalTok{(degree\_negative, }\AttributeTok{decreasing =} \ConstantTok{TRUE}\NormalTok{)}

\CommentTok{\# Identify signed hubs}
\NormalTok{pos\_threshold }\OtherTok{\textless{}{-}} \FunctionTok{quantile}\NormalTok{(degree\_positive[degree\_positive }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{], }\FloatTok{0.95}\NormalTok{)}
\NormalTok{neg\_threshold }\OtherTok{\textless{}{-}} \FunctionTok{quantile}\NormalTok{(degree\_negative[degree\_negative }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{], }\FloatTok{0.95}\NormalTok{)}

\NormalTok{hubs\_positive }\OtherTok{\textless{}{-}}\NormalTok{ degree\_positive[degree\_positive }\SpecialCharTok{\textgreater{}=}\NormalTok{ pos\_threshold }\SpecialCharTok{\&}\NormalTok{ degree\_positive }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{]}
\NormalTok{hubs\_negative }\OtherTok{\textless{}{-}}\NormalTok{ degree\_negative[degree\_negative }\SpecialCharTok{\textgreater{}=}\NormalTok{ neg\_threshold }\SpecialCharTok{\&}\NormalTok{ degree\_negative }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{]}

\FunctionTok{cat}\NormalTok{(}\StringTok{"=== SIGNED HUBS ===}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## === SIGNED HUBS ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Positive hubs (correlations ↑ in cancer): \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{length}\NormalTok{(hubs\_positive)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Positive hubs (correlations ↑ in cancer): 22
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"Negative hubs (correlations ↓ in cancer): \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, }\FunctionTok{length}\NormalTok{(hubs\_negative)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Negative hubs (correlations ↓ in cancer): 21
\end{verbatim}

\paragraph{Top 3 Positive and Negative
Hubs}\label{top-3-positive-and-negative-hubs}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Top positive hubs}
\NormalTok{top3\_pos }\OtherTok{\textless{}{-}} \FunctionTok{head}\NormalTok{(hubs\_positive, }\DecValTok{3}\NormalTok{)}
\NormalTok{top3\_pos\_symbols }\OtherTok{\textless{}{-}} \FunctionTok{sapply}\NormalTok{(}\FunctionTok{names}\NormalTok{(top3\_pos), get\_gene\_symbol)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"=== TOP 3 POSITIVE HUBS ===}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## === TOP 3 POSITIVE HUBS ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"(Genes whose correlations INCREASED in cancer)}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## (Genes whose correlations INCREASED in cancer)
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \FunctionTok{seq\_along}\NormalTok{(top3\_pos)) \{}
  \FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"\%d.  \%s {-} Positive Degree: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, i, top3\_pos\_symbols[i], top3\_pos[i]))}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 1.  LPL - Positive Degree: 53
## 2.  SLFN13 - Positive Degree: 39
## 3.  HSPA2 - Positive Degree: 36
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Top negative hubs}
\NormalTok{top3\_neg }\OtherTok{\textless{}{-}} \FunctionTok{head}\NormalTok{(hubs\_negative, }\DecValTok{3}\NormalTok{)}
\NormalTok{top3\_neg\_symbols }\OtherTok{\textless{}{-}} \FunctionTok{sapply}\NormalTok{(}\FunctionTok{names}\NormalTok{(top3\_neg), get\_gene\_symbol)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{=== TOP 3 NEGATIVE HUBS ===}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## === TOP 3 NEGATIVE HUBS ===
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"(Genes whose correlations DECREASED in cancer)}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## (Genes whose correlations DECREASED in cancer)
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \FunctionTok{seq\_along}\NormalTok{(top3\_neg)) \{}
  \FunctionTok{cat}\NormalTok{(}\FunctionTok{sprintf}\NormalTok{(}\StringTok{"\%d. \%s {-} Negative Degree: \%d}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{, i, top3\_neg\_symbols[i], top3\_neg[i]))}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 1. STRA6 - Negative Degree: 81
## 2. GPX3 - Negative Degree: 76
## 3. CES3 - Negative Degree: 75
\end{verbatim}

\subsection{5. Patient Similarity Network
(PSN)}\label{patient-similarity-network-psn}

\subsubsection{a. Computation of the
PSN}\label{a.-computation-of-the-psn}

\paragraph{Data preparation}\label{data-preparation-1}

We construct the Patient Similarity Network using normalized and
log-transformed tumor expression data restricted to the DEGs. Raw counts
are not used, as highly expressed genes would dominate similarity
estimates.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{target\_data }\OtherTok{\textless{}{-}} \FunctionTok{as.matrix}\NormalTok{(expr\_deg\_cancer)}
\end{Highlighting}
\end{Shaded}

Check dimension

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(}\FunctionTok{dim}\NormalTok{(target\_data)) }\CommentTok{\# we expect N\_DEGs x 32 patients}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 724  32
\end{verbatim}

\paragraph{Calculation of the similarity
matrix}\label{calculation-of-the-similarity-matrix}

Calculate Pearson correlation between patients (columns)

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Patient similarity matrix (patients x patients)}
\NormalTok{psn\_full }\OtherTok{\textless{}{-}} \FunctionTok{cor}\NormalTok{(target\_data, }\AttributeTok{method =} \StringTok{"pearson"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\paragraph{Clean-up of the matrix}\label{clean-up-of-the-matrix}

Set the diagonal to 0 to remove self-loops

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Remove self{-}loops}
\FunctionTok{diag}\NormalTok{(psn\_full) }\OtherTok{\textless{}{-}} \DecValTok{0}

\CommentTok{\# Keep only positive similarities}
\NormalTok{psn\_full[psn\_full }\SpecialCharTok{\textless{}} \DecValTok{0}\NormalTok{] }\OtherTok{\textless{}{-}} \DecValTok{0}

\CommentTok{\# Sanity check}
\FunctionTok{stopifnot}\NormalTok{(}\FunctionTok{all}\NormalTok{(}\FunctionTok{is.finite}\NormalTok{(psn\_full)))}
\FunctionTok{dim}\NormalTok{(psn\_full)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 32 32
\end{verbatim}

\subsubsection{b. Community detection}\label{b.-community-detection}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Build graph from FULL similarity matrix}
\NormalTok{psn\_graph }\OtherTok{\textless{}{-}} \FunctionTok{graph\_from\_adjacency\_matrix}\NormalTok{(}
\NormalTok{  psn\_full,}
  \AttributeTok{mode =} \StringTok{"undirected"}\NormalTok{,}
  \AttributeTok{weighted =} \ConstantTok{TRUE}\NormalTok{,}
  \AttributeTok{diag =} \ConstantTok{FALSE}
\NormalTok{)}

\CommentTok{\# Louvain community detection}
\NormalTok{louvain\_psn }\OtherTok{\textless{}{-}} \FunctionTok{cluster\_louvain}\NormalTok{(psn\_graph)}

\CommentTok{\# Community membership}
\NormalTok{comm\_res }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{patient   =} \FunctionTok{names}\NormalTok{(}\FunctionTok{membership}\NormalTok{(louvain\_psn)),}
  \AttributeTok{community =} \FunctionTok{as.character}\NormalTok{(}\FunctionTok{membership}\NormalTok{(louvain\_psn)),}
  \AttributeTok{stringsAsFactors =} \ConstantTok{FALSE}
\NormalTok{)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Number of PSN communities:"}\NormalTok{,}
    \FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(comm\_res}\SpecialCharTok{$}\NormalTok{community)), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Number of PSN communities: 2
\end{verbatim}

We also apply a threshold to keep only the most significant edges

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# Thresholded PSN — visualization only}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{threshold\_plot }\OtherTok{\textless{}{-}} \FloatTok{0.7}

\NormalTok{psn\_plot }\OtherTok{\textless{}{-}}\NormalTok{ psn\_full}
\NormalTok{psn\_plot[psn\_plot }\SpecialCharTok{\textless{}}\NormalTok{ threshold\_plot] }\OtherTok{\textless{}{-}} \DecValTok{0}
\FunctionTok{diag}\NormalTok{(psn\_plot) }\OtherTok{\textless{}{-}} \DecValTok{0}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Edges shown in plot:"}\NormalTok{, }\FunctionTok{sum}\NormalTok{(psn\_plot }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{) }\SpecialCharTok{/} \DecValTok{2}\NormalTok{, }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Edges shown in plot: 112
\end{verbatim}

\paragraph{Create the network object}\label{create-the-network-object}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{net\_obj }\OtherTok{\textless{}{-}} \FunctionTok{network}\NormalTok{(}
\NormalTok{  psn\_plot,}
  \AttributeTok{matrix.type =} \StringTok{"adjacency"}\NormalTok{,}
  \AttributeTok{ignore.eval =} \ConstantTok{FALSE}\NormalTok{,   }\CommentTok{\# keep weights}
  \AttributeTok{names.eval  =} \StringTok{"weights"}\NormalTok{,}
  \AttributeTok{directed    =} \ConstantTok{FALSE}
\NormalTok{)}

\CommentTok{\# Attach community labels from FULL PSN clustering}
\NormalTok{net\_names }\OtherTok{\textless{}{-}} \FunctionTok{network.vertex.names}\NormalTok{(net\_obj)}
\NormalTok{net\_obj }\SpecialCharTok{\%v\%} \StringTok{"community"} \OtherTok{\textless{}{-}}
\NormalTok{  comm\_res}\SpecialCharTok{$}\NormalTok{community[}\FunctionTok{match}\NormalTok{(net\_names, comm\_res}\SpecialCharTok{$}\NormalTok{patient)]}
\end{Highlighting}
\end{Shaded}

\paragraph{Check of the graph}\label{check-of-the-graph}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"\# of nodes (patients):"}\NormalTok{, }\FunctionTok{vcount}\NormalTok{(psn\_graph), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # of nodes (patients): 32
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"\# of edges:"}\NormalTok{, }\FunctionTok{ecount}\NormalTok{(psn\_graph), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # of edges: 496
\end{verbatim}

Plot of the PSN

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Define a manual color palette}
\NormalTok{pal }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"1"} \OtherTok{=} \StringTok{"\#1f78b4"}\NormalTok{,   }\CommentTok{\# blue}
         \StringTok{"2"} \OtherTok{=} \StringTok{"\#e31a1c"}\NormalTok{)   }\CommentTok{\# red}

\FunctionTok{ggnet2}\NormalTok{(net\_obj,}
       \AttributeTok{color =} \StringTok{"community"}\NormalTok{,}
       \AttributeTok{palette =}\NormalTok{ pal,}
       \AttributeTok{alpha =} \FloatTok{0.9}\NormalTok{,}
       \AttributeTok{size =} \DecValTok{6}\NormalTok{,}
       \AttributeTok{label =} \ConstantTok{TRUE}\NormalTok{,}
       \AttributeTok{label.size =} \DecValTok{3}\NormalTok{,}
       \AttributeTok{edge.color =} \StringTok{"gray80"}\NormalTok{,}
       \AttributeTok{edge.alpha =} \FloatTok{0.7}\NormalTok{,}
       \AttributeTok{edge.size =} \FloatTok{0.3}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{ggtitle}\NormalTok{(}\StringTok{"Patient Similarity Network (Full PSN, thresholded for visualization)"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-92-1.pdf}}

\subsubsection{4(opt). PSN and Community detection using Normal Gene
Expression}\label{opt.-psn-and-community-detection-using-normal-gene-expression}

Data preparation

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Expression matrix: DEGs x patients (normal samples)}
\NormalTok{target\_data\_normal }\OtherTok{\textless{}{-}} \FunctionTok{as.matrix}\NormalTok{(expr\_deg\_normal)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Dimension NORMAL matrix (DEGs x Patients):}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Dimension NORMAL matrix (DEGs x Patients):
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(}\FunctionTok{dim}\NormalTok{(target\_data\_normal))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 724  32
\end{verbatim}

Similarity matrix calculation

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# FULL similarity matrix (patients x patients)}
\NormalTok{psn\_normal\_full }\OtherTok{\textless{}{-}} \FunctionTok{cor}\NormalTok{(target\_data\_normal, }\AttributeTok{method =} \StringTok{"pearson"}\NormalTok{)}
\CommentTok{\# Remove self{-}loops}
\FunctionTok{diag}\NormalTok{(psn\_normal\_full) }\OtherTok{\textless{}{-}} \DecValTok{0}

\CommentTok{\# Keep only positive similarities}
\NormalTok{psn\_normal\_full[psn\_normal\_full }\SpecialCharTok{\textless{}} \DecValTok{0}\NormalTok{] }\OtherTok{\textless{}{-}} \DecValTok{0}

\CommentTok{\# Sanity check}
\FunctionTok{stopifnot}\NormalTok{(}\FunctionTok{all}\NormalTok{(}\FunctionTok{is.finite}\NormalTok{(psn\_normal\_full)))}
\FunctionTok{dim}\NormalTok{(psn\_normal\_full)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 32 32
\end{verbatim}

As before, we also apply a threshold to keep only the most significant
edges

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Build graph from FULL similarity matrix}
\NormalTok{psn\_graph\_normal }\OtherTok{\textless{}{-}} \FunctionTok{graph\_from\_adjacency\_matrix}\NormalTok{(}
\NormalTok{  psn\_normal\_full,}
  \AttributeTok{mode =} \StringTok{"undirected"}\NormalTok{,}
  \AttributeTok{weighted =} \ConstantTok{TRUE}\NormalTok{,}
  \AttributeTok{diag =} \ConstantTok{FALSE}
\NormalTok{)}

\CommentTok{\# Louvain clustering}
\NormalTok{louvain\_normal }\OtherTok{\textless{}{-}} \FunctionTok{cluster\_louvain}\NormalTok{(psn\_graph\_normal)}

\CommentTok{\# Community membership}
\NormalTok{comm\_res\_normal }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{patient          =} \FunctionTok{names}\NormalTok{(}\FunctionTok{membership}\NormalTok{(louvain\_normal)),}
  \AttributeTok{community\_normal =} \FunctionTok{as.character}\NormalTok{(}\FunctionTok{membership}\NormalTok{(louvain\_normal)),}
  \AttributeTok{stringsAsFactors =} \ConstantTok{FALSE}
\NormalTok{)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Number of communities (NORMAL):"}\NormalTok{,}
    \FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(comm\_res\_normal}\SpecialCharTok{$}\NormalTok{community\_normal)), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Number of communities (NORMAL): 2
\end{verbatim}

Create the network object

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# THRESHOLDED NETWORK — VISUALIZATION ONLY}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{plot\_thr }\OtherTok{\textless{}{-}} \FloatTok{0.85}

\NormalTok{A\_plot }\OtherTok{\textless{}{-}}\NormalTok{ psn\_normal\_full}
\NormalTok{A\_plot[A\_plot }\SpecialCharTok{\textless{}}\NormalTok{ plot\_thr] }\OtherTok{\textless{}{-}} \DecValTok{0}
\FunctionTok{diag}\NormalTok{(A\_plot) }\OtherTok{\textless{}{-}} \DecValTok{0}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Edges shown in plot:"}\NormalTok{, }\FunctionTok{sum}\NormalTok{(A\_plot }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{) }\SpecialCharTok{/} \DecValTok{2}\NormalTok{, }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Edges shown in plot: 310
\end{verbatim}

Community detection with Louvain

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{net\_normal }\OtherTok{\textless{}{-}} \FunctionTok{network}\NormalTok{(}
\NormalTok{  A\_plot,}
  \AttributeTok{matrix.type =} \StringTok{"adjacency"}\NormalTok{,}
  \AttributeTok{ignore.eval =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{names.eval =} \StringTok{"weights"}\NormalTok{,}
  \AttributeTok{directed =} \ConstantTok{FALSE}
\NormalTok{)}

\CommentTok{\# Attach FULL community labels}
\NormalTok{net\_names\_norm }\OtherTok{\textless{}{-}} \FunctionTok{network.vertex.names}\NormalTok{(net\_normal)}
\NormalTok{net\_normal }\SpecialCharTok{\%v\%} \StringTok{"community"} \OtherTok{\textless{}{-}}
\NormalTok{  comm\_res\_normal}\SpecialCharTok{$}\NormalTok{community\_normal[}
    \FunctionTok{match}\NormalTok{(net\_names\_norm, comm\_res\_normal}\SpecialCharTok{$}\NormalTok{patient)}
\NormalTok{  ]}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{set.seed}\NormalTok{(}\DecValTok{77}\NormalTok{)}
\NormalTok{n\_comm\_norm }\OtherTok{\textless{}{-}} \FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(net\_normal }\SpecialCharTok{\%v\%} \StringTok{"community"}\NormalTok{))}

\NormalTok{palette\_norm }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}
  \StringTok{"1"} \OtherTok{=} \StringTok{"\#1f78b4"}\NormalTok{,}
  \StringTok{"2"} \OtherTok{=} \StringTok{"\#33a02c"}\NormalTok{,}
  \StringTok{"3"} \OtherTok{=} \StringTok{"\#e31a1c"}
\NormalTok{)[}\FunctionTok{seq\_len}\NormalTok{(n\_comm\_norm)]}

\FunctionTok{ggnet2}\NormalTok{(net\_normal,}
       \AttributeTok{color =} \StringTok{"community"}\NormalTok{,}
       \AttributeTok{palette =}\NormalTok{ palette\_norm,}
       \AttributeTok{alpha =} \FloatTok{0.9}\NormalTok{,}
       \AttributeTok{size =} \DecValTok{6}\NormalTok{,}
       \AttributeTok{label =} \ConstantTok{TRUE}\NormalTok{,}
       \AttributeTok{label.size =} \DecValTok{3}\NormalTok{,}
       \AttributeTok{edge.color =} \StringTok{"gray85"}\NormalTok{,}
       \AttributeTok{edge.alpha =} \FloatTok{0.8}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{ggtitle}\NormalTok{(}\StringTok{"Patient Similarity Network — NORMAL (Full PSN, thresholded for visualization)"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-98-1.pdf}}

Comparison between Tumor and Normal PSNs

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{comparison\_psn }\OtherTok{\textless{}{-}} \FunctionTok{merge}\NormalTok{(comm\_res, comm\_res\_normal, }\AttributeTok{by =} \StringTok{"patient"}\NormalTok{)}
\FunctionTok{colnames}\NormalTok{(comparison\_psn) }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"Patient"}\NormalTok{, }\StringTok{"Cancer\_Community"}\NormalTok{, }\StringTok{"Normal\_Community"}\NormalTok{)}
\NormalTok{conf\_mat\_norm }\OtherTok{\textless{}{-}} \FunctionTok{table}\NormalTok{(}
  \AttributeTok{Cancer =}\NormalTok{ comparison\_psn}\SpecialCharTok{$}\NormalTok{Cancer\_Community,}
  \AttributeTok{Normal =}\NormalTok{ comparison\_psn}\SpecialCharTok{$}\NormalTok{Normal\_Community}
\NormalTok{)}
\FunctionTok{print}\NormalTok{(}\StringTok{"Confusion Matrix (Cancer vs Normal PSN):"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "Confusion Matrix (Cancer vs Normal PSN):"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(conf\_mat\_norm)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##       Normal
## Cancer  1  2
##      1 13  6
##      2  9  4
\end{verbatim}

Heatmap of the confusion matrix

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{melt\_norm }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(}\FunctionTok{as.table}\NormalTok{(conf\_mat\_norm))}
\FunctionTok{colnames}\NormalTok{(melt\_norm) }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"Cancer\_Group"}\NormalTok{, }\StringTok{"Normal\_Group"}\NormalTok{, }\StringTok{"Count"}\NormalTok{)}

\FunctionTok{ggplot}\NormalTok{(melt\_norm, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ Normal\_Group, }\AttributeTok{y =}\NormalTok{ Cancer\_Group, }\AttributeTok{fill =}\NormalTok{ Count)) }\SpecialCharTok{+}
  \FunctionTok{geom\_tile}\NormalTok{(}\AttributeTok{color =} \StringTok{"white"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_text}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{label =}\NormalTok{ Count), }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{size =} \DecValTok{15}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_gradient}\NormalTok{(}\AttributeTok{low =} \StringTok{"white"}\NormalTok{, }\AttributeTok{high =} \StringTok{"steelblue"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}\AttributeTok{title =} \StringTok{"Comparison: PSN Cancer vs PSN Normal"}\NormalTok{,}
       \AttributeTok{x =} \StringTok{"Normal communities"}\NormalTok{,}
       \AttributeTok{y =} \StringTok{"Cancer communities"}\NormalTok{)   }\SpecialCharTok{+}
  \FunctionTok{guides}\NormalTok{(}\AttributeTok{fill =} \FunctionTok{guide\_colourbar}\NormalTok{(}
  \AttributeTok{barheight =} \FunctionTok{unit}\NormalTok{(}\DecValTok{3}\NormalTok{, }\StringTok{"in"}\NormalTok{),}
  \AttributeTok{label.theme =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{30}\NormalTok{),  }
  \AttributeTok{title.theme =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{0}\NormalTok{)    }
\NormalTok{)) }\SpecialCharTok{+}
\NormalTok{  base\_theme}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-100-1.pdf}}

\subsubsection{c.~Similarity Network
Fusion}\label{c.-similarity-network-fusion}

\paragraph{Download \& Processing of Mutation
Data}\label{download-processing-of-mutation-data}

We download the Simple Nucleotide Variation data for TCGA-KIRP and
convert it into a binary mutation matrix.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{query\_mut }\OtherTok{\textless{}{-}}\NormalTok{ TCGAbiolinks}\SpecialCharTok{::}\FunctionTok{GDCquery}\NormalTok{(}
  \AttributeTok{project =}\NormalTok{ proj, }
  \AttributeTok{data.category =} \StringTok{"Simple Nucleotide Variation"}\NormalTok{, }
  \AttributeTok{data.type =} \StringTok{"Masked Somatic Mutation"}\NormalTok{,}
  \AttributeTok{workflow.type =} \StringTok{"Aliquot Ensemble Somatic Variant Merging and Masking"} \CommentTok{\# current standard }
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Actual download \textbf{run only once}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{GDCdownload}\NormalTok{(}\AttributeTok{query =}\NormalTok{ query\_mut, }\AttributeTok{directory =} \StringTok{"GDCdata"}\NormalTok{, }\AttributeTok{method =} \StringTok{"api"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Convert the downloaded files into a SummarizedExperiment object

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{mut\_data\_prepared }\OtherTok{\textless{}{-}} \FunctionTok{GDCprepare}\NormalTok{(query\_mut)}
\end{Highlighting}
\end{Shaded}

Reading with maftools

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{maf\_obj }\OtherTok{\textless{}{-}} \FunctionTok{read.maf}\NormalTok{(}\AttributeTok{maf =}\NormalTok{ mut\_data\_prepared)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## -Validating
## -Silent variants: 4635 
## -Summarizing
## --Mutiple centers found
## BI;BCM--Possible FLAGS among top ten genes:
##   TTN
##   MUC16
## -Processing clinical data
## --Missing clinical data
## -Finished in 0.598s elapsed (0.618s cpu)
\end{verbatim}

Create the binary matrix (Patients x Genes)

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{mut\_count\_matrix }\OtherTok{\textless{}{-}} \FunctionTok{mutCountMatrix}\NormalTok{(maf\_obj) }\CommentTok{\# matrix with the count of the mutations}
\end{Highlighting}
\end{Shaded}

Transpose the matrix in order to have patients on the rows and genes on
the columns

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{mut\_matrix\_binary }\OtherTok{\textless{}{-}} \FunctionTok{t}\NormalTok{(mut\_count\_matrix)}
\end{Highlighting}
\end{Shaded}

Convert in binary (1 if mutated, 0 otherwise)

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{mut\_matrix\_binary[mut\_matrix\_binary }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{] }\OtherTok{\textless{}{-}} \DecValTok{1}
\end{Highlighting}
\end{Shaded}

Use only the 12-character patient ID as row names

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{rownames}\NormalTok{(mut\_matrix\_binary) }\OtherTok{\textless{}{-}} \FunctionTok{substr}\NormalTok{(}\FunctionTok{rownames}\NormalTok{(mut\_matrix\_binary), }\DecValTok{1}\NormalTok{, }\DecValTok{12}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Remove possible duplicates

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{mut\_matrix\_binary }\OtherTok{\textless{}{-}}\NormalTok{ mut\_matrix\_binary[}\SpecialCharTok{!}\FunctionTok{duplicated}\NormalTok{(}\FunctionTok{rownames}\NormalTok{(mut\_matrix\_binary)), ]}
\end{Highlighting}
\end{Shaded}

\paragraph{Patient Alignment (Expression vs
Mutation)}\label{patient-alignment-expression-vs-mutation}

We need to ensure both matrices analyze exactly the same set of
patients.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{patients\_rna }\OtherTok{\textless{}{-}} \FunctionTok{substr}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(target\_data), }\DecValTok{1}\NormalTok{, }\DecValTok{12}\NormalTok{) }\CommentTok{\# patients used in the 5a section (expression)}
\NormalTok{patients\_mut }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(mut\_matrix\_binary) }\CommentTok{\# patients available in mutations}
\end{Highlighting}
\end{Shaded}

Find the intersection (common patients)

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{common\_patients }\OtherTok{\textless{}{-}} \FunctionTok{intersect}\NormalTok{(patients\_rna, patients\_mut)}
\end{Highlighting}
\end{Shaded}

Quick check

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"Esempio RNA:"}\NormalTok{, }\FunctionTok{head}\NormalTok{(patients\_rna, }\DecValTok{3}\NormalTok{), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Esempio RNA: TCGA-BQ-5875 TCGA-B9-4115 TCGA-BQ-5879
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"Esempio MUT:"}\NormalTok{, }\FunctionTok{head}\NormalTok{(patients\_mut, }\DecValTok{3}\NormalTok{), }\StringTok{"}\SpecialCharTok{\textbackslash{}n\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Esempio MUT: TCGA-V9-A7HT TCGA-BQ-5878 TCGA-PJ-A8JU
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"Patients with RNA data:"}\NormalTok{, }\FunctionTok{length}\NormalTok{(patients\_rna), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Patients with RNA data: 32
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"Patients with Mutation data:"}\NormalTok{, }\FunctionTok{length}\NormalTok{(patients\_mut), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Patients with Mutation data: 278
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"Common patients for SNF:"}\NormalTok{, }\FunctionTok{length}\NormalTok{(common\_patients), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Common patients for SNF: 32
\end{verbatim}

\textbf{That means that each patient of which we have the genic
expression, has also a mutation}

Matrix reorder (both matrices must have columns in the same order):

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Expression Matrix (must be Patients x Genes but in 5a the dim was Genes x Patients}
\CommentTok{\# and SNF needs the patients on the rows)}
\NormalTok{data\_expression }\OtherTok{\textless{}{-}} \FunctionTok{t}\NormalTok{(target\_data[, common\_patients])}

\CommentTok{\# Mutation Matrix (Patients x Genes)}
\NormalTok{data\_mutation }\OtherTok{\textless{}{-}}\NormalTok{ mut\_matrix\_binary[common\_patients, ]}
\end{Highlighting}
\end{Shaded}

Final check

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{all}\NormalTok{(}\FunctionTok{rownames}\NormalTok{(data\_expression) }\SpecialCharTok{==} \FunctionTok{rownames}\NormalTok{(data\_mutation)) }\CommentTok{\# Must be TRUE}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] TRUE
\end{verbatim}

\paragraph{Calculation of Affinity
Matrices}\label{calculation-of-affinity-matrices}

Standard parameters for SNF

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{K }\OtherTok{\textless{}{-}} \DecValTok{20}      \CommentTok{\# number of neighbors}
\NormalTok{alpha }\OtherTok{\textless{}{-}} \FloatTok{0.5} \CommentTok{\# hyperparameter for the construction of the affinity kernel}
\NormalTok{T\_iter }\OtherTok{\textless{}{-}} \DecValTok{20} \CommentTok{\# number of iteration needed for the fusion}
\end{Highlighting}
\end{Shaded}

First Layer: Expression

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{dist\_exp }\OtherTok{\textless{}{-}} \FunctionTok{dist2}\NormalTok{(}\FunctionTok{as.matrix}\NormalTok{(data\_expression), }\FunctionTok{as.matrix}\NormalTok{(data\_expression)) }\CommentTok{\# Euclidean distance }
\NormalTok{W\_exp }\OtherTok{\textless{}{-}} \FunctionTok{affinityMatrix}\NormalTok{(dist\_exp, K, alpha) }\CommentTok{\# Affinity Matrix (W1)}
\end{Highlighting}
\end{Shaded}

Second Layer: Mutation

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{dist\_mut }\OtherTok{\textless{}{-}} \FunctionTok{dist2}\NormalTok{(}\FunctionTok{as.matrix}\NormalTok{(data\_mutation), }\FunctionTok{as.matrix}\NormalTok{(data\_mutation)) }\CommentTok{\# Euclidean distance}
\NormalTok{W\_mut }\OtherTok{\textless{}{-}} \FunctionTok{affinityMatrix}\NormalTok{(dist\_mut, K, alpha) }\CommentTok{\# Affinity Matrix (W2)}
\end{Highlighting}
\end{Shaded}

Fusion (Similarity Network Fusion)

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{W\_fused }\OtherTok{\textless{}{-}} \FunctionTok{SNF}\NormalTok{(}\FunctionTok{list}\NormalTok{(W\_exp, W\_mut), K, T\_iter)}
\end{Highlighting}
\end{Shaded}

Assign the patient's names to the fused matrix

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{rownames}\NormalTok{(W\_fused) }\OtherTok{\textless{}{-}}\NormalTok{ common\_patients}
\FunctionTok{colnames}\NormalTok{(W\_fused) }\OtherTok{\textless{}{-}}\NormalTok{ common\_patients}
\end{Highlighting}
\end{Shaded}

Same preprocessing as before

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# We\textquotesingle{}re only interested in positive correlations}
\NormalTok{W\_fused[W\_fused}\SpecialCharTok{\textless{}}\DecValTok{0}\NormalTok{] }\OtherTok{\textless{}{-}} \DecValTok{0}

\CommentTok{\# We also apply a threshold to keep only the most significant edges}
\NormalTok{threshold\_snf }\OtherTok{\textless{}{-}} \DecValTok{0}
\NormalTok{W\_fused[W\_fused }\SpecialCharTok{\textless{}}\NormalTok{ threshold\_snf] }\OtherTok{\textless{}{-}} \DecValTok{0}

\FunctionTok{diag}\NormalTok{(W\_fused) }\OtherTok{\textless{}{-}} \DecValTok{0}
\end{Highlighting}
\end{Shaded}

\paragraph{Community Detection on the Fused
Network}\label{community-detection-on-the-fused-network}

We apply Louvain clustering on the fused network to identify multi-omics
communities.

Create the graph of the fused network:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{graph\_snf }\OtherTok{\textless{}{-}} \FunctionTok{graph\_from\_adjacency\_matrix}\NormalTok{(W\_fused, }\AttributeTok{mode =} \StringTok{"undirected"}\NormalTok{, }\AttributeTok{weighted =} \ConstantTok{TRUE}\NormalTok{, }\AttributeTok{diag =} \ConstantTok{FALSE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Apply Louvain

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{louvain\_snf }\OtherTok{\textless{}{-}} \FunctionTok{cluster\_louvain}\NormalTok{(graph\_snf)}
\end{Highlighting}
\end{Shaded}

Create the dataframe of the SNF results

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{comm\_res\_snf }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{patient =} \FunctionTok{names}\NormalTok{(}\FunctionTok{membership}\NormalTok{(louvain\_snf)),}
  \AttributeTok{community\_snf =} \FunctionTok{as.character}\NormalTok{(}\FunctionTok{membership}\NormalTok{(louvain\_snf)),}
  \AttributeTok{stringsAsFactors =} \ConstantTok{FALSE}
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Check

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"Number of communities found with SNF:"}\NormalTok{, }\FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(comm\_res\_snf}\SpecialCharTok{$}\NormalTok{community\_snf)), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Number of communities found with SNF: 3
\end{verbatim}

\paragraph{Plot of the Fused Network}\label{plot-of-the-fused-network}

Thresholding the matrix to improve readability

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#l\_comp\_snf \textless{}{-} W\_fused}
\CommentTok{\#threshold\_snf \textless{}{-} 0.02}
\CommentTok{\#l\_comp\_snf[l\_comp\_snf \textless{} threshold\_snf] \textless{}{-} 0}
\CommentTok{\#diag(l\_comp\_snf) \textless{}{-} 0}
\end{Highlighting}
\end{Shaded}

Network object

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# THRESHOLDED SNF NETWORK (FOR VISUALIZATION ONLY)}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{plot\_thr\_snf }\OtherTok{\textless{}{-}} \FloatTok{0.02}   \CommentTok{\# tipico per SNF (0.01–0.05)}

\NormalTok{A\_snf\_plot }\OtherTok{\textless{}{-}}\NormalTok{ W\_fused}
\NormalTok{A\_snf\_plot[A\_snf\_plot }\SpecialCharTok{\textless{}}\NormalTok{ plot\_thr\_snf] }\OtherTok{\textless{}{-}} \DecValTok{0}
\FunctionTok{diag}\NormalTok{(A\_snf\_plot) }\OtherTok{\textless{}{-}} \DecValTok{0}

\FunctionTok{cat}\NormalTok{(}\StringTok{"SNF plot edges:"}\NormalTok{, }\FunctionTok{sum}\NormalTok{(A\_snf\_plot }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{) }\SpecialCharTok{/} \DecValTok{2}\NormalTok{, }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## SNF plot edges: 67
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{net\_snf }\OtherTok{\textless{}{-}} \FunctionTok{network}\NormalTok{(}
\NormalTok{  A\_snf\_plot,}
  \AttributeTok{matrix.type =} \StringTok{"adjacency"}\NormalTok{,}
  \AttributeTok{ignore.eval =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{names.eval =} \StringTok{"weights"}\NormalTok{,}
  \AttributeTok{directed =} \ConstantTok{FALSE}
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Assign attributes

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{net\_names\_snf }\OtherTok{\textless{}{-}} \FunctionTok{network.vertex.names}\NormalTok{(net\_snf)}
\NormalTok{net\_snf }\SpecialCharTok{\%v\%} \StringTok{"community"} \OtherTok{\textless{}{-}}\NormalTok{ comm\_res\_snf}\SpecialCharTok{$}\NormalTok{community\_snf[}\FunctionTok{match}\NormalTok{(net\_names\_snf, comm\_res\_snf}\SpecialCharTok{$}\NormalTok{patient)]}

\NormalTok{n\_comm\_snf }\OtherTok{\textless{}{-}} \FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(net\_snf }\SpecialCharTok{\%v\%} \StringTok{"community"}\NormalTok{))}
\FunctionTok{set.seed}\NormalTok{(}\DecValTok{123}\NormalTok{)}
\NormalTok{pal\_snf }\OtherTok{\textless{}{-}} \FunctionTok{sample}\NormalTok{(}\FunctionTok{colors}\NormalTok{(}\AttributeTok{distinct =} \ConstantTok{TRUE}\NormalTok{), n\_comm\_snf)}
\FunctionTok{names}\NormalTok{(pal\_snf) }\OtherTok{\textless{}{-}} \FunctionTok{sort}\NormalTok{(}\FunctionTok{unique}\NormalTok{(net\_snf }\SpecialCharTok{\%v\%} \StringTok{"community"}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

Plot

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggnet2}\NormalTok{(net\_snf, }
       \AttributeTok{color =} \StringTok{"community"}\NormalTok{, }
       \AttributeTok{palette =}\NormalTok{ pal\_snf, }
       \AttributeTok{size =} \DecValTok{6}\NormalTok{, }
       \AttributeTok{label =} \ConstantTok{TRUE}\NormalTok{, }
       \AttributeTok{label.size =} \DecValTok{3}\NormalTok{,}
       \AttributeTok{edge.color =} \StringTok{"gray80"}\NormalTok{, }
       \AttributeTok{edge.alpha =} \FloatTok{0.7}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{ggtitle}\NormalTok{(}\StringTok{"SNF Community Structure (Expression + Mutation, thresholded)"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-128-1.pdf}}

\paragraph{Final comparison}\label{final-comparison}

Merge the results into a single dataframe

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{comparison\_df }\OtherTok{\textless{}{-}} \FunctionTok{merge}\NormalTok{(comm\_res, comm\_res\_snf, }\AttributeTok{by =} \StringTok{"patient"}\NormalTok{)}
\FunctionTok{colnames}\NormalTok{(comparison\_df) }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"Patient"}\NormalTok{, }\StringTok{"Cluster\_RNA\_Only"}\NormalTok{, }\StringTok{"Cluster\_SNF"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Create a Confusion Matrix to show how patients move among groups when
adding mutation data

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{conf\_matrix }\OtherTok{\textless{}{-}} \FunctionTok{table}\NormalTok{(}\AttributeTok{RNA\_Only =}\NormalTok{ comparison\_df}\SpecialCharTok{$}\NormalTok{Cluster\_RNA\_Only, }
                     \AttributeTok{SNF =}\NormalTok{ comparison\_df}\SpecialCharTok{$}\NormalTok{Cluster\_SNF)}

\FunctionTok{print}\NormalTok{(}\StringTok{"Confusion Matrix (Rows: RNA Clusters, Cols: SNF Clusters):"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "Confusion Matrix (Rows: RNA Clusters, Cols: SNF Clusters):"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(conf\_matrix)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##         SNF
## RNA_Only  1  2  3
##        1 11  8  0
##        2  0  1 12
\end{verbatim}

Transform the table in a dataframe for the plot

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{melted\_cmat }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(}\FunctionTok{as.table}\NormalTok{(conf\_matrix))}
\FunctionTok{colnames}\NormalTok{(melted\_cmat) }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"RNA\_Group"}\NormalTok{, }\StringTok{"SNF\_Group"}\NormalTok{, }\StringTok{"Count"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Plot

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggplot}\NormalTok{(melted\_cmat, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x =}\NormalTok{ SNF\_Group, }\AttributeTok{y =}\NormalTok{ RNA\_Group, }\AttributeTok{fill =}\NormalTok{ Count)) }\SpecialCharTok{+}
  \FunctionTok{geom\_tile}\NormalTok{(}\AttributeTok{color =} \StringTok{"white"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{geom\_text}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{label =}\NormalTok{ Count), }\AttributeTok{color =} \StringTok{"black"}\NormalTok{, }\AttributeTok{size =} \DecValTok{10}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_gradient}\NormalTok{(}\AttributeTok{low =} \StringTok{"white"}\NormalTok{, }\AttributeTok{high =} \StringTok{"tomato"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}
    \AttributeTok{title =} \StringTok{"Comparison: RNA{-}Seq vs SNF Clusters"}\NormalTok{,}
    \AttributeTok{x =} \StringTok{"SNF Communities (RNA + Mutation)"}\NormalTok{,}
    \AttributeTok{y =} \StringTok{"RNA{-}Only Communities (5.1)"}
\NormalTok{  ) }\SpecialCharTok{+}
  \FunctionTok{guides}\NormalTok{(}\AttributeTok{fill =} \FunctionTok{guide\_colourbar}\NormalTok{(}
  \AttributeTok{barheight =} \FunctionTok{unit}\NormalTok{(}\DecValTok{3}\NormalTok{, }\StringTok{"in"}\NormalTok{),}
  \AttributeTok{label.theme =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{30}\NormalTok{),   }
  \AttributeTok{title.theme =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{0}\NormalTok{)   }
\NormalTok{)) }\SpecialCharTok{+}
\NormalTok{  base\_theme}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-132-1.pdf}}

\subsection{3(opt). Gene Set Enrichment Analysis (cancer
hubs)}\label{opt.-gene-set-enrichment-analysis-cancer-hubs}

\subsubsection{Load required libraries}\label{load-required-libraries}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(clusterProfiler)}
\FunctionTok{library}\NormalTok{(org.Hs.eg.db)}
\FunctionTok{library}\NormalTok{(enrichplot) }\CommentTok{\# Useful for advanced visualizations}
\FunctionTok{library}\NormalTok{(pathview)   }\CommentTok{\# For visualizing colorful pathways}
\FunctionTok{library}\NormalTok{(stringr)}
\end{Highlighting}
\end{Shaded}

\subsubsection{Prepare Data for Gene Ontology
(GO)}\label{prepare-data-for-gene-ontology-go}

We extract the gene symbols from the hubs identified in \textbf{Task 3}.
We will compare ``Cancer Hubs'' (\texttt{hubs\_c}) against ``Normal
Hubs'' (\texttt{hubs\_n}) to see the functional differences.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{genes\_cancer }\OtherTok{\textless{}{-}} \FunctionTok{names}\NormalTok{(hubs\_c)}
\NormalTok{genes\_normal }\OtherTok{\textless{}{-}} \FunctionTok{names}\NormalTok{(hubs\_n)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(genes\_cancer)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "ENSG00000167613.16" "ENSG00000142347.19" "ENSG00000162511.8" 
## [4] "ENSG00000066336.12" "ENSG00000088325.16" "ENSG00000112799.9"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(genes\_normal)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "ENSG00000131183.11" "ENSG00000158296.14" "ENSG00000196549.12"
## [4] "ENSG00000136872.20" "ENSG00000009765.15" "ENSG00000180432.6"
\end{verbatim}

Remove the version suffix from the end of the IDs

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# gsub replaces the "\textbackslash{}\textbackslash{}..*" patter (the dot and all the rest) with ""}
\NormalTok{genes\_cancer }\OtherTok{\textless{}{-}} \FunctionTok{gsub}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{..*"}\NormalTok{, }\StringTok{""}\NormalTok{, genes\_cancer)}
\NormalTok{genes\_normal }\OtherTok{\textless{}{-}} \FunctionTok{gsub}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{..*"}\NormalTok{, }\StringTok{""}\NormalTok{, genes\_normal)}
\end{Highlighting}
\end{Shaded}

Let's see the type of the ID of the genes

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Check of the the first gene to know its format}
\NormalTok{first\_gene }\OtherTok{\textless{}{-}}\NormalTok{ genes\_cancer[}\DecValTok{1}\NormalTok{]}
\NormalTok{input\_type }\OtherTok{\textless{}{-}} \FunctionTok{ifelse}\NormalTok{(}\FunctionTok{startsWith}\NormalTok{(first\_gene, }\StringTok{"ENSG"}\NormalTok{), }\StringTok{"ENSEMBL"}\NormalTok{, }\StringTok{"SYMBOL"}\NormalTok{)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Clean ID:"}\NormalTok{, genes\_cancer[}\DecValTok{1}\NormalTok{], }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Clean ID: ENSG00000167613
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"The genes appear to be in the format:"}\NormalTok{, input\_type, }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## The genes appear to be in the format: ENSEMBL
\end{verbatim}

Let's convert them in the \textbf{ENTREZ ID} format (better supported
also for the KEGG later..)

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Cancer Genes}
\NormalTok{entrez\_cancer }\OtherTok{\textless{}{-}} \FunctionTok{bitr}\NormalTok{(genes\_cancer, }
                      \AttributeTok{fromType =}\NormalTok{ input\_type, }
                      \AttributeTok{toType =} \StringTok{"ENTREZID"}\NormalTok{, }
                      \AttributeTok{OrgDb =} \StringTok{"org.Hs.eg.db"}\NormalTok{)}\SpecialCharTok{$}\NormalTok{ENTREZID}

\CommentTok{\# Normal Genes}
\NormalTok{entrez\_normal }\OtherTok{\textless{}{-}} \FunctionTok{bitr}\NormalTok{(genes\_normal, }
                      \AttributeTok{fromType =}\NormalTok{ input\_type, }
                      \AttributeTok{toType =} \StringTok{"ENTREZID"}\NormalTok{, }
                      \AttributeTok{OrgDb =} \StringTok{"org.Hs.eg.db"}\NormalTok{)}\SpecialCharTok{$}\NormalTok{ENTREZID}
\end{Highlighting}
\end{Shaded}

Create a list for the compareCluster function

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{hubs\_list\_entrez }\OtherTok{\textless{}{-}} \FunctionTok{list}\NormalTok{(}\AttributeTok{Cancer =}\NormalTok{ entrez\_cancer, }\AttributeTok{Normal =}\NormalTok{ entrez\_normal)}
\end{Highlighting}
\end{Shaded}

Quick check

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(hubs\_list\_entrez}\SpecialCharTok{$}\NormalTok{Cancer)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "3903"  "4542"  "7805"  "6688"  "22974" "9450"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{head}\NormalTok{(hubs\_list\_entrez}\SpecialCharTok{$}\NormalTok{Normal)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "6569"   "64849"  "4311"   "229"    "389434" "1582"
\end{verbatim}

\subsubsection{Gene Ontology Enrichment (Biological
Process)}\label{gene-ontology-enrichment-biological-process}

We use \texttt{compareCluster} to perform GO enrichment for Biological
Processes (\texttt{ont="BP"}). This allows us to visualize parallel
functionalities between cancer and normal tissue hubs.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# GO Analysis}
\NormalTok{comp\_GO\_BP }\OtherTok{\textless{}{-}} \FunctionTok{compareCluster}\NormalTok{(}\AttributeTok{geneCluster =}\NormalTok{ hubs\_list\_entrez, }
                             \AttributeTok{fun =} \StringTok{"enrichGO"}\NormalTok{,}
                             \AttributeTok{OrgDb =} \StringTok{"org.Hs.eg.db"}\NormalTok{,}
                             \AttributeTok{keyType =} \StringTok{"ENTREZID"}\NormalTok{,}
                             \AttributeTok{ont =} \StringTok{"BP"}\NormalTok{, }
                             \AttributeTok{pvalueCutoff =} \FloatTok{0.05}\NormalTok{, }
                             \AttributeTok{pAdjustMethod =} \StringTok{"BH"}\NormalTok{,}
                             \AttributeTok{readable =} \ConstantTok{TRUE}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\paragraph{Plot}\label{plot}

Visualize results with a DotPlot to see which biological processes the
genes belong to

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Simplify to reduce redundancy}
\NormalTok{comp\_clean }\OtherTok{\textless{}{-}} \FunctionTok{simplify}\NormalTok{(comp\_GO\_BP, }\AttributeTok{cutoff =} \FloatTok{0.6}\NormalTok{, }\AttributeTok{by =} \StringTok{"p.adjust"}\NormalTok{, }\AttributeTok{select\_fun =}\NormalTok{ min)}

\CommentTok{\# Remove duplicate descriptions}
\NormalTok{comp\_clean}\SpecialCharTok{@}\NormalTok{compareClusterResult }\OtherTok{\textless{}{-}}\NormalTok{ comp\_clean}\SpecialCharTok{@}\NormalTok{compareClusterResult[}\SpecialCharTok{!}\FunctionTok{duplicated}\NormalTok{(comp\_clean}\SpecialCharTok{@}\NormalTok{compareClusterResult}\SpecialCharTok{$}\NormalTok{Description), ]}

\CommentTok{\# Create readable plot}
\FunctionTok{dotplot}\NormalTok{(comp\_clean, }
      \AttributeTok{showCategory =} \DecValTok{8}\NormalTok{, }
      \AttributeTok{title =} \StringTok{"GO Enrichment: Biological Processes"}\NormalTok{) }\SpecialCharTok{+}

\FunctionTok{scale\_y\_discrete}\NormalTok{(}\AttributeTok{labels =} \ControlFlowTok{function}\NormalTok{(x) }\FunctionTok{str\_wrap}\NormalTok{(x, }\AttributeTok{width =} \DecValTok{70}\NormalTok{)) }\SpecialCharTok{+} \CommentTok{\# Wrap text at 70 characters}

\FunctionTok{theme}\NormalTok{(}
  \AttributeTok{plot.title =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{hjust =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{size =} \DecValTok{12}\NormalTok{, }\AttributeTok{face =} \StringTok{"bold"}\NormalTok{),}
  \AttributeTok{axis.text.y =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{9}\NormalTok{, }\AttributeTok{hjust =} \DecValTok{1}\NormalTok{),  }\CommentTok{\# Increase size slightly}
  \AttributeTok{axis.text.x =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{angle =} \DecValTok{45}\NormalTok{, }\AttributeTok{hjust =} \DecValTok{1}\NormalTok{, }\AttributeTok{size =} \DecValTok{10}\NormalTok{),}
  \AttributeTok{plot.margin =} \FunctionTok{margin}\NormalTok{(}\DecValTok{5}\NormalTok{, }\DecValTok{5}\NormalTok{, }\DecValTok{5}\NormalTok{, }\DecValTok{5}\NormalTok{, }\StringTok{"pt"}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-142-1.pdf}}

Right column, Normal tissue: healthy tissue hubs are genes that maintain
the physiological functionality of the kidney. The kidney serves to
filter the blood and manage nutrients/salts; healthy hub genes do
exactly that.

Left column, Cancer tissue: Instead, cancer hubs are predominantly
enriched in processes related to immune response and inflammation. This
suggests that the tumor microenvironment is characterized by a strong
infiltration of immune cells and a chronic inflammatory state, which is
a known feature of Papillary Renal Cell Carcinoma.

\subsubsection{KEGG Pathway Enrichment}\label{kegg-pathway-enrichment}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{comp\_KEGG }\OtherTok{\textless{}{-}} \FunctionTok{compareCluster}\NormalTok{(}\AttributeTok{geneCluster =}\NormalTok{ hubs\_list\_entrez, }
                              \AttributeTok{fun =} \StringTok{"enrichKEGG"}\NormalTok{,}
                              \AttributeTok{organism =} \StringTok{"hsa"}\NormalTok{,   }\CommentTok{\# homo sapiens}
                              \AttributeTok{pvalueCutoff =} \FloatTok{0.05}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\paragraph{Plot}\label{plot-1}

Plot to see which metabolic pathways the genes belong to

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{dotplot}\NormalTok{(comp\_KEGG, }
            \AttributeTok{showCategory =} \DecValTok{10}\NormalTok{, }
            \AttributeTok{title =} \StringTok{"KEGG Pathway Enrichment (Hubs)"}\NormalTok{) }\SpecialCharTok{+}
      
      \CommentTok{\# Manda a capo i nomi dei pathway lunghi}
      \FunctionTok{scale\_y\_discrete}\NormalTok{(}\AttributeTok{labels =} \ControlFlowTok{function}\NormalTok{(x) }\FunctionTok{str\_wrap}\NormalTok{(x, }\AttributeTok{width =} \DecValTok{50}\NormalTok{)) }\SpecialCharTok{+}
      
      \FunctionTok{theme}\NormalTok{(}
        \AttributeTok{plot.title =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{hjust =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{size =} \DecValTok{12}\NormalTok{, }\AttributeTok{face =} \StringTok{"bold"}\NormalTok{),}
        \AttributeTok{axis.text.y =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{9}\NormalTok{), }
        \AttributeTok{axis.text.x =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{angle =} \DecValTok{45}\NormalTok{, }\AttributeTok{hjust =} \DecValTok{1}\NormalTok{, }\AttributeTok{size =} \DecValTok{10}\NormalTok{),}
        \AttributeTok{plot.margin =} \FunctionTok{margin}\NormalTok{(}\DecValTok{10}\NormalTok{, }\DecValTok{10}\NormalTok{, }\DecValTok{10}\NormalTok{, }\DecValTok{10}\NormalTok{)}
\NormalTok{      )}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-144-1.pdf}}

24/38 Cancer Genes and 19/27 Normal Genes were successfully mapped

Right column, Normal tissue: the related pathways confirm the metabolic
nature of the healthy kidney. The kidney not only serves to filter, but
actively reabsorbs proteins and sugars from the filtrate so as not to
lose them in the urine. Normal hubs maintain this metabolic homeostasis.

Left column, Cancer tissue: conversely, the cancer network reveals a
radical functional shift towards immune modulation and tissue
remodeling. Dominant pathways like \emph{Osteoclast differentiation} and
\emph{B cell receptor signaling} indicate a highly inflamed
microenvironment. Cancer hubs no longer support filtration but rather
drive structural changes and recruit immune cells to sustain tumor
progression.

\subsubsection{Pathway Visualization}\label{pathway-visualization}

\paragraph{Data Preparation}\label{data-preparation-2}

We recover the log2fc calculated in Task 2

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{gene\_data\_fc }\OtherTok{\textless{}{-}}\NormalTok{ log2fc}
\end{Highlighting}
\end{Shaded}

Remove the version suffix from the end of the IDs (same as before)

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{gene\_names\_clean }\OtherTok{\textless{}{-}} \FunctionTok{gsub}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{..*"}\NormalTok{, }\StringTok{""}\NormalTok{, }\FunctionTok{names}\NormalTok{(gene\_data\_fc))}
\end{Highlighting}
\end{Shaded}

Let's convert them in the \textbf{ENTREZ ID} format

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{mapping\_entrez }\OtherTok{\textless{}{-}} \FunctionTok{bitr}\NormalTok{(gene\_names\_clean, }
                       \AttributeTok{fromType =}\NormalTok{ input\_type, }\CommentTok{\# "SYMBOL" o "ENSEMBL" (computed before)}
                       \AttributeTok{toType =} \StringTok{"ENTREZID"}\NormalTok{, }
                       \AttributeTok{OrgDb =} \StringTok{"org.Hs.eg.db"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

Create a final vector: VALUES = Log2FC, NAMES= EntrezID

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# We need to align the data: we only keep the genes that have been successfully converted}
\NormalTok{idx\_mapped }\OtherTok{\textless{}{-}} \FunctionTok{match}\NormalTok{(gene\_names\_clean, mapping\_entrez[, input\_type])}
\NormalTok{gene\_data\_entrez }\OtherTok{\textless{}{-}}\NormalTok{ gene\_data\_fc[}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(idx\_mapped)]}
\FunctionTok{names}\NormalTok{(gene\_data\_entrez) }\OtherTok{\textless{}{-}}\NormalTok{ mapping\_entrez}\SpecialCharTok{$}\NormalTok{ENTREZID[}\FunctionTok{na.omit}\NormalTok{(idx\_mapped)]}
\end{Highlighting}
\end{Shaded}

\paragraph{Pathway selection}\label{pathway-selection}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# We automatically search for the most significant pathway for the "Cancer" cluster}
\CommentTok{\# if comp\_KEGG has results:}
\ControlFlowTok{if}\NormalTok{ (}\SpecialCharTok{!}\FunctionTok{is.null}\NormalTok{(comp\_KEGG) }\SpecialCharTok{\&\&} \FunctionTok{nrow}\NormalTok{(comp\_KEGG}\SpecialCharTok{@}\NormalTok{compareClusterResult) }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{) \{}
\NormalTok{  res\_df }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(comp\_KEGG)}
  
  \CommentTok{\# we filter for the Cancer cluster}
\NormalTok{  top\_cancer\_pathway }\OtherTok{\textless{}{-}} \FunctionTok{subset}\NormalTok{(res\_df, Cluster }\SpecialCharTok{==} \StringTok{"Cancer"}\NormalTok{)}
  
  \ControlFlowTok{if}\NormalTok{ (}\FunctionTok{nrow}\NormalTok{(top\_cancer\_pathway) }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{) \{}
    \CommentTok{\# take the ID of the first pathway (es. hsa04060)}
\NormalTok{    path\_id }\OtherTok{\textless{}{-}}\NormalTok{ top\_cancer\_pathway}\SpecialCharTok{$}\NormalTok{ID[}\DecValTok{1}\NormalTok{]}
\NormalTok{    path\_name }\OtherTok{\textless{}{-}}\NormalTok{ top\_cancer\_pathway}\SpecialCharTok{$}\NormalTok{Description[}\DecValTok{1}\NormalTok{]}
    
    \FunctionTok{cat}\NormalTok{(}\StringTok{"Map generation for the pathway:"}\NormalTok{, path\_name, }\StringTok{"("}\NormalTok{, path\_id, }\StringTok{")}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
    
    \CommentTok{\# PATHVIEW EXECUTION}
\NormalTok{    pv.out }\OtherTok{\textless{}{-}} \FunctionTok{pathview}\NormalTok{(}\AttributeTok{gene.data =}\NormalTok{ gene\_data\_entrez, }
                       \AttributeTok{pathway.id =}\NormalTok{ path\_id, }
                       \AttributeTok{species =} \StringTok{"hsa"}\NormalTok{, }
                       \AttributeTok{out.suffix =} \StringTok{"cancer\_hubs"}\NormalTok{, }\CommentTok{\# adds this name to the file}
                       \AttributeTok{keys.align =} \StringTok{"y"}\NormalTok{, }
                       \AttributeTok{keb.dir =} \StringTok{"."}\NormalTok{,      }\CommentTok{\# saves the image in the current directory}
                       \AttributeTok{limit =} \FunctionTok{list}\NormalTok{(}\AttributeTok{gene =} \DecValTok{2}\NormalTok{, }\AttributeTok{cpd =} \DecValTok{1}\NormalTok{), }\CommentTok{\# Color scale: from {-}2 (Blue) to +2 (Red)}
                       \AttributeTok{low =} \FunctionTok{list}\NormalTok{(}\AttributeTok{gene =} \StringTok{"blue"}\NormalTok{), }
                       \AttributeTok{mid =} \FunctionTok{list}\NormalTok{(}\AttributeTok{gene =} \StringTok{"gray"}\NormalTok{), }
                       \AttributeTok{high =} \FunctionTok{list}\NormalTok{(}\AttributeTok{gene =} \StringTok{"red"}\NormalTok{))}
    
    \FunctionTok{cat}\NormalTok{(}\StringTok{"Image saved as:"}\NormalTok{, }\FunctionTok{paste0}\NormalTok{(path\_id, }\StringTok{".cancer\_hubs.png"}\NormalTok{), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
    
\NormalTok{  \} }\ControlFlowTok{else}\NormalTok{ \{}
    \FunctionTok{cat}\NormalTok{(}\StringTok{"No cancer{-}specific pathways found in KEGG results.}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\NormalTok{  \}}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Map generation for the pathway: Complement and coagulation cascades ( hsa04610 )
\end{verbatim}

\begin{verbatim}
## Image saved as: hsa04610.cancer_hubs.png
\end{verbatim}

\paragraph{Plot}\label{plot-2}

Below there's an image because this library saves the plot as image
within the main directory

\pandocbounded{\includegraphics[keepaspectratio]{hsa04610.cancer_hubs.png}}

This pathway indicates a shift towards chronic inflammation and tissue
remodeling. Key components of the classical complement pathway,
including the initiator \textbf{C1q} and the central hub \textbf{C3},
are upregulated (red nodes). Crucially, the strong upregulation of
receptors such as \textbf{C5aR1}, \textbf{C3AR1}, and the integrin
receptors \textbf{CR3/CR4} suggests active signaling for immune cell
recruitment (chemotaxis) and adhesion, rather than direct cell lysis (as
the terminal MAC complex remains downregulated). Furthermore, the
upregulation of \textbf{uPAR} (PLAUR) and coagulation factors like
\textbf{F10} and \textbf{F12} highlights a mechanism for extracellular
matrix degradation and a pro-thrombotic state, facilitating tumor
spreading and metastasis.

\subsection{5(opt). PSN communities
characterization}\label{opt.-psn-communities-characterization}

To address PSN community characterization in a statistically meaningful
way, we adopted a tumor-only strategy, using all available primary tumor
samples and retaining one tumor sample per patient. This choice allows
us to substantially increase the number of patients included in the
analysis, overcoming the severe sample size limitation that arises when
restricting the cohort to paired tumor--normal samples only.

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\#  Tumor{-}only cohort: one Primary Tumor per patient}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\CommentTok{\# Raw tumor counts (genes x samples)}
\NormalTok{expr.C\_all }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(rna.expr.data.C)}

\CommentTok{\# Extract patient IDs (12 characters)}
\NormalTok{patients.C\_all }\OtherTok{\textless{}{-}} \FunctionTok{substr}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(expr.C\_all), }\DecValTok{1}\NormalTok{, }\DecValTok{12}\NormalTok{)}

\CommentTok{\# Count tumor samples per patient}
\NormalTok{tab.C\_all }\OtherTok{\textless{}{-}} \FunctionTok{table}\NormalTok{(patients.C\_all)}

\CommentTok{\# Keep only patients with exactly ONE tumor sample}
\NormalTok{single\_patients\_all }\OtherTok{\textless{}{-}} \FunctionTok{names}\NormalTok{(tab.C\_all[tab.C\_all }\SpecialCharTok{==} \DecValTok{1}\NormalTok{])}

\CommentTok{\# Subset expression matrix}
\NormalTok{idx.single.all }\OtherTok{\textless{}{-}} \FunctionTok{which}\NormalTok{(patients.C\_all }\SpecialCharTok{\%in\%}\NormalTok{ single\_patients\_all)}
\NormalTok{expr.C\_all }\OtherTok{\textless{}{-}}\NormalTok{ expr.C\_all[, idx.single.all, drop }\OtherTok{=} \ConstantTok{FALSE}\NormalTok{]}

\CommentTok{\# Rename columns to pure patient IDs}
\FunctionTok{colnames}\NormalTok{(expr.C\_all) }\OtherTok{\textless{}{-}} \FunctionTok{substr}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(expr.C\_all), }\DecValTok{1}\NormalTok{, }\DecValTok{12}\NormalTok{)}

\CommentTok{\# Final sanity checks}
\FunctionTok{stopifnot}\NormalTok{(}\FunctionTok{sum}\NormalTok{(}\FunctionTok{duplicated}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(expr.C\_all))) }\SpecialCharTok{==} \DecValTok{0}\NormalTok{)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Tumor{-}only cohort (genes x patients):"}\NormalTok{,}
    \FunctionTok{nrow}\NormalTok{(expr.C\_all), }\StringTok{"x"}\NormalTok{, }\FunctionTok{ncol}\NormalTok{(expr.C\_all), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Tumor-only cohort (genes x patients): 60660 x 290
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\#  Tumor{-}only normalization }

\CommentTok{\# Compute library sizes}
\NormalTok{lib\_size }\OtherTok{\textless{}{-}} \FunctionTok{colSums}\NormalTok{(expr.C\_all)}

\CommentTok{\# CPM normalization}
\NormalTok{tumor\_norm }\OtherTok{\textless{}{-}} \FunctionTok{sweep}\NormalTok{(expr.C\_all, }\DecValTok{2}\NormalTok{, lib\_size, }\AttributeTok{FUN =} \StringTok{"/"}\NormalTok{) }\SpecialCharTok{*} \FloatTok{1e6}

\CommentTok{\# Log{-}transform}
\NormalTok{tumor\_norm\_log }\OtherTok{\textless{}{-}} \FunctionTok{log2}\NormalTok{(tumor\_norm }\SpecialCharTok{+} \DecValTok{1}\NormalTok{)}


\FunctionTok{cat}\NormalTok{(}\StringTok{"Tumor{-}only normalized matrix:"}\NormalTok{,}
    \FunctionTok{nrow}\NormalTok{(tumor\_norm\_log), }\StringTok{"genes x"}\NormalTok{,}
    \FunctionTok{ncol}\NormalTok{(tumor\_norm\_log), }\StringTok{"patients}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Tumor-only normalized matrix: 60660 genes x 290 patients
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(survival)}
\FunctionTok{library}\NormalTok{(survminer)}
\FunctionTok{library}\NormalTok{(maftools)}
\FunctionTok{library}\NormalTok{(TCGAbiolinks)}
\FunctionTok{library}\NormalTok{(stringr)}
\FunctionTok{library}\NormalTok{(dplyr)}
\FunctionTok{library}\NormalTok{(tidyr)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{query\_clin }\OtherTok{\textless{}{-}} \FunctionTok{GDCquery}\NormalTok{(}
  \AttributeTok{project =}\NormalTok{ proj,}
  \AttributeTok{data.category =} \StringTok{"Clinical"}\NormalTok{,}
  \AttributeTok{data.type =} \StringTok{"Clinical Supplement"}\NormalTok{,}
  \AttributeTok{data.format =} \StringTok{"BCR XML"}
\NormalTok{)}

\CommentTok{\# Only run once (then comment)}
\FunctionTok{GDCdownload}\NormalTok{(}\AttributeTok{query =}\NormalTok{ query\_clin)}

\NormalTok{clinical\_kirp }\OtherTok{\textless{}{-}} \FunctionTok{GDCprepare\_clinic}\NormalTok{(query\_clin, }\AttributeTok{clinical.info =} \StringTok{"patient"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{clinical\_kirp }\OtherTok{\textless{}{-}}\NormalTok{ clinical\_kirp }\SpecialCharTok{\%\textgreater{}\%}
\NormalTok{  dplyr}\SpecialCharTok{::}\FunctionTok{mutate}\NormalTok{(}
    \AttributeTok{patient =}\NormalTok{ bcr\_patient\_barcode,}
    \AttributeTok{survival\_time =} \FunctionTok{ifelse}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(days\_to\_death),}
\NormalTok{                           days\_to\_death,}
\NormalTok{                           days\_to\_last\_followup),}
    \AttributeTok{survival\_status =} \FunctionTok{ifelse}\NormalTok{(vital\_status }\SpecialCharTok{==} \StringTok{"Dead"}\NormalTok{, }\DecValTok{1}\NormalTok{, }\DecValTok{0}\NormalTok{)}
\NormalTok{  ) }\SpecialCharTok{\%\textgreater{}\%}
\NormalTok{  dplyr}\SpecialCharTok{::}\FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(survival\_time)) }\SpecialCharTok{\%\textgreater{}\%}
\NormalTok{  dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(}
\NormalTok{    patient,}
\NormalTok{    survival\_time,}
\NormalTok{    survival\_status,}
\NormalTok{    age\_at\_initial\_pathologic\_diagnosis,}
\NormalTok{    gender}
\NormalTok{  )}

\FunctionTok{head}\NormalTok{(clinical\_kirp)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##        patient survival_time survival_status
## 1 TCGA-2K-A9WE           214               0
## 2 TCGA-2Z-A9J1          2298               0
## 3 TCGA-2Z-A9J2          1795               0
## 4 TCGA-2Z-A9J3          1771              NA
## 5 TCGA-2Z-A9J5          3050               0
## 6 TCGA-2Z-A9J6          1731               0
##   age_at_initial_pathologic_diagnosis gender
## 1                                  53   MALE
## 2                                  71   MALE
## 3                                  71 FEMALE
## 4                                  67   MALE
## 5                                  80   MALE
## 6                                  60   MALE
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{DDR\_genes  }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"TP53"}\NormalTok{, }\StringTok{"ATM"}\NormalTok{, }\StringTok{"ATR"}\NormalTok{, }\StringTok{"BRCA1"}\NormalTok{, }\StringTok{"BRCA2"}\NormalTok{, }\StringTok{"CHEK1"}\NormalTok{, }\StringTok{"CHEK2"}\NormalTok{)}
\NormalTok{TP53\_path  }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"TP53"}\NormalTok{)}
\NormalTok{WNT\_genes  }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"APC"}\NormalTok{, }\StringTok{"CTNNB1"}\NormalTok{, }\StringTok{"AXIN1"}\NormalTok{, }\StringTok{"LRP5"}\NormalTok{, }\StringTok{"LRP6"}\NormalTok{)}
\NormalTok{RTK\_RAS\_genes }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"KRAS"}\NormalTok{, }\StringTok{"NRAS"}\NormalTok{, }\StringTok{"HRAS"}\NormalTok{, }\StringTok{"BRAF"}\NormalTok{, }\StringTok{"EGFR"}\NormalTok{, }\StringTok{"ERBB2"}\NormalTok{)}
\NormalTok{PI3K\_genes }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"PIK3CA"}\NormalTok{, }\StringTok{"PTEN"}\NormalTok{, }\StringTok{"AKT1"}\NormalTok{, }\StringTok{"AKT2"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{DDR\_genes      }\OtherTok{\textless{}{-}} \FunctionTok{intersect}\NormalTok{(DDR\_genes, }\FunctionTok{colnames}\NormalTok{(mut\_matrix\_binary))}
\NormalTok{TP53\_path      }\OtherTok{\textless{}{-}} \FunctionTok{intersect}\NormalTok{(TP53\_path, }\FunctionTok{colnames}\NormalTok{(mut\_matrix\_binary))}
\NormalTok{WNT\_genes      }\OtherTok{\textless{}{-}} \FunctionTok{intersect}\NormalTok{(WNT\_genes, }\FunctionTok{colnames}\NormalTok{(mut\_matrix\_binary))}
\NormalTok{RTK\_RAS\_genes  }\OtherTok{\textless{}{-}} \FunctionTok{intersect}\NormalTok{(RTK\_RAS\_genes, }\FunctionTok{colnames}\NormalTok{(mut\_matrix\_binary))}
\NormalTok{PI3K\_genes     }\OtherTok{\textless{}{-}} \FunctionTok{intersect}\NormalTok{(PI3K\_genes, }\FunctionTok{colnames}\NormalTok{(mut\_matrix\_binary))}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pathway\_matrix }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{patient =} \FunctionTok{rownames}\NormalTok{(mut\_matrix\_binary),}
  \AttributeTok{DDR      =} \FunctionTok{as.integer}\NormalTok{(}\FunctionTok{rowSums}\NormalTok{(mut\_matrix\_binary[, DDR\_genes, }\AttributeTok{drop =} \ConstantTok{FALSE}\NormalTok{]) }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{),}
  \AttributeTok{TP53     =} \FunctionTok{as.integer}\NormalTok{(}\FunctionTok{rowSums}\NormalTok{(mut\_matrix\_binary[, TP53\_path, }\AttributeTok{drop =} \ConstantTok{FALSE}\NormalTok{]) }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{),}
  \AttributeTok{WNT      =} \FunctionTok{as.integer}\NormalTok{(}\FunctionTok{rowSums}\NormalTok{(mut\_matrix\_binary[, WNT\_genes, }\AttributeTok{drop =} \ConstantTok{FALSE}\NormalTok{]) }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{),}
  \AttributeTok{RTK\_RAS  =} \FunctionTok{as.integer}\NormalTok{(}\FunctionTok{rowSums}\NormalTok{(mut\_matrix\_binary[, RTK\_RAS\_genes, }\AttributeTok{drop =} \ConstantTok{FALSE}\NormalTok{]) }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{),}
  \AttributeTok{PI3K     =} \FunctionTok{as.integer}\NormalTok{(}\FunctionTok{rowSums}\NormalTok{(mut\_matrix\_binary[, PI3K\_genes, }\AttributeTok{drop =} \ConstantTok{FALSE}\NormalTok{]) }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{)}
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Compute TMB}
\NormalTok{tmb\_res }\OtherTok{\textless{}{-}} \FunctionTok{tmb}\NormalTok{(maf\_obj)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## --Mutiple centers found
## BI;BCM--Possible FLAGS among top ten genes:
##   TTN
##   MUC16
## -Processing clinical data
\end{verbatim}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-158-1.pdf}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Convert data.table {-}\textgreater{} data.frame BEFORE dplyr}
\NormalTok{tmb\_df }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(tmb\_res) }\SpecialCharTok{\%\textgreater{}\%}
\NormalTok{  dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(Tumor\_Sample\_Barcode, total\_perMB) }\SpecialCharTok{\%\textgreater{}\%}
\NormalTok{  dplyr}\SpecialCharTok{::}\FunctionTok{mutate}\NormalTok{(}
    \AttributeTok{patient  =} \FunctionTok{substr}\NormalTok{(Tumor\_Sample\_Barcode, }\DecValTok{1}\NormalTok{, }\DecValTok{12}\NormalTok{),}
    \AttributeTok{TMB      =}\NormalTok{ total\_perMB,}
    \AttributeTok{TMB\_high =}\NormalTok{ TMB }\SpecialCharTok{\textgreater{}} \DecValTok{10}
\NormalTok{  ) }\SpecialCharTok{\%\textgreater{}\%}
\NormalTok{  dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(patient, TMB, TMB\_high)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{clinical\_kirp }\OtherTok{\textless{}{-}}\NormalTok{ clinical\_kirp }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{distinct}\NormalTok{(patient, }\AttributeTok{.keep\_all =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{clinical\_kirp }\OtherTok{\textless{}{-}}\NormalTok{ clinical\_kirp }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(survival\_time), }\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(survival\_status))}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# Alignment checks (RNA vs mutation vs clinical)}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{patients\_rna }\OtherTok{\textless{}{-}} \FunctionTok{colnames}\NormalTok{(tumor\_norm\_log)}
\NormalTok{patients\_mut }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(mut\_matrix\_binary)}
\NormalTok{patients\_clin }\OtherTok{\textless{}{-}}\NormalTok{ clinical\_kirp}\SpecialCharTok{$}\NormalTok{patient}

\FunctionTok{cat}\NormalTok{(}\StringTok{"RNA tumor patients:"}\NormalTok{, }\FunctionTok{length}\NormalTok{(patients\_rna), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## RNA tumor patients: 290
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"Mutation patients:"}\NormalTok{, }\FunctionTok{length}\NormalTok{(patients\_mut), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Mutation patients: 278
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"Clinical patients:"}\NormalTok{, }\FunctionTok{length}\NormalTok{(patients\_clin), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Clinical patients: 249
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"RNA ∩ Mutation:"}\NormalTok{,}
    \FunctionTok{length}\NormalTok{(}\FunctionTok{intersect}\NormalTok{(patients\_rna, patients\_mut)), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## RNA ∩ Mutation: 277
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"RNA ∩ Clinical:"}\NormalTok{,}
    \FunctionTok{length}\NormalTok{(}\FunctionTok{intersect}\NormalTok{(patients\_rna, patients\_clin)), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## RNA ∩ Clinical: 248
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\#  DEG panel availability in tumor{-}only cohort}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\CommentTok{\# DEG\_list is already defined from paired analysis}
\FunctionTok{length}\NormalTok{(DEG\_list)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 724
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Intersection with tumor{-}only expression matrix}
\NormalTok{DEG\_list\_all }\OtherTok{\textless{}{-}} \FunctionTok{intersect}\NormalTok{(DEG\_list, }\FunctionTok{rownames}\NormalTok{(tumor\_norm\_log))}

\FunctionTok{cat}\NormalTok{(}\StringTok{"DEGs from paired analysis:"}\NormalTok{, }\FunctionTok{length}\NormalTok{(DEG\_list), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## DEGs from paired analysis: 724
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"DEGs available in tumor{-}only matrix:"}\NormalTok{, }\FunctionTok{length}\NormalTok{(DEG\_list\_all), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## DEGs available in tumor-only matrix: 724
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{X\_tumor\_deg }\OtherTok{\textless{}{-}}\NormalTok{ tumor\_norm\_log[DEG\_list, , drop }\OtherTok{=} \ConstantTok{FALSE}\NormalTok{]}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Final PSN feature matrix:"}\NormalTok{,}
    \FunctionTok{nrow}\NormalTok{(X\_tumor\_deg), }\StringTok{"genes x"}\NormalTok{,}
    \FunctionTok{ncol}\NormalTok{(X\_tumor\_deg), }\StringTok{"patients}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Final PSN feature matrix: 724 genes x 290 patients
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# PSN (RNA{-}only, FULL)}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{target\_data\_all }\OtherTok{\textless{}{-}} \FunctionTok{as.matrix}\NormalTok{(X\_tumor\_deg)}

\NormalTok{psn\_full }\OtherTok{\textless{}{-}} \FunctionTok{cor}\NormalTok{(target\_data\_all, }\AttributeTok{method =} \StringTok{"pearson"}\NormalTok{)}
\FunctionTok{diag}\NormalTok{(psn\_full) }\OtherTok{\textless{}{-}} \DecValTok{0}
\NormalTok{psn\_full[psn\_full }\SpecialCharTok{\textless{}} \DecValTok{0}\NormalTok{] }\OtherTok{\textless{}{-}} \DecValTok{0}   \CommentTok{\# opzionale ma ok}

\FunctionTok{stopifnot}\NormalTok{(}\FunctionTok{all}\NormalTok{(}\FunctionTok{is.finite}\NormalTok{(psn\_full)))}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Full PSN:"}\NormalTok{, }\FunctionTok{dim}\NormalTok{(psn\_full), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Full PSN: 290 290
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# Louvain on FULL PSN}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{psn\_graph\_full }\OtherTok{\textless{}{-}} \FunctionTok{graph\_from\_adjacency\_matrix}\NormalTok{(}
\NormalTok{  psn\_full,}
  \AttributeTok{mode =} \StringTok{"undirected"}\NormalTok{,}
  \AttributeTok{weighted =} \ConstantTok{TRUE}\NormalTok{,}
  \AttributeTok{diag =} \ConstantTok{FALSE}
\NormalTok{)}

\NormalTok{louvain\_psn\_all }\OtherTok{\textless{}{-}} \FunctionTok{cluster\_louvain}\NormalTok{(psn\_graph\_full)}

\NormalTok{comm\_res\_psn\_all }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{patient   =} \FunctionTok{names}\NormalTok{(}\FunctionTok{membership}\NormalTok{(louvain\_psn\_all)),}
  \AttributeTok{community =} \FunctionTok{as.character}\NormalTok{(}\FunctionTok{membership}\NormalTok{(louvain\_psn\_all)),}
  \AttributeTok{stringsAsFactors =} \ConstantTok{FALSE}
\NormalTok{)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Number of PSN communities:"}\NormalTok{,}
    \FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(comm\_res\_psn\_all}\SpecialCharTok{$}\NormalTok{community)), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Number of PSN communities: 3
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# PSN for visualization ONLY}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{plot\_thr }\OtherTok{\textless{}{-}} \FloatTok{0.8}

\NormalTok{psn\_plot }\OtherTok{\textless{}{-}}\NormalTok{ psn\_full}
\NormalTok{psn\_plot[psn\_plot }\SpecialCharTok{\textless{}}\NormalTok{ plot\_thr] }\OtherTok{\textless{}{-}} \DecValTok{0}
\FunctionTok{diag}\NormalTok{(psn\_plot) }\OtherTok{\textless{}{-}} \DecValTok{0}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Edges for plot:"}\NormalTok{, }\FunctionTok{sum}\NormalTok{(psn\_plot }\SpecialCharTok{\textgreater{}} \DecValTok{0}\NormalTok{) }\SpecialCharTok{/} \DecValTok{2}\NormalTok{, }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Edges for plot: 9095
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# Network object for ggnet2}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{net\_psn\_all }\OtherTok{\textless{}{-}} \FunctionTok{network}\NormalTok{(}
\NormalTok{  psn\_plot,}
  \AttributeTok{matrix.type =} \StringTok{"adjacency"}\NormalTok{,}
  \AttributeTok{ignore.eval =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{names.eval =} \StringTok{"weights"}\NormalTok{,}
  \AttributeTok{directed =} \ConstantTok{FALSE}
\NormalTok{)}

\NormalTok{net\_names }\OtherTok{\textless{}{-}} \FunctionTok{network.vertex.names}\NormalTok{(net\_psn\_all)}
\NormalTok{net\_psn\_all }\SpecialCharTok{\%v\%} \StringTok{"community"} \OtherTok{\textless{}{-}}
\NormalTok{  comm\_res\_psn\_all}\SpecialCharTok{$}\NormalTok{community[}\FunctionTok{match}\NormalTok{(net\_names, comm\_res\_psn\_all}\SpecialCharTok{$}\NormalTok{patient)]}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# Plot}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{n\_comm }\OtherTok{\textless{}{-}} \FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(net\_psn\_all }\SpecialCharTok{\%v\%} \StringTok{"community"}\NormalTok{))}
\FunctionTok{set.seed}\NormalTok{(}\DecValTok{13}\NormalTok{)}
\NormalTok{pal }\OtherTok{\textless{}{-}} \FunctionTok{sample}\NormalTok{(}\FunctionTok{colors}\NormalTok{(}\AttributeTok{distinct =} \ConstantTok{TRUE}\NormalTok{), n\_comm)}
\FunctionTok{names}\NormalTok{(pal) }\OtherTok{\textless{}{-}} \FunctionTok{sort}\NormalTok{(}\FunctionTok{unique}\NormalTok{(net\_psn\_all }\SpecialCharTok{\%v\%} \StringTok{"community"}\NormalTok{))}

\FunctionTok{ggnet2}\NormalTok{(net\_psn\_all,}
       \AttributeTok{color =} \StringTok{"community"}\NormalTok{,}
       \AttributeTok{palette =}\NormalTok{ pal,}
       \AttributeTok{alpha =} \FloatTok{0.9}\NormalTok{,}
       \AttributeTok{size =} \DecValTok{5}\NormalTok{,}
       \AttributeTok{label =} \ConstantTok{FALSE}\NormalTok{,}
       \AttributeTok{edge.color =} \StringTok{"gray80"}\NormalTok{,}
       \AttributeTok{edge.alpha =} \FloatTok{0.6}\NormalTok{,}
       \AttributeTok{edge.size =} \FloatTok{0.3}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{ggtitle}\NormalTok{(}\StringTok{"PSN Communities (Full clustering, edges r \textgreater{} 0.8)"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-167-1.pdf}}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# Patient alignment (RNA vs Mutation)}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{patients\_rna\_all }\OtherTok{\textless{}{-}} \FunctionTok{colnames}\NormalTok{(X\_tumor\_deg)}
\NormalTok{patients\_mut }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(mut\_matrix\_binary)}

\NormalTok{common\_patients\_all }\OtherTok{\textless{}{-}} \FunctionTok{intersect}\NormalTok{(patients\_rna\_all, patients\_mut)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Patients RNA:"}\NormalTok{, }\FunctionTok{length}\NormalTok{(patients\_rna\_all), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Patients RNA: 290
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"Patients Mutation:"}\NormalTok{, }\FunctionTok{length}\NormalTok{(patients\_mut), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Patients Mutation: 278
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"Common patients for SNF:"}\NormalTok{, }\FunctionTok{length}\NormalTok{(common\_patients\_all), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Common patients for SNF: 277
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# Prepare SNF input matrices}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\CommentTok{\# RNA view: patients x genes}
\NormalTok{data\_expression\_all }\OtherTok{\textless{}{-}} \FunctionTok{t}\NormalTok{(X\_tumor\_deg[, common\_patients\_all, }\AttributeTok{drop =} \ConstantTok{FALSE}\NormalTok{])}

\CommentTok{\# Mutation view: patients x genes}
\NormalTok{data\_mutation\_all }\OtherTok{\textless{}{-}}\NormalTok{ mut\_matrix\_binary[common\_patients\_all, , drop }\OtherTok{=} \ConstantTok{FALSE}\NormalTok{]}

\CommentTok{\# Final check}
\FunctionTok{stopifnot}\NormalTok{(}\FunctionTok{all}\NormalTok{(}\FunctionTok{rownames}\NormalTok{(data\_expression\_all) }\SpecialCharTok{==} \FunctionTok{rownames}\NormalTok{(data\_mutation\_all)))}

\FunctionTok{cat}\NormalTok{(}\StringTok{"RNA view:"}\NormalTok{, }\FunctionTok{dim}\NormalTok{(data\_expression\_all), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## RNA view: 277 724
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"Mutation view:"}\NormalTok{, }\FunctionTok{dim}\NormalTok{(data\_mutation\_all), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Mutation view: 277 8373
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# RNA affinity matrix}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{dist\_exp\_all }\OtherTok{\textless{}{-}} \FunctionTok{dist2}\NormalTok{(}
  \FunctionTok{as.matrix}\NormalTok{(data\_expression\_all),}
  \FunctionTok{as.matrix}\NormalTok{(data\_expression\_all)}
\NormalTok{)}

\NormalTok{W\_exp\_all }\OtherTok{\textless{}{-}} \FunctionTok{affinityMatrix}\NormalTok{(dist\_exp\_all, K, alpha)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# Mutation affinity matrix}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{dist\_mut\_all }\OtherTok{\textless{}{-}} \FunctionTok{dist2}\NormalTok{(}
  \FunctionTok{as.matrix}\NormalTok{(data\_mutation\_all),}
  \FunctionTok{as.matrix}\NormalTok{(data\_mutation\_all)}
\NormalTok{)}

\NormalTok{W\_mut\_all }\OtherTok{\textless{}{-}} \FunctionTok{affinityMatrix}\NormalTok{(dist\_mut\_all, K, alpha)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# Network fusion}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{W\_fused\_all }\OtherTok{\textless{}{-}} \FunctionTok{SNF}\NormalTok{(}
  \FunctionTok{list}\NormalTok{(W\_exp\_all, W\_mut\_all),}
\NormalTok{  K,}
\NormalTok{  T\_iter}
\NormalTok{)}

\FunctionTok{rownames}\NormalTok{(W\_fused\_all) }\OtherTok{\textless{}{-}}\NormalTok{ common\_patients\_all}
\FunctionTok{colnames}\NormalTok{(W\_fused\_all) }\OtherTok{\textless{}{-}}\NormalTok{ common\_patients\_all}

\CommentTok{\# Remove self{-}loops}
\FunctionTok{diag}\NormalTok{(W\_fused\_all) }\OtherTok{\textless{}{-}} \DecValTok{0}


\NormalTok{W\_fused\_all[W\_fused\_all }\SpecialCharTok{\textless{}} \DecValTok{0}\NormalTok{] }\OtherTok{\textless{}{-}} \DecValTok{0}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{graph\_snf\_all }\OtherTok{\textless{}{-}} \FunctionTok{graph\_from\_adjacency\_matrix}\NormalTok{(}
\NormalTok{  W\_fused\_all,}
  \AttributeTok{mode =} \StringTok{"undirected"}\NormalTok{,}
  \AttributeTok{weighted =} \ConstantTok{TRUE}\NormalTok{,}
  \AttributeTok{diag =} \ConstantTok{FALSE}
\NormalTok{)}

\NormalTok{louvain\_snf\_all }\OtherTok{\textless{}{-}} \FunctionTok{cluster\_louvain}\NormalTok{(graph\_snf\_all)}

\NormalTok{comm\_res\_snf\_all }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}
  \AttributeTok{patient =} \FunctionTok{names}\NormalTok{(}\FunctionTok{membership}\NormalTok{(louvain\_snf\_all)),}
  \AttributeTok{community\_snf =} \FunctionTok{as.character}\NormalTok{(}\FunctionTok{membership}\NormalTok{(louvain\_snf\_all)),}
  \AttributeTok{stringsAsFactors =} \ConstantTok{FALSE}
\NormalTok{)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Number of SNF communities:"}\NormalTok{,}
    \FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(comm\_res\_snf\_all}\SpecialCharTok{$}\NormalTok{community\_snf)), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Number of SNF communities: 6
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{net\_snf\_all }\OtherTok{\textless{}{-}} \FunctionTok{network}\NormalTok{(}
\NormalTok{  W\_fused\_all,}
  \AttributeTok{matrix.type =} \StringTok{"adjacency"}\NormalTok{,}
  \AttributeTok{ignore.eval =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{names.eval =} \StringTok{"weights"}\NormalTok{,}
  \AttributeTok{directed =} \ConstantTok{FALSE}
\NormalTok{)}

\NormalTok{net\_names }\OtherTok{\textless{}{-}} \FunctionTok{network.vertex.names}\NormalTok{(net\_snf\_all)}
\NormalTok{net\_snf\_all }\SpecialCharTok{\%v\%} \StringTok{"community"} \OtherTok{\textless{}{-}}
\NormalTok{  comm\_res\_snf\_all}\SpecialCharTok{$}\NormalTok{community\_snf[}
    \FunctionTok{match}\NormalTok{(net\_names, comm\_res\_snf\_all}\SpecialCharTok{$}\NormalTok{patient)}
\NormalTok{  ]}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# Standardize patient IDs (12 chars) + deduplicate}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{comm\_res\_snf\_all}\SpecialCharTok{$}\NormalTok{patient }\OtherTok{\textless{}{-}} \FunctionTok{substr}\NormalTok{(comm\_res\_snf\_all}\SpecialCharTok{$}\NormalTok{patient, }\DecValTok{1}\NormalTok{, }\DecValTok{12}\NormalTok{)}
\NormalTok{clinical\_kirp}\SpecialCharTok{$}\NormalTok{patient    }\OtherTok{\textless{}{-}} \FunctionTok{substr}\NormalTok{(clinical\_kirp}\SpecialCharTok{$}\NormalTok{patient, }\DecValTok{1}\NormalTok{, }\DecValTok{12}\NormalTok{)}
\NormalTok{pathway\_matrix}\SpecialCharTok{$}\NormalTok{patient   }\OtherTok{\textless{}{-}} \FunctionTok{substr}\NormalTok{(pathway\_matrix}\SpecialCharTok{$}\NormalTok{patient, }\DecValTok{1}\NormalTok{, }\DecValTok{12}\NormalTok{)}
\NormalTok{tmb\_df}\SpecialCharTok{$}\NormalTok{patient           }\OtherTok{\textless{}{-}} \FunctionTok{substr}\NormalTok{(tmb\_df}\SpecialCharTok{$}\NormalTok{patient, }\DecValTok{1}\NormalTok{, }\DecValTok{12}\NormalTok{)}

\NormalTok{clinical\_kirp  }\OtherTok{\textless{}{-}}\NormalTok{ clinical\_kirp }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{distinct}\NormalTok{(patient, }\AttributeTok{.keep\_all =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{pathway\_matrix }\OtherTok{\textless{}{-}}\NormalTok{ pathway\_matrix }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{distinct}\NormalTok{(patient, }\AttributeTok{.keep\_all =} \ConstantTok{TRUE}\NormalTok{)}
\NormalTok{tmb\_df         }\OtherTok{\textless{}{-}}\NormalTok{ tmb\_df }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{distinct}\NormalTok{(patient, }\AttributeTok{.keep\_all =} \ConstantTok{TRUE}\NormalTok{)}

\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# Build master table (SNF + clinical + enrichment features)}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{master\_table\_all }\OtherTok{\textless{}{-}}\NormalTok{ comm\_res\_snf\_all }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{rename}\NormalTok{(}\AttributeTok{cluster\_snf =}\NormalTok{ community\_snf) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{left\_join}\NormalTok{(clinical\_kirp,  }\AttributeTok{by =} \StringTok{"patient"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{left\_join}\NormalTok{(pathway\_matrix, }\AttributeTok{by =} \StringTok{"patient"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{left\_join}\NormalTok{(tmb\_df,         }\AttributeTok{by =} \StringTok{"patient"}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{distinct}\NormalTok{(patient, }\AttributeTok{.keep\_all =} \ConstantTok{TRUE}\NormalTok{)}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Patients in master table:"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(master\_table\_all), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Patients in master table: 277
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"SNF clusters:"}\NormalTok{, }\FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(master\_table\_all}\SpecialCharTok{$}\NormalTok{cluster\_snf)), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## SNF clusters: 6
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{summary}\NormalTok{(master\_table\_all[, }\FunctionTok{c}\NormalTok{(}\StringTok{"survival\_time"}\NormalTok{,}\StringTok{"survival\_status"}\NormalTok{)])}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##  survival_time    survival_status  
##  Min.   :   4.0   Min.   :0.00000  
##  1st Qu.: 450.2   1st Qu.:0.00000  
##  Median : 768.5   Median :0.00000  
##  Mean   :1045.7   Mean   :0.06667  
##  3rd Qu.:1384.0   3rd Qu.:0.00000  
##  Max.   :5925.0   Max.   :1.00000  
##  NA's   :37       NA's   :37
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# Survival: Kaplan–Meier by SNF cluster}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{surv\_df }\OtherTok{\textless{}{-}}\NormalTok{ master\_table\_all }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(survival\_time), }\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(survival\_status), }\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(cluster\_snf)) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}
    \AttributeTok{cluster\_snf =} \FunctionTok{factor}\NormalTok{(cluster\_snf),}
    \AttributeTok{survival\_time =} \FunctionTok{as.numeric}\NormalTok{(survival\_time),}
    \AttributeTok{survival\_status =} \FunctionTok{as.numeric}\NormalTok{(survival\_status)}
\NormalTok{  )}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Patients usable for survival:"}\NormalTok{, }\FunctionTok{nrow}\NormalTok{(surv\_df), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Patients usable for survival: 240
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{table}\NormalTok{(surv\_df}\SpecialCharTok{$}\NormalTok{cluster\_snf)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  1  2  3  4  5  6 
## 57 39 29 42 39 34
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{km\_snf }\OtherTok{\textless{}{-}} \FunctionTok{survfit}\NormalTok{(}\FunctionTok{Surv}\NormalTok{(survival\_time, survival\_status) }\SpecialCharTok{\textasciitilde{}}\NormalTok{ cluster\_snf, }\AttributeTok{data =}\NormalTok{ surv\_df)}

\FunctionTok{ggsurvplot}\NormalTok{(}
\NormalTok{  km\_snf,}
  \AttributeTok{data =}\NormalTok{ surv\_df,}
  \AttributeTok{palette =} \StringTok{"Dark2"}\NormalTok{,}
  \AttributeTok{size =} \FloatTok{1.3}\NormalTok{,}
  \AttributeTok{censor.size =} \DecValTok{3}\NormalTok{,}
  \AttributeTok{pval =} \ConstantTok{TRUE}\NormalTok{,}
  \AttributeTok{conf.int =} \ConstantTok{FALSE}\NormalTok{,}
  \AttributeTok{xlab =} \StringTok{"Time (days)"}\NormalTok{,}
  \AttributeTok{ylab =} \StringTok{"Overall survival probability"}\NormalTok{,}
  \AttributeTok{legend.title =} \StringTok{"SNF community"}\NormalTok{,}
  \AttributeTok{ggtheme =} \FunctionTok{theme\_classic}\NormalTok{(}\AttributeTok{base\_size =} \DecValTok{15}\NormalTok{)}
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-176-1.pdf}}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# Helper functions for enrichment tests}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\CommentTok{\# Robust Fisher test (fallback to simulation if needed)}
\NormalTok{safe\_fisher }\OtherTok{\textless{}{-}} \ControlFlowTok{function}\NormalTok{(tab) \{}
\NormalTok{  out }\OtherTok{\textless{}{-}} \FunctionTok{tryCatch}\NormalTok{(}
    \FunctionTok{fisher.test}\NormalTok{(tab),}
    \AttributeTok{error =} \ControlFlowTok{function}\NormalTok{(e)}
      \FunctionTok{fisher.test}\NormalTok{(tab, }\AttributeTok{simulate.p.value =} \ConstantTok{TRUE}\NormalTok{, }\AttributeTok{B =} \DecValTok{20000}\NormalTok{)}
\NormalTok{  )}
\NormalTok{  out}\SpecialCharTok{$}\NormalTok{p.value}
\NormalTok{\}}

\CommentTok{\# Categorical enrichment (e.g. TMB\_high, pathway mutation indicators)}
\NormalTok{cluster\_enrichment\_categorical }\OtherTok{\textless{}{-}} \ControlFlowTok{function}\NormalTok{(df, cluster\_col, var\_col) \{}

\NormalTok{  tmp }\OtherTok{\textless{}{-}}\NormalTok{ df }\SpecialCharTok{\%\textgreater{}\%}
\NormalTok{    dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(}
\NormalTok{      dplyr}\SpecialCharTok{::}\FunctionTok{all\_of}\NormalTok{(cluster\_col),}
\NormalTok{      dplyr}\SpecialCharTok{::}\FunctionTok{all\_of}\NormalTok{(var\_col)}
\NormalTok{    ) }\SpecialCharTok{\%\textgreater{}\%}
\NormalTok{    dplyr}\SpecialCharTok{::}\FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(.data[[var\_col]]))}

\NormalTok{  tab }\OtherTok{\textless{}{-}} \FunctionTok{table}\NormalTok{(tmp[[cluster\_col]], tmp[[var\_col]])}

  \CommentTok{\# Fisher test not defined if table has \textless{}2 rows or columns}
  \ControlFlowTok{if}\NormalTok{ (}\FunctionTok{min}\NormalTok{(}\FunctionTok{dim}\NormalTok{(tab)) }\SpecialCharTok{\textless{}} \DecValTok{2}\NormalTok{) \{}
    \FunctionTok{return}\NormalTok{(}\FunctionTok{list}\NormalTok{(}
      \AttributeTok{table =}\NormalTok{ tab,}
      \AttributeTok{p.value =} \ConstantTok{NA}
\NormalTok{    ))}
\NormalTok{  \}}

\NormalTok{  p }\OtherTok{\textless{}{-}} \FunctionTok{safe\_fisher}\NormalTok{(tab)}

  \FunctionTok{list}\NormalTok{(}
    \AttributeTok{table =}\NormalTok{ tab,}
    \AttributeTok{p.value =}\NormalTok{ p}
\NormalTok{  )}
\NormalTok{\}}

\CommentTok{\# Continuous enrichment (e.g. TMB)}
\NormalTok{cluster\_enrichment\_continuous }\OtherTok{\textless{}{-}} \ControlFlowTok{function}\NormalTok{(df, cluster\_col, var\_col) \{}

\NormalTok{  tmp }\OtherTok{\textless{}{-}}\NormalTok{ df }\SpecialCharTok{\%\textgreater{}\%}
\NormalTok{    dplyr}\SpecialCharTok{::}\FunctionTok{select}\NormalTok{(}
\NormalTok{      dplyr}\SpecialCharTok{::}\FunctionTok{all\_of}\NormalTok{(cluster\_col),}
\NormalTok{      dplyr}\SpecialCharTok{::}\FunctionTok{all\_of}\NormalTok{(var\_col)}
\NormalTok{    ) }\SpecialCharTok{\%\textgreater{}\%}
\NormalTok{    dplyr}\SpecialCharTok{::}\FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(.data[[var\_col]]))}

  \CommentTok{\# Kruskal–Wallis requires at least 2 groups}
  \ControlFlowTok{if}\NormalTok{ (}\FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(tmp[[cluster\_col]])) }\SpecialCharTok{\textless{}} \DecValTok{2}\NormalTok{) \{}
    \FunctionTok{return}\NormalTok{(}\ConstantTok{NA}\NormalTok{)}
\NormalTok{  \}}

  \FunctionTok{kruskal.test}\NormalTok{(tmp[[var\_col]] }\SpecialCharTok{\textasciitilde{}}\NormalTok{ tmp[[cluster\_col]])}\SpecialCharTok{$}\NormalTok{p.value}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\#  Enrichment: TMB\_high and pathway mutation indicators}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\CommentTok{\# TMB\_high (categorical)}
\NormalTok{res\_tmb\_high }\OtherTok{\textless{}{-}} \FunctionTok{cluster\_enrichment\_categorical}\NormalTok{(master\_table\_all, }\StringTok{"cluster\_snf"}\NormalTok{, }\StringTok{"TMB\_high"}\NormalTok{)}
\NormalTok{res\_tmb\_high}\SpecialCharTok{$}\NormalTok{table}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##    
##     FALSE
##   1    64
##   2    44
##   3    37
##   4    50
##   5    41
##   6    41
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{cat}\NormalTok{(}\StringTok{"Enrichment p{-}value (TMB\_high):"}\NormalTok{, res\_tmb\_high}\SpecialCharTok{$}\NormalTok{p.value, }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Enrichment p-value (TMB_high): NA
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# TMB (continuous)}
\NormalTok{p\_tmb }\OtherTok{\textless{}{-}} \FunctionTok{cluster\_enrichment\_continuous}\NormalTok{(master\_table\_all, }\StringTok{"cluster\_snf"}\NormalTok{, }\StringTok{"TMB"}\NormalTok{)}
\FunctionTok{cat}\NormalTok{(}\StringTok{"Kruskal p{-}value (TMB):"}\NormalTok{, p\_tmb, }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Kruskal p-value (TMB): 5.443439e-09
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Pathway indicators (categorical 0/1)}
\NormalTok{path\_vars }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(}\StringTok{"DDR"}\NormalTok{,}\StringTok{"TP53"}\NormalTok{,}\StringTok{"WNT"}\NormalTok{,}\StringTok{"RTK\_RAS"}\NormalTok{,}\StringTok{"PI3K"}\NormalTok{)}
\NormalTok{path\_vars }\OtherTok{\textless{}{-}}\NormalTok{ path\_vars[path\_vars }\SpecialCharTok{\%in\%} \FunctionTok{colnames}\NormalTok{(master\_table\_all)]}

\NormalTok{enrich\_path }\OtherTok{\textless{}{-}} \FunctionTok{lapply}\NormalTok{(path\_vars, }\ControlFlowTok{function}\NormalTok{(v) \{}
\NormalTok{  out }\OtherTok{\textless{}{-}} \FunctionTok{cluster\_enrichment\_categorical}\NormalTok{(master\_table\_all, }\StringTok{"cluster\_snf"}\NormalTok{, v)}
  \FunctionTok{data.frame}\NormalTok{(}\AttributeTok{variable =}\NormalTok{ v, }\AttributeTok{p.value =}\NormalTok{ out}\SpecialCharTok{$}\NormalTok{p.value)}
\NormalTok{\})}

\NormalTok{enrich\_path\_df }\OtherTok{\textless{}{-}} \FunctionTok{do.call}\NormalTok{(rbind, enrich\_path) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{p.adj =} \FunctionTok{p.adjust}\NormalTok{(p.value, }\AttributeTok{method =} \StringTok{"fdr"}\NormalTok{)) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{arrange}\NormalTok{(p.adj)}

\NormalTok{enrich\_path\_df}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   variable    p.value     p.adj
## 1      DDR 0.05643461 0.1410865
## 2     TP53 0.05064859 0.1410865
## 3  RTK_RAS 0.15878474 0.2646412
## 4      WNT 0.31239111 0.3904889
## 5     PI3K 0.43518209 0.4351821
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\#  Clinical enrichment: age + gender}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\CommentTok{\# Age}
\ControlFlowTok{if}\NormalTok{ (}\StringTok{"age\_at\_initial\_pathologic\_diagnosis"} \SpecialCharTok{\%in\%} \FunctionTok{colnames}\NormalTok{(master\_table\_all)) \{}
\NormalTok{  p\_age }\OtherTok{\textless{}{-}} \FunctionTok{cluster\_enrichment\_continuous}\NormalTok{(master\_table\_all, }\StringTok{"cluster\_snf"}\NormalTok{, }\StringTok{"age\_at\_initial\_pathologic\_diagnosis"}\NormalTok{)}
  \FunctionTok{cat}\NormalTok{(}\StringTok{"Kruskal p{-}value (age):"}\NormalTok{, p\_age, }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Kruskal p-value (age): 0.0460415
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Gender}
\ControlFlowTok{if}\NormalTok{ (}\StringTok{"gender"} \SpecialCharTok{\%in\%} \FunctionTok{colnames}\NormalTok{(master\_table\_all)) \{}
\NormalTok{  res\_gender }\OtherTok{\textless{}{-}} \FunctionTok{cluster\_enrichment\_categorical}\NormalTok{(master\_table\_all, }\StringTok{"cluster\_snf"}\NormalTok{, }\StringTok{"gender"}\NormalTok{)}
\NormalTok{  res\_gender}\SpecialCharTok{$}\NormalTok{table}
  \FunctionTok{cat}\NormalTok{(}\StringTok{"Enrichment p{-}value (gender):"}\NormalTok{, res\_gender}\SpecialCharTok{$}\NormalTok{p.value, }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Enrichment p-value (gender): 0.00179991
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# Prepare expression + metadata for community DE}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\CommentTok{\# Ensure same patients}
\NormalTok{common\_patients }\OtherTok{\textless{}{-}} \FunctionTok{intersect}\NormalTok{(}
  \FunctionTok{colnames}\NormalTok{(tumor\_norm\_log),}
\NormalTok{  comm\_res\_snf\_all}\SpecialCharTok{$}\NormalTok{patient}
\NormalTok{)}

\NormalTok{expr\_comm }\OtherTok{\textless{}{-}}\NormalTok{ tumor\_norm\_log[, common\_patients, drop }\OtherTok{=} \ConstantTok{FALSE}\NormalTok{]}

\NormalTok{meta\_comm }\OtherTok{\textless{}{-}}\NormalTok{ comm\_res\_snf\_all }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{filter}\NormalTok{(patient }\SpecialCharTok{\%in\%}\NormalTok{ common\_patients) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{community\_snf =} \FunctionTok{factor}\NormalTok{(community\_snf)) }\SpecialCharTok{\%\textgreater{}\%}
  \FunctionTok{arrange}\NormalTok{(}\FunctionTok{match}\NormalTok{(patient, common\_patients))}

\FunctionTok{stopifnot}\NormalTok{(}\FunctionTok{all}\NormalTok{(meta\_comm}\SpecialCharTok{$}\NormalTok{patient }\SpecialCharTok{==} \FunctionTok{colnames}\NormalTok{(expr\_comm)))}

\FunctionTok{cat}\NormalTok{(}\StringTok{"Patients for community DE:"}\NormalTok{, }\FunctionTok{ncol}\NormalTok{(expr\_comm), }\StringTok{"}\SpecialCharTok{\textbackslash{}n}\StringTok{"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Patients for community DE: 277
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{table}\NormalTok{(meta\_comm}\SpecialCharTok{$}\NormalTok{community\_snf)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
##  1  2  3  4  5  6 
## 64 44 37 50 41 41
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# Differential expression: community vs rest (LIMMA)}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\FunctionTok{library}\NormalTok{(limma)}

\CommentTok{\# Expression matrix: genes x samples}
\NormalTok{expr\_mat }\OtherTok{\textless{}{-}} \FunctionTok{as.matrix}\NormalTok{(expr\_comm)}

\CommentTok{\# Ensure syntactically valid community labels (e.g. C1, C2, ...)}
\NormalTok{meta\_comm}\SpecialCharTok{$}\NormalTok{community\_snf }\OtherTok{\textless{}{-}} \FunctionTok{factor}\NormalTok{(}
\NormalTok{  meta\_comm}\SpecialCharTok{$}\NormalTok{community\_snf,}
  \AttributeTok{labels =} \FunctionTok{paste0}\NormalTok{(}\StringTok{"C"}\NormalTok{, }\FunctionTok{levels}\NormalTok{(meta\_comm}\SpecialCharTok{$}\NormalTok{community\_snf))}
\NormalTok{)}

\CommentTok{\# Design matrix (no intercept)}
\NormalTok{design }\OtherTok{\textless{}{-}} \FunctionTok{model.matrix}\NormalTok{(}\SpecialCharTok{\textasciitilde{}} \DecValTok{0} \SpecialCharTok{+}\NormalTok{ community\_snf, }\AttributeTok{data =}\NormalTok{ meta\_comm)}
\FunctionTok{colnames}\NormalTok{(design) }\OtherTok{\textless{}{-}} \FunctionTok{levels}\NormalTok{(meta\_comm}\SpecialCharTok{$}\NormalTok{community\_snf)}

\CommentTok{\# Fit linear model}
\NormalTok{fit }\OtherTok{\textless{}{-}} \FunctionTok{lmFit}\NormalTok{(expr\_mat, design)}

\CommentTok{\# Build contrasts: each community vs the average of the others}
\NormalTok{communities }\OtherTok{\textless{}{-}} \FunctionTok{colnames}\NormalTok{(design)}

\NormalTok{contrast\_list }\OtherTok{\textless{}{-}} \FunctionTok{sapply}\NormalTok{(}
\NormalTok{  communities,}
  \ControlFlowTok{function}\NormalTok{(cl) \{}
    \FunctionTok{paste0}\NormalTok{(}
\NormalTok{      cl, }\StringTok{" {-} ("}\NormalTok{,}
      \FunctionTok{paste}\NormalTok{(}\FunctionTok{setdiff}\NormalTok{(communities, cl), }\AttributeTok{collapse =} \StringTok{" + "}\NormalTok{),}
      \StringTok{")/"}\NormalTok{, }\FunctionTok{length}\NormalTok{(}\FunctionTok{setdiff}\NormalTok{(communities, cl))}
\NormalTok{    )}
\NormalTok{  \}}
\NormalTok{)}

\NormalTok{contrast\_matrix }\OtherTok{\textless{}{-}} \FunctionTok{makeContrasts}\NormalTok{(}
  \AttributeTok{contrasts =}\NormalTok{ contrast\_list,}
  \AttributeTok{levels =}\NormalTok{ design}
\NormalTok{)}

\CommentTok{\# Apply contrasts and empirical Bayes moderation}
\NormalTok{fit2 }\OtherTok{\textless{}{-}} \FunctionTok{contrasts.fit}\NormalTok{(fit, contrast\_matrix)}
\NormalTok{fit2 }\OtherTok{\textless{}{-}} \FunctionTok{eBayes}\NormalTok{(fit2)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# Extract community{-}specific DEGs }
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{community\_deg\_list }\OtherTok{\textless{}{-}} \FunctionTok{list}\NormalTok{()}

\NormalTok{coef\_names }\OtherTok{\textless{}{-}} \FunctionTok{colnames}\NormalTok{(fit2}\SpecialCharTok{$}\NormalTok{coefficients)}

\ControlFlowTok{for}\NormalTok{ (i }\ControlFlowTok{in} \FunctionTok{seq\_along}\NormalTok{(communities)) \{}

\NormalTok{  cl }\OtherTok{\textless{}{-}}\NormalTok{ communities[i]}

\NormalTok{  res }\OtherTok{\textless{}{-}} \FunctionTok{topTable}\NormalTok{(}
\NormalTok{    fit2,}
    \AttributeTok{coef =}\NormalTok{ i,              }\CommentTok{\# \textless{}{-}{-} USE INDEX, NOT NAME}
    \AttributeTok{number =} \ConstantTok{Inf}\NormalTok{,}
    \AttributeTok{adjust.method =} \StringTok{"BH"}
\NormalTok{  )}

\NormalTok{  res}\SpecialCharTok{$}\NormalTok{gene }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(res)}

\NormalTok{  res\_sig }\OtherTok{\textless{}{-}}\NormalTok{ res }\SpecialCharTok{\%\textgreater{}\%}
\NormalTok{    dplyr}\SpecialCharTok{::}\FunctionTok{filter}\NormalTok{(}
\NormalTok{      adj.P.Val }\SpecialCharTok{\textless{}} \FloatTok{0.05}\NormalTok{,}
      \FunctionTok{abs}\NormalTok{(logFC) }\SpecialCharTok{\textgreater{}=} \DecValTok{1}
\NormalTok{    )}

\NormalTok{  community\_deg\_list[[cl]] }\OtherTok{\textless{}{-}}\NormalTok{ res\_sig}

  \FunctionTok{cat}\NormalTok{(}
    \StringTok{"Community"}\NormalTok{, cl, }\StringTok{"("}\NormalTok{, coef\_names[i], }\StringTok{"):"}\NormalTok{,}
    \FunctionTok{nrow}\NormalTok{(res\_sig), }\StringTok{"DEGs}\SpecialCharTok{\textbackslash{}n}\StringTok{"}
\NormalTok{  )}
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Community C1 ( C1 ): 913 DEGs
## Community C2 ( C2 ): 561 DEGs
## Community C3 ( C3 ): 1574 DEGs
## Community C4 ( C4 ): 786 DEGs
## Community C5 ( C5 ): 366 DEGs
## Community C6 ( C6 ): 522 DEGs
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\# Prepare gene sets for enrichment (ENTREZ) + background universe}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\CommentTok{\# 1) Universe (background): all tested genes in limma (recommended for ORA)}
\NormalTok{universe\_genes }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(expr\_comm)}
\NormalTok{universe\_genes }\OtherTok{\textless{}{-}} \FunctionTok{gsub}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{..*"}\NormalTok{, }\StringTok{""}\NormalTok{, universe\_genes)  }\CommentTok{\# remove version if ENSG.x}

\CommentTok{\# 2) Convert universe to ENTREZ (once)}
\NormalTok{universe\_map }\OtherTok{\textless{}{-}} \FunctionTok{bitr}\NormalTok{(}
\NormalTok{  universe\_genes,}
  \AttributeTok{fromType =} \FunctionTok{ifelse}\NormalTok{(}\FunctionTok{startsWith}\NormalTok{(universe\_genes[}\DecValTok{1}\NormalTok{], }\StringTok{"ENSG"}\NormalTok{), }\StringTok{"ENSEMBL"}\NormalTok{, }\StringTok{"SYMBOL"}\NormalTok{),}
  \AttributeTok{toType   =} \StringTok{"ENTREZID"}\NormalTok{,}
  \AttributeTok{OrgDb    =}\NormalTok{ org.Hs.eg.db}
\NormalTok{)}
\NormalTok{universe\_entrez }\OtherTok{\textless{}{-}} \FunctionTok{unique}\NormalTok{(universe\_map}\SpecialCharTok{$}\NormalTok{ENTREZID)}

\CommentTok{\# 3) Convert each community DEG list to ENTREZ}
\NormalTok{community\_entrez }\OtherTok{\textless{}{-}} \FunctionTok{lapply}\NormalTok{(}\FunctionTok{names}\NormalTok{(community\_deg\_list), }\ControlFlowTok{function}\NormalTok{(cl) \{}
\NormalTok{  df }\OtherTok{\textless{}{-}}\NormalTok{ community\_deg\_list[[cl]]}
  \ControlFlowTok{if}\NormalTok{ (}\FunctionTok{nrow}\NormalTok{(df) }\SpecialCharTok{==} \DecValTok{0}\NormalTok{) }\FunctionTok{return}\NormalTok{(}\FunctionTok{character}\NormalTok{(}\DecValTok{0}\NormalTok{))}

\NormalTok{  genes }\OtherTok{\textless{}{-}} \FunctionTok{gsub}\NormalTok{(}\StringTok{"}\SpecialCharTok{\textbackslash{}\textbackslash{}}\StringTok{..*"}\NormalTok{, }\StringTok{""}\NormalTok{, df}\SpecialCharTok{$}\NormalTok{gene)}

\NormalTok{  keyType }\OtherTok{\textless{}{-}} \FunctionTok{ifelse}\NormalTok{(}\FunctionTok{startsWith}\NormalTok{(genes[}\DecValTok{1}\NormalTok{], }\StringTok{"ENSG"}\NormalTok{), }\StringTok{"ENSEMBL"}\NormalTok{, }\StringTok{"SYMBOL"}\NormalTok{)}

\NormalTok{  mapped }\OtherTok{\textless{}{-}} \FunctionTok{bitr}\NormalTok{(}
\NormalTok{    genes,}
    \AttributeTok{fromType =}\NormalTok{ keyType,}
    \AttributeTok{toType   =} \StringTok{"ENTREZID"}\NormalTok{,}
    \AttributeTok{OrgDb    =}\NormalTok{ org.Hs.eg.db}
\NormalTok{  )}

  \FunctionTok{unique}\NormalTok{(mapped}\SpecialCharTok{$}\NormalTok{ENTREZID)}
\NormalTok{\})}
\FunctionTok{names}\NormalTok{(community\_entrez) }\OtherTok{\textless{}{-}} \FunctionTok{names}\NormalTok{(community\_deg\_list)}

\CommentTok{\# Quick checks}
\FunctionTok{sapply}\NormalTok{(community\_entrez, length)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##   C1   C2   C3   C4   C5   C6 
##  866  548 1522  762  353  513
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{length}\NormalTok{(universe\_entrez)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 37540
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{go\_results }\OtherTok{\textless{}{-}} \FunctionTok{compareCluster}\NormalTok{(}
  \AttributeTok{geneCluster   =}\NormalTok{ community\_entrez,}
  \AttributeTok{fun           =} \StringTok{"enrichGO"}\NormalTok{,}
  \AttributeTok{OrgDb         =}\NormalTok{ org.Hs.eg.db,}
  \AttributeTok{keyType       =} \StringTok{"ENTREZID"}\NormalTok{,}
  \AttributeTok{ont           =} \StringTok{"BP"}\NormalTok{,}
  \AttributeTok{universe      =}\NormalTok{ universe\_entrez,     }\CommentTok{\# IMPORTANT: correct background}
  \AttributeTok{pvalueCutoff  =} \FloatTok{0.05}\NormalTok{,}
  \AttributeTok{pAdjustMethod =} \StringTok{"BH"}\NormalTok{,}
  \AttributeTok{readable      =} \ConstantTok{TRUE}
\NormalTok{)}

\CommentTok{\# Remove redundant GO terms (recommended)}
\NormalTok{go\_results\_simpl }\OtherTok{\textless{}{-}} \FunctionTok{simplify}\NormalTok{(}
\NormalTok{  go\_results,}
  \AttributeTok{cutoff      =} \FloatTok{0.6}\NormalTok{,}
  \AttributeTok{by          =} \StringTok{"p.adjust"}\NormalTok{,}
  \AttributeTok{select\_fun  =}\NormalTok{ min}
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{dotplot}\NormalTok{(}
\NormalTok{  go\_results\_simpl,}
  \AttributeTok{showCategory =} \DecValTok{6}\NormalTok{,}
  \AttributeTok{title =} \StringTok{"GO Biological Process enrichment by SNF community"}
\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{theme\_bw}\NormalTok{(}\AttributeTok{base\_size =} \DecValTok{13}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}
    \AttributeTok{plot.title =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{hjust =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{face =} \StringTok{"bold"}\NormalTok{),}
    \AttributeTok{axis.text.y =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{9}\NormalTok{)}
\NormalTok{  )}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-185-1.pdf}}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}
\CommentTok{\#  KEGG pathway enrichment per SNF community}
\DocumentationTok{\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#}

\NormalTok{kegg\_results }\OtherTok{\textless{}{-}} \FunctionTok{compareCluster}\NormalTok{(}
  \AttributeTok{geneCluster   =}\NormalTok{ community\_entrez,}
  \AttributeTok{fun           =} \StringTok{"enrichKEGG"}\NormalTok{,}
  \AttributeTok{organism      =} \StringTok{"hsa"}\NormalTok{,}
  \AttributeTok{universe      =}\NormalTok{ universe\_entrez,}
  \AttributeTok{pvalueCutoff  =} \FloatTok{0.05}
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{dotplot}\NormalTok{(}
\NormalTok{  kegg\_results,}
  \AttributeTok{showCategory =} \DecValTok{5}\NormalTok{,}
  \AttributeTok{title =} \StringTok{"KEGG pathway enrichment by SNF community"}
\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{theme\_bw}\NormalTok{(}\AttributeTok{base\_size =} \DecValTok{13}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}
    \AttributeTok{plot.title =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{hjust =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{face =} \StringTok{"bold"}\NormalTok{),}
    \AttributeTok{axis.text.y =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size =} \DecValTok{9}\NormalTok{)}
\NormalTok{  )}
\end{Highlighting}
\end{Shaded}

\pandocbounded{\includegraphics[keepaspectratio]{Project_files/figure-latex/unnamed-chunk-187-1.pdf}}

\end{document}
